<!DOCTYPE html><html class="no-js" lang="en"><head>
<meta charset="utf-8">
<meta content="width=device-width,initial-scale=1" name="viewport">
<link href="https://msk-access.github.io/krewlyzer/dev/features/variant/mfsd/" rel="canonical">
<link href="../../regulatory/region-mds/" rel="prev">
<link href="../../methylation/uxm/" rel="next">
<link href="../../../assets/images/favicon.png" rel="icon">
<meta content="mkdocs-1.6.1, mkdocs-material-9.7.3" name="generator">
<title>mFSD - Krewlyzer</title>
<link href="../../../assets/stylesheets/main.484c7ddc.min.css" rel="stylesheet">
<link href="../../../assets/stylesheets/palette.ab4e12ef.min.css" rel="stylesheet">
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect">
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet">
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<link href="../../../css/timeago.css" rel="stylesheet">
<link href="../../../stylesheets/extra.css" rel="stylesheet">
<link href="../../../assets/stylesheets/panzoom.css" rel="stylesheet">
<script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<meta content="website" property="og:type">
<meta content="mFSD - Krewlyzer" property="og:title">
<meta content="https://msk-access.github.io/krewlyzer/dev/assets/images/social/features/variant/mfsd.png" property="og:image">
<meta content="image/png" property="og:image:type">
<meta content="1200" property="og:image:width">
<meta content="630" property="og:image:height">
<meta content="https://msk-access.github.io/krewlyzer/dev/features/variant/mfsd/" property="og:url">
<meta content="summary_large_image" property="twitter:card">
<meta content="mFSD - Krewlyzer" property="twitter:title">
<meta content="https://msk-access.github.io/krewlyzer/dev/assets/images/social/features/variant/mfsd.png" property="twitter:image">
<link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style><meta name="panzoom-theme" content="material">
<meta name="panzoom-data" content='{"selectors": [".mermaid", ".d2"], "initial_zoom_level": 1.0}'> </head>
<body data-md-color-accent="custom" data-md-color-primary="custom" data-md-color-scheme="default" dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox">
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox">
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#mutant-fragment-size-distribution-mfsd">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<div data-md-color-scheme="default" data-md-component="outdated" hidden="">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Krewlyzer" class="md-header__button md-logo" data-md-component="logo" href="../../.." title="Krewlyzer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Krewlyzer
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              mFSD
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="custom" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="custom" data-md-color-scheme="default" id="__palette_0" name="__palette" type="radio">
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
</label>
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="custom" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="custom" data-md-color-scheme="slate" id="__palette_1" name="__palette" type="radio">
<label class="md-header__button md-icon" for="__palette_0" hidden="" title="Switch to light mode">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
</label>
</form>
<script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
<label class="md-header__button md-icon" for="__search">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text">
<label class="md-search__icon md-icon" for="__search">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/msk-access/krewlyzer" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Krewlyzer" class="md-nav__button md-logo" data-md-component="logo" href="../../.." title="Krewlyzer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
    Krewlyzer
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/msk-access/krewlyzer" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../..">
<span class="md-ellipsis">
    
  
    Home
  

    
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../getting-started/">
<span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/installation/">
<span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/quickstart/">
<span class="md-ellipsis">
    
  
    Quickstart
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/concepts/">
<span class="md-ellipsis">
    
  
    Concepts
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../cli/">
<span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
            
  
    CLI Reference
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../cli/run-all/">
<span class="md-ellipsis">
    
  
    run-all
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../nextflow/">
<span class="md-ellipsis">
    
  
    Nextflow Pipeline
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
            
  
    Nextflow Pipeline
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/samplesheet/">
<span class="md-ellipsis">
    
  
    Samplesheet
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/parameters/">
<span class="md-ellipsis">
    
  
    Parameters
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/outputs/">
<span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/examples/">
<span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../">
<span class="md-ellipsis">
    
  
    Features
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="true" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
            
  
    Features
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_2" type="checkbox">
<label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Core
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_2_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_2">
<span class="md-nav__icon md-icon"></span>
            
  
    Core
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/extract/">
<span class="md-ellipsis">
    
  
    Fragment Extraction
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/fsc/">
<span class="md-ellipsis">
    
  
    FSC
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/fsd/">
<span class="md-ellipsis">
    
  
    FSD
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/fsr/">
<span class="md-ellipsis">
    
  
    FSR
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/wps/">
<span class="md-ellipsis">
    
  
    WPS
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_3" type="checkbox">
<label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Regulatory
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_3_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_3">
<span class="md-nav__icon md-icon"></span>
            
  
    Regulatory
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/motif/">
<span class="md-ellipsis">
    
  
    End Motif
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/ocf/">
<span class="md-ellipsis">
    
  
    OCF
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/region-entropy/">
<span class="md-ellipsis">
    
  
    Region Entropy
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/region-mds/">
<span class="md-ellipsis">
    
  
    Region MDS
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5_4" type="checkbox">
<label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Variant
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_5_4_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_4">
<span class="md-nav__icon md-icon"></span>
            
  
    Variant
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox">
<label class="md-nav__link md-nav__link--active" for="__toc">
<span class="md-ellipsis">
    
  
    mFSD
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
<span class="md-ellipsis">
    
  
    mFSD
  

    
  </span>
</a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#purpose">
<span class="md-ellipsis">
      
        Purpose
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#processing-flowchart">
<span class="md-ellipsis">
      
        Processing Flowchart
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-context">
<span class="md-ellipsis">
      
        Biological Context
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#variant-types-supported">
<span class="md-ellipsis">
      
        Variant Types Supported
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usage">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dual-bam-support">
<span class="md-ellipsis">
      
        Dual BAM Support
      
    </span>
</a>
<nav aria-label="Dual BAM Support" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#via-run-all-cli">
<span class="md-ellipsis">
      
        Via run-all CLI
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#via-nextflow-samplesheet">
<span class="md-ellipsis">
      
        Via Nextflow Samplesheet
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cli-options">
<span class="md-ellipsis">
      
        CLI Options
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#duplex-sequencing-support">
<span class="md-ellipsis">
      
        Duplex Sequencing Support
      
    </span>
</a>
<nav aria-label="Duplex Sequencing Support" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#supported-duplex-formats">
<span class="md-ellipsis">
      
        Supported Duplex Formats
      
    </span>
</a>
<nav aria-label="Supported Duplex Formats" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#xs2-fgbiopicard">
<span class="md-ellipsis">
      
        XS2: fgbio/Picard
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#xs1-marianas">
<span class="md-ellipsis">
      
        XS1: Marianas
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#duplex-weighting-formula">
<span class="md-ellipsis">
      
        Duplex Weighting Formula
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#interpretation-in-duplex-mode">
<span class="md-ellipsis">
      
        Interpretation in Duplex Mode
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#log-likelihood-ratio-llr-scoring">
<span class="md-ellipsis">
      
        Log-Likelihood Ratio (LLR) Scoring
      
    </span>
</a>
<nav aria-label="Log-Likelihood Ratio (LLR) Scoring" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#llr-formula">
<span class="md-ellipsis">
      
        LLR Formula
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#llr-output-columns">
<span class="md-ellipsis">
      
        LLR Output Columns
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#llr-interpretation-guide">
<span class="md-ellipsis">
      
        LLR Interpretation Guide
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#clinical-example-mrd-detection">
<span class="md-ellipsis">
      
        Clinical Example: MRD Detection
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cross-species-and-assay-support">
<span class="md-ellipsis">
      
        Cross-Species and Assay Support
      
    </span>
</a>
<nav aria-label="Cross-Species and Assay Support" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#built-in-presets">
<span class="md-ellipsis">
      
        Built-in Presets
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-rationale">
<span class="md-ellipsis">
      
        Biological Rationale
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#python-api-access">
<span class="md-ellipsis">
      
        Python API Access
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#fragment-classification">
<span class="md-ellipsis">
      
        Fragment Classification
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#formulas">
<span class="md-ellipsis">
      
        Formulas
      
    </span>
</a>
<nav aria-label="Formulas" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#ks-test-kolmogorov-smirnov">
<span class="md-ellipsis">
      
        KS Test (Kolmogorov-Smirnov)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#size-delta">
<span class="md-ellipsis">
      
        Size Delta
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#vaf-proxy">
<span class="md-ellipsis">
      
        VAF Proxy
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-format">
<span class="md-ellipsis">
      
        Output Format
      
    </span>
</a>
<nav aria-label="Output Format" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#main-output-samplemfsdtsv">
<span class="md-ellipsis">
      
        Main Output: {sample}.mFSD.tsv
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-samplemfsddistributionstsv">
<span class="md-ellipsis">
      
        Optional: {sample}.mFSD.distributions.tsv
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#clinical-interpretation">
<span class="md-ellipsis">
      
        Clinical Interpretation
      
    </span>
</a>
<nav aria-label="Clinical Interpretation" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#mrd-settings">
<span class="md-ellipsis">
      
        MRD Settings
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#see-also">
<span class="md-ellipsis">
      
        See Also
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_5" type="checkbox">
<label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Methylation
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_5_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_5">
<span class="md-nav__icon md-icon"></span>
            
  
    Methylation
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../methylation/uxm/">
<span class="md-ellipsis">
    
  
    UXM
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_6" type="checkbox">
<label class="md-nav__link" for="__nav_5_6" id="__nav_5_6_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Output
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_6_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_6">
<span class="md-nav__icon md-icon"></span>
            
  
    Output
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../output/json-output/">
<span class="md-ellipsis">
    
  
    JSON Export
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox">
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="">
<span class="md-ellipsis">
    
  
    Guides
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
            
  
    Guides
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/panel-mode/">
<span class="md-ellipsis">
    
  
    Panel Mode
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/msk-access-quickstart/">
<span class="md-ellipsis">
    
  
    MSK-ACCESS Quickstart
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/building-pon/">
<span class="md-ellipsis">
    
  
    Building PON
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/gc-correction/">
<span class="md-ellipsis">
    
  
    GC Correction
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox">
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="">
<span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/pon-models/">
<span class="md-ellipsis">
    
  
    PON Models
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/input-formats/">
<span class="md-ellipsis">
    
  
    Input Formats
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/output-files/">
<span class="md-ellipsis">
    
  
    Output Files
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/architecture/">
<span class="md-ellipsis">
    
  
    Architecture
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/glossary/">
<span class="md-ellipsis">
    
  
    Glossary
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_8" type="checkbox">
<label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="">
<span class="md-ellipsis">
    
  
    Development
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_8_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_8">
<span class="md-nav__icon md-icon"></span>
            
  
    Development
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/developer-guide/">
<span class="md-ellipsis">
    
  
    Developer Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/testing-guide/">
<span class="md-ellipsis">
    
  
    Testing Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/release-guide/">
<span class="md-ellipsis">
    
  
    Release Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/changelog/">
<span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/contributing/">
<span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/contributing-data/">
<span class="md-ellipsis">
    
  
    Contributing Data
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_9" type="checkbox">
<label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="">
<span class="md-ellipsis">
    
  
    Resources
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_9_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_9">
<span class="md-nav__icon md-icon"></span>
            
  
    Resources
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/overview/">
<span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/citation/">
<span class="md-ellipsis">
    
  
    Citation
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/troubleshooting/">
<span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#purpose">
<span class="md-ellipsis">
      
        Purpose
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#processing-flowchart">
<span class="md-ellipsis">
      
        Processing Flowchart
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-context">
<span class="md-ellipsis">
      
        Biological Context
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#variant-types-supported">
<span class="md-ellipsis">
      
        Variant Types Supported
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usage">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#dual-bam-support">
<span class="md-ellipsis">
      
        Dual BAM Support
      
    </span>
</a>
<nav aria-label="Dual BAM Support" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#via-run-all-cli">
<span class="md-ellipsis">
      
        Via run-all CLI
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#via-nextflow-samplesheet">
<span class="md-ellipsis">
      
        Via Nextflow Samplesheet
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cli-options">
<span class="md-ellipsis">
      
        CLI Options
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#duplex-sequencing-support">
<span class="md-ellipsis">
      
        Duplex Sequencing Support
      
    </span>
</a>
<nav aria-label="Duplex Sequencing Support" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#supported-duplex-formats">
<span class="md-ellipsis">
      
        Supported Duplex Formats
      
    </span>
</a>
<nav aria-label="Supported Duplex Formats" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#xs2-fgbiopicard">
<span class="md-ellipsis">
      
        XS2: fgbio/Picard
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#xs1-marianas">
<span class="md-ellipsis">
      
        XS1: Marianas
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#duplex-weighting-formula">
<span class="md-ellipsis">
      
        Duplex Weighting Formula
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#interpretation-in-duplex-mode">
<span class="md-ellipsis">
      
        Interpretation in Duplex Mode
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#log-likelihood-ratio-llr-scoring">
<span class="md-ellipsis">
      
        Log-Likelihood Ratio (LLR) Scoring
      
    </span>
</a>
<nav aria-label="Log-Likelihood Ratio (LLR) Scoring" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#llr-formula">
<span class="md-ellipsis">
      
        LLR Formula
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#llr-output-columns">
<span class="md-ellipsis">
      
        LLR Output Columns
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#llr-interpretation-guide">
<span class="md-ellipsis">
      
        LLR Interpretation Guide
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#clinical-example-mrd-detection">
<span class="md-ellipsis">
      
        Clinical Example: MRD Detection
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cross-species-and-assay-support">
<span class="md-ellipsis">
      
        Cross-Species and Assay Support
      
    </span>
</a>
<nav aria-label="Cross-Species and Assay Support" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#built-in-presets">
<span class="md-ellipsis">
      
        Built-in Presets
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-rationale">
<span class="md-ellipsis">
      
        Biological Rationale
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#python-api-access">
<span class="md-ellipsis">
      
        Python API Access
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#fragment-classification">
<span class="md-ellipsis">
      
        Fragment Classification
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#formulas">
<span class="md-ellipsis">
      
        Formulas
      
    </span>
</a>
<nav aria-label="Formulas" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#ks-test-kolmogorov-smirnov">
<span class="md-ellipsis">
      
        KS Test (Kolmogorov-Smirnov)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#size-delta">
<span class="md-ellipsis">
      
        Size Delta
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#vaf-proxy">
<span class="md-ellipsis">
      
        VAF Proxy
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-format">
<span class="md-ellipsis">
      
        Output Format
      
    </span>
</a>
<nav aria-label="Output Format" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#main-output-samplemfsdtsv">
<span class="md-ellipsis">
      
        Main Output: {sample}.mFSD.tsv
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#optional-samplemfsddistributionstsv">
<span class="md-ellipsis">
      
        Optional: {sample}.mFSD.distributions.tsv
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#clinical-interpretation">
<span class="md-ellipsis">
      
        Clinical Interpretation
      
    </span>
</a>
<nav aria-label="Clinical Interpretation" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#mrd-settings">
<span class="md-ellipsis">
      
        MRD Settings
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#see-also">
<span class="md-ellipsis">
      
        See Also
      
    </span>
</a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<nav aria-label="Navigation" class="md-path">
<ol class="md-path__list">
<li class="md-path__item">
<a class="md-path__link" href="../../..">
<span class="md-ellipsis">
    Home
  </span>
</a>
</li>
<li class="md-path__item">
<a class="md-path__link" href="../../">
<span class="md-ellipsis">
    Features
  </span>
</a>
</li>
<li class="md-path__item">
<a class="md-path__link" href="./">
<span class="md-ellipsis">
    Variant
  </span>
</a>
</li>
</ol>
</nav>
<article class="md-content__inner md-typeset">
<h1 id="mutant-fragment-size-distribution-mfsd">Mutant Fragment Size Distribution (<abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr>)</h1>
<p><strong>Command</strong>: <code>krewlyzer mfsd</code></p>
<blockquote>
<p><strong>Plain English</strong>: <abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> compares fragment sizes at known mutation sites.
Fragments carrying the mutation (ALT) tend to be shorter than healthy fragments (REF).</p>
<p><strong>Use case</strong>: MRD monitoring - track tumor DNA by comparing mutant vs. wild-type fragment sizes.</p>
</blockquote>
<hr>
<h2 id="purpose">Purpose</h2>
<p>Compares the size distribution of mutant vs. wild-type fragments at variant sites, with support for all small variant types and 4-way fragment classification.</p>
<hr>
<h2 id="processing-flowchart">Processing Flowchart</h2>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart LR
    BAM["sample.bam"] --&gt; RUST["Rust Backend"]
    VCF["variants.vcf/maf"] --&gt; RUST
    REF["Reference FASTA"] --&gt; RUST

    RUST --&gt; MFSD["mFSD.tsv"]

    subgraph "Per Variant"
        RUST --&gt; CLASS{"Classify Reads"}
        CLASS --&gt; REF_F["REF fragments"]
        CLASS --&gt; ALT_F["ALT fragments"]
        CLASS --&gt; NONREF["NonREF"]
        CLASS --&gt; N_F["N fragments"]
    end

    subgraph "Statistics"
        REF_F --&gt; KS["KS Test"]
        ALT_F --&gt; KS
        KS --&gt; PVAL["p-value, Delta"]
    end
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<hr>
<h2 id="biological-context">Biological Context</h2>
<p>Mutant <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> fragments are typically <strong>shorter</strong> (~145bp) than wild-type <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> (~166bp) due to:
- Altered nucleosome positioning in tumor cells
- Different chromatin accessibility
- Enhanced apoptosis patterns</p>
<p>This module quantifies this difference using high-depth targeted sequencing data.</p>
<hr>
<h2 id="variant-types-supported">Variant Types Supported</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Example</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>SNV</strong></td>
<td>A&gt;T</td>
<td>Single nucleotide variant</td>
</tr>
<tr>
<td><strong>MNV</strong></td>
<td>AT&gt;<abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr></td>
<td>Multi-nucleotide variant</td>
</tr>
<tr>
<td><strong>Insertion</strong></td>
<td>A&gt;ATG</td>
<td>Pure insertion</td>
</tr>
<tr>
<td><strong>Deletion</strong></td>
<td>ATG&gt;A</td>
<td>Pure deletion</td>
</tr>
<tr>
<td><strong>Complex</strong></td>
<td>ATG&gt;CT</td>
<td>Mixed substitution + indel</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="usage">Usage</h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1"># Basic usage with VCF</span>
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>krewlyzer<span class="w"> </span>mfsd<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-V<span class="w"> </span>variants.vcf<span class="w"> </span>-o<span class="w"> </span>output_dir/
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="c1"># With MAF file and GC correction</span>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>krewlyzer<span class="w"> </span>mfsd<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-V<span class="w"> </span>variants.maf<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a><span class="w">    </span>-r<span class="w"> </span>hg19.fa<span class="w"> </span>--correction-factors<span class="w"> </span>factors.tsv
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a><span class="c1"># With per-variant distributions</span>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a>krewlyzer<span class="w"> </span>mfsd<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-V<span class="w"> </span>variants.vcf<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a><span class="w">    </span>--output-distributions
</code></pre></div>
<hr>
<h2 id="dual-bam-support">Dual <abbr title="Binary Alignment Map - Compressed sequence alignment file format">BAM</abbr> Support</h2>
<p>For optimal results with duplex sequencing panels (e.g., MSK-ACCESS), use separate BAMs:</p>
<table>
<thead>
<tr>
<th><abbr title="Binary Alignment Map - Compressed sequence alignment file format">BAM</abbr> Type</th>
<th>Use Case</th>
<th>Why</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>all_unique</code></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr>, <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr>, <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr></td>
<td>Maximum coverage for background features</td>
</tr>
<tr>
<td><code>duplex</code></td>
<td><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr></td>
<td>Highest accuracy for variant detection</td>
</tr>
</tbody>
</table>
<h3 id="via-run-all-cli">Via run-all CLI</h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="c1"># Provide dedicated duplex BAM for mFSD</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>-i<span class="w"> </span>sample.all_unique.bam<span class="w"> </span>--mfsd-bam<span class="w"> </span>sample.duplex.bam<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="w">    </span>-r<span class="w"> </span>hg19.fa<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span>--assay<span class="w"> </span>xs2<span class="w"> </span>--variants<span class="w"> </span>sample.maf
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>When <code>--mfsd-bam</code> is provided, duplex weighting is <strong>auto-enabled</strong>.
If no duplex tags (cD/Marianas) are found, a warning is logged but processing continues with weight=1.0.</p>
</div>
<h3 id="via-nextflow-samplesheet">Via Nextflow Samplesheet</h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>sample,bam,mfsd_bam,meth_bam,vcf,bed,maf,single_sample_maf,assay,pon,targets
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>ACCESS_001,/path/to/sample.all_unique.bam,/path/to/sample.duplex.bam,,,/path/to/variants.maf,true,XS2,,
</code></pre></div>
<hr>
<h2 id="cli-options">CLI Options</h2>
<table>
<thead>
<tr>
<th>Option</th>
<th>Short</th>
<th>Type</th>
<th>Default</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--input</code></td>
<td><code>-i</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>Input <abbr title="Binary Alignment Map - Compressed sequence alignment file format">BAM</abbr> file</td>
</tr>
<tr>
<td><code>--variants</code></td>
<td><code>-V</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>VCF or MAF file with variants</td>
</tr>
<tr>
<td><code>--output</code></td>
<td><code>-o</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>Output directory</td>
</tr>
<tr>
<td><code>--sample-name</code></td>
<td><code>-s</code></td>
<td>TEXT</td>
<td></td>
<td>Override sample name</td>
</tr>
<tr>
<td><code>--reference</code></td>
<td><code>-r</code></td>
<td>PATH</td>
<td></td>
<td>Reference FASTA for <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction</td>
</tr>
<tr>
<td><code>--correction-factors</code></td>
<td><code>-F</code></td>
<td>PATH</td>
<td></td>
<td>Pre-computed correction factors CSV</td>
</tr>
<tr>
<td><code>--duplex</code></td>
<td><code>-D</code></td>
<td>FLAG</td>
<td></td>
<td>Enable duplex consensus weighting (fgbio/Marianas)</td>
</tr>
<tr>
<td><code>--mapq</code></td>
<td><code>-q</code></td>
<td>INT</td>
<td>20</td>
<td>Minimum mapping quality</td>
</tr>
<tr>
<td><code>--minlen</code></td>
<td></td>
<td>INT</td>
<td>65</td>
<td>Minimum fragment length</td>
</tr>
<tr>
<td><code>--maxlen</code></td>
<td></td>
<td>INT</td>
<td>1000</td>
<td>Maximum fragment length</td>
</tr>
<tr>
<td><code>--skip-duplicates</code></td>
<td></td>
<td>FLAG</td>
<td>True</td>
<td>Skip duplicate reads (always enabled in backend)</td>
</tr>
<tr>
<td><code>--require-proper-pair</code></td>
<td></td>
<td>FLAG</td>
<td>False</td>
<td>Require proper pairs (disable for duplex BAMs)</td>
</tr>
<tr>
<td><code>--output-distributions</code></td>
<td><code>-d</code></td>
<td>FLAG</td>
<td></td>
<td>Output per-variant size distributions</td>
</tr>
<tr>
<td><code>--verbose</code></td>
<td><code>-v</code></td>
<td>FLAG</td>
<td></td>
<td>Enable verbose logging</td>
</tr>
<tr>
<td><code>--threads</code></td>
<td><code>-t</code></td>
<td>INT</td>
<td>0</td>
<td>Number of threads (0=all)</td>
</tr>
</tbody>
</table>
<hr>
<div class="admonition note">
<p class="admonition-title">Base Quality Filtering</p>
<p>Base quality filtering (<code>--min-baseq</code>) is available when <abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> is invoked via <code>krewlyzer run-all --min-baseq &lt;N&gt;</code>.
The standalone <code>krewlyzer mfsd</code> command does not expose this flag directly.</p>
</div>
<hr>
<h2 id="duplex-sequencing-support">Duplex Sequencing Support</h2>
<p><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> supports <strong>duplex consensus sequencing</strong> for ultra-sensitive variant detection. When <code>--duplex</code> is enabled, fragments from high-confidence duplex families receive higher weight in statistical calculations.</p>
<h3 id="supported-duplex-formats">Supported Duplex Formats</h3>
<h4 id="xs2-fgbiopicard">XS2: fgbio/Picard</h4>
<p>Uses SAM auxiliary tags set by <a href="https://fulcrumgenomics.github.io/fgbio/tools/latest/CallDuplexConsensusReads.html">fgbio CallDuplexConsensusReads</a>:</p>
<table>
<thead>
<tr>
<th>Tag</th>
<th>Meaning</th>
<th>Use in <abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr></th>
</tr>
</thead>
<tbody>
<tr>
<td><code>aD</code></td>
<td>Max depth of strand A single-strand consensus</td>
<td>SS-A depth</td>
</tr>
<tr>
<td><code>bD</code></td>
<td>Max depth of strand B single-strand consensus</td>
<td>SS-B depth</td>
</tr>
<tr>
<td><code>cD</code></td>
<td><strong>Max depth of duplex consensus</strong></td>
<td><strong>Primary weight source</strong></td>
</tr>
<tr>
<td><code>aM/bM/cM</code></td>
<td>Min depth at any point</td>
<td>Quality check</td>
</tr>
<tr>
<td><code>aE/bE/cE</code></td>
<td>Error rate</td>
<td>Quality filter</td>
</tr>
</tbody>
</table>
<p><strong>Example SAM record:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>read1   0   chr1   1000   60   100M   *   0   150   ACGT...   IIII...   cD:i:5   aD:i:8   bD:i:7
</code></pre></div><p></p>
<h4 id="xs1-marianas">XS1: Marianas</h4>
<p>Encodes family size in the read name per <a href="https://cmo-ci.gitbook.io/marianas/read-name-information">Marianas documentation</a>:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>Marianas:UMI1+UMI2:contig:start:posCount:negCount:contig2:start2:pos2:neg2
</code></pre></div>
<table>
<thead>
<tr>
<th>Position</th>
<th>Field</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>0</td>
<td><code>Marianas</code></td>
<td>Marianas</td>
</tr>
<tr>
<td>1</td>
<td>UMI</td>
<td>ACT+TTA</td>
</tr>
<tr>
<td>2</td>
<td>Read1 contig</td>
<td>2</td>
</tr>
<tr>
<td>3</td>
<td>Read1 start</td>
<td>48033828</td>
</tr>
<tr>
<td>4</td>
<td>Read1 (+) count</td>
<td>4</td>
</tr>
<tr>
<td>5</td>
<td>Read1 (-) count</td>
<td>3</td>
</tr>
</tbody>
</table>
<p><strong>Family size = posCount + negCount</strong> (e.g., 4 + 3 = 7)</p>
<h3 id="duplex-weighting-formula">Duplex Weighting Formula</h3>
<p>When <code>--duplex</code> is enabled, each fragment receives a weight based on its duplex family size:</p>
<div class="arithmatex">\[
\text{weight} = \max(\ln(\text{family_size}), 1.0)
\]</div>
<table>
<thead>
<tr>
<th>Family Size</th>
<th>Weight</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>1.0</td>
<td>Single read (no UMI collapse)</td>
</tr>
<tr>
<td>2</td>
<td>1.0</td>
<td>Minimum duplex (capped)</td>
</tr>
<tr>
<td>3</td>
<td>1.1</td>
<td>Low-confidence duplex</td>
</tr>
<tr>
<td>5</td>
<td>1.6</td>
<td>Moderate confidence</td>
</tr>
<tr>
<td>10</td>
<td>2.3</td>
<td>High confidence</td>
</tr>
<tr>
<td>50</td>
<td>3.9</td>
<td>Very high confidence</td>
</tr>
</tbody>
</table>
<h3 id="interpretation-in-duplex-mode">Interpretation in Duplex Mode</h3>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p>When <code>--duplex</code> is enabled, <code>ALT_Weighted</code> and <code>REF_Weighted</code> will exceed raw counts.
A ratio of <strong>~1.6x</strong> indicates typical duplex family sizes of 3-5.</p>
</div>
<table>
<thead>
<tr>
<th>Column</th>
<th>Without Duplex</th>
<th>With Duplex</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>ALT_Count</code></td>
<td>Raw fragment count</td>
<td>Raw fragment count</td>
</tr>
<tr>
<td><code>ALT_Weighted</code></td>
<td>= ALT_Count</td>
<td>= (weight  fragment)</td>
</tr>
<tr>
<td><code>VAF_GC_Corrected</code></td>
<td>Based on raw</td>
<td>Based on weighted counts</td>
</tr>
</tbody>
</table>
<p><strong>Example:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>Variant at chr1:156845927
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>  ALT_Count    = 393
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>  ALT_Weighted = 691  (ratio = 1.76)
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>   Indicates high-confidence duplex families
</code></pre></div><p></p>
<hr>
<h2 id="log-likelihood-ratio-llr-scoring">Log-Likelihood Ratio (LLR) Scoring</h2>
<p>For <strong>low fragment count scenarios</strong> (common in duplex/panel sequencing), traditional KS tests are unreliable. <abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> provides LLR scoring as a probabilistic alternative.</p>
<h3 id="llr-formula">LLR Formula</h3>
<div class="arithmatex">\[
\text{LLR} = \sum_{i=1}^{n} \left[ \log P(\text{size}_i | \text{Tumor}) - \log P(\text{size}_i | \text{Healthy}) \right]
\]</div>
<p>Using Gaussian models:
- <strong>Healthy</strong>:  = 167bp,  = 35bp (nucleosomal periodicity)
- <strong>Tumor</strong>:  = 145bp,  = 25bp (sub-nucleosomal fragments)</p>
<h3 id="llr-output-columns">LLR Output Columns</h3>
<table>
<thead>
<tr>
<th>Column</th>
<th>Range</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>ALT_LLR</code></td>
<td>any</td>
<td>Log-likelihood ratio for ALT fragments</td>
</tr>
<tr>
<td><code>REF_LLR</code></td>
<td>any</td>
<td>Log-likelihood ratio for REF fragments</td>
</tr>
</tbody>
</table>
<h3 id="llr-interpretation-guide">LLR Interpretation Guide</h3>
<table>
<thead>
<tr>
<th>LLR Value</th>
<th>Interpretation</th>
<th>Action</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>&gt; 5</strong></td>
<td>Strong tumor signal</td>
<td>High confidence in tumor-derived fragments</td>
</tr>
<tr>
<td><strong>0 to 5</strong></td>
<td>Weak tumor signal</td>
<td>Possible tumor, verify with other evidence</td>
</tr>
<tr>
<td><strong>-5 to 0</strong></td>
<td>Weak healthy signal</td>
<td>Likely healthy, low tumor content</td>
</tr>
<tr>
<td><strong>&lt; -5</strong></td>
<td>Strong healthy signal</td>
<td>Consistent with healthy <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr></td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For low-N variants (ALT_Count &lt; 5), use <code>ALT_LLR</code> instead of <code>KS_Pval_ALT_REF</code>.
The LLR is robust with even 1-2 fragments, while KS tests require 5.</p>
</div>
<h3 id="clinical-example-mrd-detection">Clinical Example: MRD Detection</h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>Variant at TP53:chr17:7577539
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a>  ALT_Count = 3           # Too few for KS test
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a>  KS_Valid  = FALSE       # KS test unreliable
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a>  ALT_LLR   = 4.2         # Positive = tumor-like fragments
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a>  REF_LLR   = -89.5       # Negative = healthy-like REF population
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a>
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a>   Interpretation: ALT fragments show tumor signature despite low count
</code></pre></div>
<hr>
<h2 id="cross-species-and-assay-support">Cross-Species and Assay Support</h2>
<p>The LLR model uses Gaussian distributions for healthy and tumor fragment length peaks. These parameters can be customized for different species or library preparations.</p>
<h3 id="built-in-presets">Built-in Presets</h3>
<table>
<thead>
<tr>
<th>Preset</th>
<th>Healthy </th>
<th>Healthy </th>
<th>Tumor </th>
<th>Tumor </th>
<th>Use Case</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>human</strong> (default)</td>
<td>167bp</td>
<td>35bp</td>
<td>145bp</td>
<td>25bp</td>
<td>Human <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr></td>
</tr>
<tr>
<td><strong>canine</strong></td>
<td>153bp</td>
<td>30bp</td>
<td>135bp</td>
<td>22bp</td>
<td>Canine <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr></td>
</tr>
<tr>
<td><strong>ssdna</strong></td>
<td>160bp</td>
<td>40bp</td>
<td>140bp</td>
<td>30bp</td>
<td>Single-stranded library prep</td>
</tr>
</tbody>
</table>
<h3 id="biological-rationale">Biological Rationale</h3>
<p>Fragment length peaks vary across species due to:
- <strong>Nucleosome spacing differences</strong> - Canine nucleosomes are more tightly packed
- <strong>Chromatin structure</strong> - Different histone modifications
- <strong>Library preparation</strong> - ssDNA preps capture shorter fragments</p>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart LR
    subgraph "Human cfDNA"
        H_HEALTHY["Healthy: ~167bp"] 
        H_TUMOR["Tumor: ~145bp"]
    end

    subgraph "Canine cfDNA"
        C_HEALTHY["Healthy: ~153bp"]
        C_TUMOR["Tumor: ~135bp"]
    end

    H_HEALTHY --&gt; DELTA1[" = -22bp"]
    H_TUMOR --&gt; DELTA1
    C_HEALTHY --&gt; DELTA2[" = -18bp"]
    C_TUMOR --&gt; DELTA2
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<h3 id="python-api-access">Python API Access</h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="kn">from</span><span class="w"> </span><span class="nn">krewlyzer._core</span><span class="w"> </span><span class="kn">import</span> <span class="n">LLRModelParams</span>
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a><span class="c1"># Use built-in presets</span>
<a href="#__codelineno-7-4" id="__codelineno-7-4" name="__codelineno-7-4"></a><span class="n">human_params</span> <span class="o">=</span> <span class="n">LLRModelParams</span><span class="o">.</span><span class="n">human</span><span class="p">()</span>
<a href="#__codelineno-7-5" id="__codelineno-7-5" name="__codelineno-7-5"></a><span class="n">canine_params</span> <span class="o">=</span> <span class="n">LLRModelParams</span><span class="o">.</span><span class="n">canine</span><span class="p">()</span>
<a href="#__codelineno-7-6" id="__codelineno-7-6" name="__codelineno-7-6"></a><span class="n">ssdna_params</span> <span class="o">=</span> <span class="n">LLRModelParams</span><span class="o">.</span><span class="n">ssdna</span><span class="p">()</span>
<a href="#__codelineno-7-7" id="__codelineno-7-7" name="__codelineno-7-7"></a>
<a href="#__codelineno-7-8" id="__codelineno-7-8" name="__codelineno-7-8"></a><span class="c1"># Custom parameters</span>
<a href="#__codelineno-7-9" id="__codelineno-7-9" name="__codelineno-7-9"></a><span class="n">custom</span> <span class="o">=</span> <span class="n">LLRModelParams</span><span class="p">(</span>
<a href="#__codelineno-7-10" id="__codelineno-7-10" name="__codelineno-7-10"></a>    <span class="n">healthy_mu</span><span class="o">=</span><span class="mf">160.0</span><span class="p">,</span>
<a href="#__codelineno-7-11" id="__codelineno-7-11" name="__codelineno-7-11"></a>    <span class="n">healthy_sigma</span><span class="o">=</span><span class="mf">32.0</span><span class="p">,</span>
<a href="#__codelineno-7-12" id="__codelineno-7-12" name="__codelineno-7-12"></a>    <span class="n">tumor_mu</span><span class="o">=</span><span class="mf">140.0</span><span class="p">,</span>
<a href="#__codelineno-7-13" id="__codelineno-7-13" name="__codelineno-7-13"></a>    <span class="n">tumor_sigma</span><span class="o">=</span><span class="mf">20.0</span>
<a href="#__codelineno-7-14" id="__codelineno-7-14" name="__codelineno-7-14"></a><span class="p">)</span>
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>CLI support for preset selection is planned for a future release.
Currently, the Rust backend defaults to human parameters.</p>
</div>
<hr>
<h2 id="fragment-classification">Fragment Classification</h2>
<p>Fragments are classified into 4 categories:</p>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart TB
    READ["Fragment at variant site"] --&gt; CHECK{"Base at variant?"}
    CHECK --&gt;|"Matches REF"| REF["REF category"]
    CHECK --&gt;|"Matches ALT"| ALT["ALT category"]
    CHECK --&gt;|"Other base"| NONREF["NonREF category"]
    CHECK --&gt;|"N base"| N["N category"]
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<table>
<thead>
<tr>
<th>Category</th>
<th>Description</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>REF</strong></td>
<td>Supports reference allele</td>
<td>Healthy <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr></td>
</tr>
<tr>
<td><strong>ALT</strong></td>
<td>Supports alternate allele</td>
<td>Tumor signal</td>
</tr>
<tr>
<td><strong>NonREF</strong></td>
<td>Other base (not REF/ALT/N)</td>
<td>Sequencing errors</td>
</tr>
<tr>
<td><strong>N</strong></td>
<td>Contains N at variant</td>
<td>Low quality</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="formulas">Formulas</h2>
<h3 id="ks-test-kolmogorov-smirnov">KS Test (Kolmogorov-Smirnov)</h3>
<div class="arithmatex">\[
\text{KS statistic} = \max |F_1(x) - F_2(x)|
\]</div>
<p>Where:
- <span class="arithmatex">\(F_1(x)\)</span> = CDF of ALT fragment sizes
- <span class="arithmatex">\(F_2(x)\)</span> = CDF of REF fragment sizes</p>
<h3 id="size-delta">Size Delta</h3>
<div class="arithmatex">\[
\Delta_{\text{ALT-REF}} = \text{ALT_MeanSize} - \text{REF_MeanSize}
\]</div>
<p><strong>Expected:</strong>
- Healthy: <span class="arithmatex">\(\approx 0\)</span>
- Cancer: <span class="arithmatex">\(&lt; 0\)</span> (ALT shorter)</p>
<h3 id="vaf-proxy">VAF Proxy</h3>
<div class="arithmatex">\[
\text{VAF_Proxy} = \frac{\text{ALT_Count}}{\text{REF_Count} + \text{ALT_Count}}
\]</div>
<hr>
<h2 id="output-format">Output Format</h2>
<h3 id="main-output-samplemfsdtsv">Main Output: <code>{sample}.mFSD.tsv</code></h3>
<table>
<thead>
<tr>
<th>Column Group</th>
<th>Columns</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Variant Info</strong> (5)</td>
<td><code>Chrom</code>, <code>Pos</code>, <code>Ref</code>, <code>Alt</code>, <code>VarType</code></td>
<td>Variant coordinates and type</td>
</tr>
<tr>
<td><strong>Raw Counts</strong> (5)</td>
<td><code>REF_Count</code>, <code>ALT_Count</code>, <code>NonREF_Count</code>, <code>N_Count</code>, <code>Total_Count</code></td>
<td>Fragment counts per allele class</td>
</tr>
<tr>
<td><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-Weighted Counts</strong> (5)</td>
<td><code>REF_Weighted</code>, <code>ALT_Weighted</code>, <code>NonREF_Weighted</code>, <code>N_Weighted</code>, <code>VAF_GC_Corrected</code></td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-bias corrected counts and VAF</td>
</tr>
<tr>
<td><strong>Log-Likelihood Ratio</strong> (2)</td>
<td><code>ALT_LLR</code>, <code>REF_LLR</code></td>
<td>Log-likelihood ratios (useful for low-N duplex variants where KS unreliable)</td>
</tr>
<tr>
<td><strong>Mean Sizes</strong> (4)</td>
<td><code>REF_MeanSize</code>, <code>ALT_MeanSize</code>, <code>NonREF_MeanSize</code>, <code>N_MeanSize</code></td>
<td>Mean fragment size per allele class</td>
</tr>
<tr>
<td><strong>KS Tests</strong> (18)</td>
<td><code>Delta_*/KS_*/KS_Pval_*</code>  6 allele pairings</td>
<td>KolmogorovSmirnov distance + p-value for: ALT-REF, ALT-NonREF, REF-NonREF, ALT-N, REF-N, NonREF-N</td>
</tr>
<tr>
<td><strong>Derived</strong> (5)</td>
<td><code>VAF_Proxy</code>, <code>Error_Rate</code>, <code>N_Rate</code>, <code>Size_Ratio</code>, <code>Quality_Score</code></td>
<td>Computed summary biomarkers</td>
</tr>
<tr>
<td><strong>Flags</strong> (2)</td>
<td><code>ALT_Confidence</code>, <code>KS_Valid</code></td>
<td>Quality indicators (<code>HIGH</code>/<code>LOW</code>/<code>NONE</code>; TRUE/FALSE)</td>
</tr>
</tbody>
</table>
<h3 id="optional-samplemfsddistributionstsv">Optional: <code>{sample}.mFSD.distributions.tsv</code></h3>
<p>With <code>--output-distributions</code>:
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>Chrom  Pos    Ref  Alt  Category  Size  Count
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a>chr1   12345  A    T    REF       145   3
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a>chr1   12345  A    T    REF       166   12
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a>chr1   12345  A    T    ALT       142   2
</code></pre></div><p></p>
<hr>
<h2 id="clinical-interpretation">Clinical Interpretation</h2>
<table>
<thead>
<tr>
<th>Metric</th>
<th>Healthy</th>
<th>Cancer (<abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr>)</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>Delta_ALT_REF</code></td>
<td>~0</td>
<td><strong>Negative</strong> (ALT shorter)</td>
</tr>
<tr>
<td><code>Size_Ratio</code></td>
<td>~1.0</td>
<td><strong>&lt; 1.0</strong></td>
</tr>
<tr>
<td><code>VAF_Proxy</code></td>
<td>0</td>
<td>&gt; 0 (correlates with TF)</td>
</tr>
</tbody>
</table>
<h3 id="mrd-settings">MRD Settings</h3>
<ul>
<li>Low fragment counts (1-2) produce <code>NA</code></li>
<li><code>ALT_Confidence</code>: HIGH (5), LOW (1-4), NONE (0)</li>
<li><code>KS_Valid</code>: TRUE if REF and ALT 2 fragments each</li>
</ul>
<hr>
<h2 id="see-also">See Also</h2>
<ul>
<li><a href="../../../reference/pon-models/"><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> Models</a>  <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction factors</li>
<li><a href="../../../resources/citation/#mfsd">Citation</a>  Scientific references</li>
</ul>
<!-- Abbreviation definitions for Krewlyzer documentation -->
<!-- These provide hover tooltips for technical acronyms -->
<aside class="md-source-file">
<span class="md-source-file__fact">
<span class="md-icon" title="Last update">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
</span>
<span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="March 1, 2026 03:15:26 UTC"><span class="timeago" datetime="2026-03-01T03:15:26+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="March 1, 2026 03:15:26 UTC">2026-03-01</span>
</span>
<span class="md-source-file__fact">
<span class="md-icon" title="Created">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
</span>
<span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="November 26, 2025 16:12:36 UTC"><span class="timeago" datetime="2025-11-26T16:12:36+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="November 26, 2025 16:12:36 UTC">2025-11-26</span>
</span>
</aside>
</article>
</div>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
<div class="md-copyright__highlight">
      Developed by <a href="https://github.com/rhshah">Ronak H Shah</a> @ 
<a href="https://www.mskcc.org">MSKCC</a> | 
Built with <a href="https://antigravity.google/">Antigravity</a> | 
<a href="https://github.com/msk-access/krewlyzer/blob/main/LICENSE">AGPL-3.0</a>
</div>
</div>
<div class="md-social">
<a class="md-social__link" href="https://github.com/msk-access/krewlyzer" rel="noopener" target="_blank" title="github.com">
<svg viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
</a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"annotate": null, "base": "../../..", "features": ["navigation.instant", "navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.indexes", "navigation.path", "toc.follow", "content.code.copy", "content.code.annotate"], "search": "../../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"default": "stable", "provider": "mike"}}</script>
<script src="../../../assets/javascripts/bundle.79ae519e.min.js"></script>
<script src="../../../js/timeago.min.js"></script>
<script src="../../../js/timeago_mkdocs_material.js"></script>
<script src="../../../javascripts/mathjax.js"></script>
<script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
<script src="../../../assets/javascripts/panzoom.min.js"></script>
<script src="../../../assets/javascripts/zoompan.js"></script>
<script type="module">import mermaid from "https://unpkg.com/mermaid@11.4.0/dist/mermaid.esm.min.mjs";
window.mermaidConfig = {default: {
    startOnLoad: false
}};</script>
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>