<!DOCTYPE html><html lang="en" class="no-js"><head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://msk-access.github.io/krewlyzer/dev/reference/output-files/">
      
      
        <link rel="prev" href="../input-formats/">
      
      
        <link rel="next" href="../architecture/">
      
      
        
      
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.3">
    
    
      
        <title>Output Files - Krewlyzer</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../css/timeago.css">
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
      <link rel="stylesheet" href="../../assets/stylesheets/panzoom.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  
<meta property="og:type" content="website">
<meta property="og:title" content="Output Files - Krewlyzer">
<meta property="og:image" content="https://msk-access.github.io/krewlyzer/dev/assets/images/social/reference/output-files.png">
<meta property="og:image:type" content="image/png">
<meta property="og:image:width" content="1200">
<meta property="og:image:height" content="630">
<meta property="og:url" content="https://msk-access.github.io/krewlyzer/dev/reference/output-files/">
<meta property="twitter:card" content="summary_large_image">
<meta property="twitter:title" content="Output Files - Krewlyzer">
<meta property="twitter:image" content="https://msk-access.github.io/krewlyzer/dev/assets/images/social/reference/output-files.png">
<link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style><meta name="panzoom-theme" content="material">
<meta name="panzoom-data" content='{"selectors": [".mermaid", ".d2"], "initial_zoom_level": 1.0}'> </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="custom">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#output-files-reference" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-color-scheme="default" data-md-component="outdated" hidden>
        
      </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Krewlyzer" class="md-header__button md-logo" aria-label="Krewlyzer" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Krewlyzer
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Output Files
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="custom" aria-label="Switch to dark mode" type="radio" name="__palette" id="__palette_0">
    
      <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
      </label>
    
  
    
    
    
    <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="custom" aria-label="Switch to light mode" type="radio" name="__palette" id="__palette_1">
    
      <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_0" hidden>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
      </label>
    
  
</form>
      
    
    
      <script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/msk-access/krewlyzer" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Krewlyzer" class="md-nav__button md-logo" aria-label="Krewlyzer" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>

    </a>
    Krewlyzer
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/msk-access/krewlyzer" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
  </div>
  <div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Home
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_2">
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../getting-started/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_2" id="__nav_2_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started/installation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started/quickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Quickstart
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../getting-started/concepts/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Concepts
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_3">
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../cli/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_3" id="__nav_3_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    CLI Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../cli/run-all/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    run-all
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_4">
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../nextflow/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Nextflow Pipeline
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_4" id="__nav_4_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Nextflow Pipeline
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../nextflow/samplesheet/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Samplesheet
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../nextflow/parameters/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Parameters
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../nextflow/outputs/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../nextflow/examples/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
          
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5">
        
          
          <div class="md-nav__link md-nav__container">
            <a href="../../features/" class="md-nav__link ">
              
  
  
  <span class="md-ellipsis">
    
  
    Features
  

    
  </span>
  
  

            </a>
            
              
              <label class="md-nav__link " for="__nav_5" id="__nav_5_label" tabindex="">
                <span class="md-nav__icon md-icon"></span>
              </label>
            
          </div>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Features
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5_2">
        
          
          <label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Core
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Core
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/core/extract/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Fragment Extraction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/core/fsc/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    FSC
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/core/fsd/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    FSD
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/core/fsr/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    FSR
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/core/wps/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    WPS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5_3">
        
          
          <label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Regulatory
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Regulatory
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/regulatory/motif/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    End Motif
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/regulatory/ocf/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    OCF
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/regulatory/region-entropy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Region Entropy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/regulatory/region-mds/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Region MDS
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5_4">
        
          
          <label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Variant
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_4_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_4">
            <span class="md-nav__icon md-icon"></span>
            
  
    Variant
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/variant/mfsd/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    mFSD
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5_5">
        
          
          <label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Methylation
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_5_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_5">
            <span class="md-nav__icon md-icon"></span>
            
  
    Methylation
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/methylation/uxm/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    UXM
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    
    
      
        
      
    
    
    
      
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_5_6">
        
          
          <label class="md-nav__link" for="__nav_5_6" id="__nav_5_6_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Output
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="2" aria-labelledby="__nav_5_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_5_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Output
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../features/output/json-output/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    JSON Export
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_6">
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Guides
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            
  
    Guides
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/panel-mode/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Panel Mode
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/msk-access-quickstart/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    MSK-ACCESS Quickstart
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/building-pon/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Building PON
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../guides/gc-correction/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    GC Correction
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" checked>
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../pon-models/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    PON Models
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../input-formats/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Input Formats
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    Output Files
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    Output Files
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-reference" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Reference
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#on-target-vs-off-target-files" class="md-nav__link">
    <span class="md-ellipsis">
      
        On-Target vs Off-Target Files
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="On-Target vs Off-Target Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#what-on-target-and-off-target-mean" class="md-nav__link">
    <span class="md-ellipsis">
      
        What "On-Target" and "Off-Target" Mean
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#why-the-split-matters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why the Split Matters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#which-to-use" class="md-nav__link">
    <span class="md-ellipsis">
      
        Which to Use
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#complete-on-target-off-target-file-inventory" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete On-Target / Off-Target File Inventory
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gc-correction-and-on-target" class="md-nav__link">
    <span class="md-ellipsis">
      
        GC Correction and On-Target
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-fragmentomics" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Fragmentomics
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Fragmentomics">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fsd-fragment-size-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSD (Fragment Size Distribution)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSD (Fragment Size Distribution)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsr-fragment-size-ratio" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSR (Fragment Size Ratio)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSR (Fragment Size Ratio)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-fragment-size-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC (Fragment Size Coverage)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC (Fragment Size Coverage)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-gene-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC Gene-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC Gene-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-region-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC Region-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC Region-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-e1-only" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC E1-Only
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC E1-Only">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-counts-pre-correction" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC Counts (Pre-Correction)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC Counts (Pre-Correction)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nucleosome-positioning" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nucleosome Positioning
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Nucleosome Positioning">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#wps-windowed-protection-score" class="md-nav__link">
    <span class="md-ellipsis">
      
        WPS (Windowed Protection Score)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="WPS (Windowed Protection Score)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-parquet" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns (Parquet)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#wps-panel" class="md-nav__link">
    <span class="md-ellipsis">
      
        WPS Panel
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="WPS Panel">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ml-use-case_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#wps-background" class="md-nav__link">
    <span class="md-ellipsis">
      
        WPS Background
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="WPS Background">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_8" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_8" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#motif-tissue-of-origin" class="md-nav__link">
    <span class="md-ellipsis">
      
        Motif &amp; Tissue-of-Origin
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Motif &amp; Tissue-of-Origin">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#endmotif" class="md-nav__link">
    <span class="md-ellipsis">
      
        EndMotif
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="EndMotif">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_9" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_9" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#endmotif-1-mer" class="md-nav__link">
    <span class="md-ellipsis">
      
        EndMotif 1-mer
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="EndMotif 1-mer">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-base-fraction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: base, fraction
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_10" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#breakpointmotif" class="md-nav__link">
    <span class="md-ellipsis">
      
        BreakPointMotif
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="BreakPointMotif">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#purpose-vs-endmotif" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose vs EndMotif
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mds-motif-diversity-score" class="md-nav__link">
    <span class="md-ellipsis">
      
        MDS (Motif Diversity Score)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MDS (Motif Diversity Score)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-mds-float" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: MDS (float)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_10" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mds-exon-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        MDS Exon-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MDS Exon-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_11" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mds-gene-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        MDS Gene-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MDS Gene-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_8" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_12" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ocf-orientation-aware-cfdna-fragmentation" class="md-nav__link">
    <span class="md-ellipsis">
      
        OCF (Orientation-aware cfDNA Fragmentation)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="OCF (Orientation-aware cfDNA Fragmentation)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_9" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_11" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_13" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ocf-sync" class="md-nav__link">
    <span class="md-ellipsis">
      
        OCF Sync
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="OCF Sync">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-label-count-mean_size-entropy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: label, count, mean_size, entropy
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tfbs-transcription-factor-binding-site-entropy" class="md-nav__link">
    <span class="md-ellipsis">
      
        TFBS (Transcription Factor Binding Site Entropy)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TFBS (Transcription Factor Binding Site Entropy)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_10" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_12" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_14" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tfbs-sync" class="md-nav__link">
    <span class="md-ellipsis">
      
        TFBS Sync
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TFBS Sync">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-label-size-count-proportion" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: label, size, count, proportion
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#atac-chromatin-accessibility-entropy" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC (Chromatin Accessibility Entropy)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="ATAC (Chromatin Accessibility Entropy)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_11" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_13" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_15" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#atac-sync" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC Sync
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="ATAC Sync">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-label-size-count-proportion_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: label, size, count, proportion
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        Variant-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Variant-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mfsd-mutant-fragment-size-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      
        mFSD (Mutant Fragment Size Distribution)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="mFSD (Mutant Fragment Size Distribution)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#column-groups-46-total" class="md-nav__link">
    <span class="md-ellipsis">
      
        Column Groups (46 total)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_14" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_16" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mfsd-distributions" class="md-nav__link">
    <span class="md-ellipsis">
      
        mFSD Distributions
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="mFSD Distributions">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-chrom-pos-ref-alt-category-size-count" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: Chrom, Pos, Ref, Alt, Category, Size, Count
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#methylation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Methylation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#uxm-methylation" class="md-nav__link">
    <span class="md-ellipsis">
      
        UXM (Methylation)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="UXM (Methylation)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_12" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_15" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_17" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diagnostic-qc-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diagnostic / QC Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Diagnostic / QC Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gc-correction-factors" class="md-nav__link">
    <span class="md-ellipsis">
      
        GC Correction Factors
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GC Correction Factors">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_13" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_16" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#metadata-json" class="md-nav__link">
    <span class="md-ellipsis">
      
        Metadata JSON
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#features-json" class="md-nav__link">
    <span class="md-ellipsis">
      
        Features JSON
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#feature-selection-guide-for-ml-models" class="md-nav__link">
    <span class="md-ellipsis">
      
        Feature Selection Guide for ML Models
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#see-also" class="md-nav__link">
    <span class="md-ellipsis">
      
        See Also
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../architecture/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Architecture
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../glossary/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Glossary
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8">
        
          
          <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Development
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_8">
            <span class="md-nav__icon md-icon"></span>
            
  
    Development
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../development/developer-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Developer Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../development/testing-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Testing Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../development/release-guide/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Release Guide
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../development/changelog/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../development/contributing/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../development/contributing-data/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Contributing Data
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
      
        
      
        
      
        
      
    
    
    
      
        
        
      
    
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
        
        
          
        
        <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_9">
        
          
          <label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="">
            
  
  
  <span class="md-ellipsis">
    
  
    Resources
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_9_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_9">
            <span class="md-nav__icon md-icon"></span>
            
  
    Resources
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/overview/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/citation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Citation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../../resources/troubleshooting/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#quick-reference" class="md-nav__link">
    <span class="md-ellipsis">
      
        Quick Reference
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#on-target-vs-off-target-files" class="md-nav__link">
    <span class="md-ellipsis">
      
        On-Target vs Off-Target Files
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="On-Target vs Off-Target Files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#what-on-target-and-off-target-mean" class="md-nav__link">
    <span class="md-ellipsis">
      
        What "On-Target" and "Off-Target" Mean
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#why-the-split-matters" class="md-nav__link">
    <span class="md-ellipsis">
      
        Why the Split Matters
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#which-to-use" class="md-nav__link">
    <span class="md-ellipsis">
      
        Which to Use
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#complete-on-target-off-target-file-inventory" class="md-nav__link">
    <span class="md-ellipsis">
      
        Complete On-Target / Off-Target File Inventory
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gc-correction-and-on-target" class="md-nav__link">
    <span class="md-ellipsis">
      
        GC Correction and On-Target
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#core-fragmentomics" class="md-nav__link">
    <span class="md-ellipsis">
      
        Core Fragmentomics
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Core Fragmentomics">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#fsd-fragment-size-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSD (Fragment Size Distribution)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSD (Fragment Size Distribution)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsr-fragment-size-ratio" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSR (Fragment Size Ratio)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSR (Fragment Size Ratio)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-fragment-size-coverage" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC (Fragment Size Coverage)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC (Fragment Size Coverage)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_2" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-gene-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC Gene-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC Gene-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_3" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-region-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC Region-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC Region-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_4" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-e1-only" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC E1-Only
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC E1-Only">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fsc-counts-pre-correction" class="md-nav__link">
    <span class="md-ellipsis">
      
        FSC Counts (Pre-Correction)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="FSC Counts (Pre-Correction)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nucleosome-positioning" class="md-nav__link">
    <span class="md-ellipsis">
      
        Nucleosome Positioning
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Nucleosome Positioning">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#wps-windowed-protection-score" class="md-nav__link">
    <span class="md-ellipsis">
      
        WPS (Windowed Protection Score)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="WPS (Windowed Protection Score)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-parquet" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns (Parquet)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#wps-panel" class="md-nav__link">
    <span class="md-ellipsis">
      
        WPS Panel
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="WPS Panel">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ml-use-case_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#wps-background" class="md-nav__link">
    <span class="md-ellipsis">
      
        WPS Background
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="WPS Background">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_5" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_8" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_8" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#motif-tissue-of-origin" class="md-nav__link">
    <span class="md-ellipsis">
      
        Motif &amp; Tissue-of-Origin
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Motif &amp; Tissue-of-Origin">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#endmotif" class="md-nav__link">
    <span class="md-ellipsis">
      
        EndMotif
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="EndMotif">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_6" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_9" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_9" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#endmotif-1-mer" class="md-nav__link">
    <span class="md-ellipsis">
      
        EndMotif 1-mer
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="EndMotif 1-mer">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-base-fraction" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: base, fraction
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_10" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#breakpointmotif" class="md-nav__link">
    <span class="md-ellipsis">
      
        BreakPointMotif
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="BreakPointMotif">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#purpose-vs-endmotif" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose vs EndMotif
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mds-motif-diversity-score" class="md-nav__link">
    <span class="md-ellipsis">
      
        MDS (Motif Diversity Score)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MDS (Motif Diversity Score)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-mds-float" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: MDS (float)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_10" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mds-exon-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        MDS Exon-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MDS Exon-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_7" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_11" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mds-gene-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        MDS Gene-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="MDS Gene-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_8" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_12" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ocf-orientation-aware-cfdna-fragmentation" class="md-nav__link">
    <span class="md-ellipsis">
      
        OCF (Orientation-aware cfDNA Fragmentation)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="OCF (Orientation-aware cfDNA Fragmentation)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_9" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_11" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_13" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ocf-sync" class="md-nav__link">
    <span class="md-ellipsis">
      
        OCF Sync
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="OCF Sync">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-label-count-mean_size-entropy" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: label, count, mean_size, entropy
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tfbs-transcription-factor-binding-site-entropy" class="md-nav__link">
    <span class="md-ellipsis">
      
        TFBS (Transcription Factor Binding Site Entropy)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TFBS (Transcription Factor Binding Site Entropy)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_10" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_12" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_14" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tfbs-sync" class="md-nav__link">
    <span class="md-ellipsis">
      
        TFBS Sync
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="TFBS Sync">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-label-size-count-proportion" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: label, size, count, proportion
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#atac-chromatin-accessibility-entropy" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC (Chromatin Accessibility Entropy)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="ATAC (Chromatin Accessibility Entropy)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_11" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_13" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_15" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#atac-sync" class="md-nav__link">
    <span class="md-ellipsis">
      
        ATAC Sync
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="ATAC Sync">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-label-size-count-proportion_1" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: label, size, count, proportion
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#variant-level" class="md-nav__link">
    <span class="md-ellipsis">
      
        Variant-Level
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Variant-Level">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mfsd-mutant-fragment-size-distribution" class="md-nav__link">
    <span class="md-ellipsis">
      
        mFSD (Mutant Fragment Size Distribution)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="mFSD (Mutant Fragment Size Distribution)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#column-groups-46-total" class="md-nav__link">
    <span class="md-ellipsis">
      
        Column Groups (46 total)
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_14" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_16" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#mfsd-distributions" class="md-nav__link">
    <span class="md-ellipsis">
      
        mFSD Distributions
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="mFSD Distributions">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns-chrom-pos-ref-alt-category-size-count" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns: Chrom, Pos, Ref, Alt, Category, Size, Count
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#methylation" class="md-nav__link">
    <span class="md-ellipsis">
      
        Methylation
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Methylation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#uxm-methylation" class="md-nav__link">
    <span class="md-ellipsis">
      
        UXM (Methylation)
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="UXM (Methylation)">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_12" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_15" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#ml-use-case_17" class="md-nav__link">
    <span class="md-ellipsis">
      
        ML Use Case
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diagnostic-qc-outputs" class="md-nav__link">
    <span class="md-ellipsis">
      
        Diagnostic / QC Outputs
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Diagnostic / QC Outputs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gc-correction-factors" class="md-nav__link">
    <span class="md-ellipsis">
      
        GC Correction Factors
      
    </span>
  </a>
  
    <nav class="md-nav" aria-label="GC Correction Factors">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#columns_13" class="md-nav__link">
    <span class="md-ellipsis">
      
        Columns
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#purpose-use-cases_16" class="md-nav__link">
    <span class="md-ellipsis">
      
        Purpose &amp; Use Cases
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#metadata-json" class="md-nav__link">
    <span class="md-ellipsis">
      
        Metadata JSON
      
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#features-json" class="md-nav__link">
    <span class="md-ellipsis">
      
        Features JSON
      
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#feature-selection-guide-for-ml-models" class="md-nav__link">
    <span class="md-ellipsis">
      
        Feature Selection Guide for ML Models
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#see-also" class="md-nav__link">
    <span class="md-ellipsis">
      
        See Also
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
                



  


  <nav class="md-path" aria-label="Navigation">
    <ol class="md-path__list">
      
        
  
  
    <li class="md-path__item">
      <a href="../.." class="md-path__link">
        
  <span class="md-ellipsis">
    Home
  </span>

      </a>
    </li>
  

      
      
        
  
  
    
    
      <li class="md-path__item">
        <a href="../pon-models/" class="md-path__link">
          
  <span class="md-ellipsis">
    Reference
  </span>

        </a>
      </li>
    
  

      
    </ol>
  </nav>

              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="output-files-reference">Output Files Reference</h1>
<p>Complete reference for every file Krewlyzer produces  what it contains, what each column means, when to use it, and how to apply it in ML models.</p>
<hr>
<h2 id="quick-reference">Quick Reference</h2>
<table>
<thead>
<tr>
<th>File</th>
<th>Feature</th>
<th>Resolution</th>
<th>ML Signal</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="#fsd-fragment-size-distribution"><code>{s}.FSD.tsv</code></a></td>
<td><abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr></td>
<td>Per chr arm</td>
<td>Arm-level fragmentation shift</td>
</tr>
<tr>
<td><a href="#fsr-fragment-size-ratio"><code>{s}.FSR.tsv</code></a></td>
<td><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr></td>
<td>5 Mb window</td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized short/long ratio</td>
</tr>
<tr>
<td><a href="#fsc-fragment-size-coverage"><code>{s}.FSC.tsv</code></a></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
<td>5 Mb window</td>
<td>Multi-channel size coverage</td>
</tr>
<tr>
<td><a href="#fsc-gene-level"><code>{s}.FSC.gene.tsv</code></a></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
<td>Per gene</td>
<td>Gene fragmentation composition</td>
</tr>
<tr>
<td><a href="#fsc-region-level"><code>{s}.FSC.regions.tsv</code></a></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
<td>Per exon/target</td>
<td>Exon fragmentation composition</td>
</tr>
<tr>
<td><a href="#fsc-e1-only"><code>{s}.FSC.regions.e1only.tsv</code></a></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>-E1</td>
<td>Per gene (E1)</td>
<td>Promoter-proximal signal</td>
</tr>
<tr>
<td><a href="#fsc-counts-pre-correction"><code>{s}.fsc_counts.tsv</code></a></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>-raw</td>
<td>Per <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bin</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction diagnostics</td>
</tr>
<tr>
<td><a href="#wps-windowed-protection-score"><code>{s}.WPS.parquet</code></a></td>
<td><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr></td>
<td>Per anchor</td>
<td>Nucleosome positioning</td>
</tr>
<tr>
<td><a href="#wps-panel"><code>{s}.WPS.panel.parquet</code></a></td>
<td><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr></td>
<td>Panel anchors</td>
<td>Gene-level nucleosome</td>
</tr>
<tr>
<td><a href="#wps-background"><code>{s}.WPS_background.parquet</code></a></td>
<td><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr>-bg</td>
<td>Alu stacks</td>
<td>Global chromatin state</td>
</tr>
<tr>
<td><a href="#endmotif"><code>{s}.EndMotif.tsv</code></a></td>
<td>Motif</td>
<td>Global (1 row)</td>
<td>256 4-mer end frequencies</td>
</tr>
<tr>
<td><a href="#endmotif-1-mer"><code>{s}.EndMotif1mer.tsv</code></a></td>
<td>Motif</td>
<td>Global (4 rows)</td>
<td>Base composition at ends</td>
</tr>
<tr>
<td><a href="#breakpointmotif"><code>{s}.BreakPointMotif.tsv</code></a></td>
<td>Motif</td>
<td>Global (1 row)</td>
<td>256 4-mer break frequencies</td>
</tr>
<tr>
<td><a href="#mds-motif-diversity-score"><code>{s}.MDS.tsv</code></a></td>
<td><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr></td>
<td>Global (1 row)</td>
<td>Motif diversity scalar</td>
</tr>
<tr>
<td><a href="#mds-exon-level"><code>{s}.MDS.exon.tsv</code></a></td>
<td>Region-<abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr></td>
<td>Per exon</td>
<td>Per-exon motif diversity</td>
</tr>
<tr>
<td><a href="#mds-gene-level"><code>{s}.MDS.gene.tsv</code></a></td>
<td>Region-<abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr></td>
<td>Per gene</td>
<td>Per-gene <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> + E1</td>
</tr>
<tr>
<td><a href="#ocf-orientation-aware-fragmentation"><code>{s}.OCF.tsv</code></a></td>
<td><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr></td>
<td>Per tissue</td>
<td>Tissue-of-origin score</td>
</tr>
<tr>
<td><a href="#ocf-sync"><code>{s}.OCF.sync.tsv</code></a></td>
<td><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr></td>
<td>Positional</td>
<td>Strand-phased profiles</td>
</tr>
<tr>
<td><a href="#tfbs-transcription-factor-binding-site-entropy"><code>{s}.TFBS.tsv</code></a></td>
<td><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr></td>
<td>Per TF</td>
<td>TF footprint entropy</td>
</tr>
<tr>
<td><a href="#tfbs-sync"><code>{s}.TFBS.sync.tsv</code></a></td>
<td><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr></td>
<td>Per TF  size</td>
<td>Size-resolved TF profiles</td>
</tr>
<tr>
<td><a href="#atac-chromatin-accessibility-entropy"><code>{s}.ATAC.tsv</code></a></td>
<td><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr></td>
<td>Per tissue</td>
<td><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> region entropy</td>
</tr>
<tr>
<td><a href="#atac-sync"><code>{s}.ATAC.sync.tsv</code></a></td>
<td><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr></td>
<td>Per tissue  size</td>
<td>Size-resolved <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> profiles</td>
</tr>
<tr>
<td><a href="#mfsd-mutant-fragment-size-distribution"><code>{s}.mFSD.tsv</code></a></td>
<td><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr></td>
<td>Per variant</td>
<td>Variant fragment size metrics</td>
</tr>
<tr>
<td><a href="#mfsd-distributions"><code>{s}.mFSD.distributions.tsv</code></a></td>
<td><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr></td>
<td>Per variant  size</td>
<td>Raw size histograms</td>
</tr>
<tr>
<td><a href="#uxm-methylation"><code>{s}.UXM.tsv</code></a></td>
<td><abbr title="Fragment-level Methylation - Methylation state of individual fragments">UXM</abbr></td>
<td>Per region</td>
<td>U/X/M methylation fractions</td>
</tr>
<tr>
<td><a href="#gc-correction-factors"><code>{s}.correction_factors.tsv</code></a></td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr></td>
<td>Per (len, <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>) bin</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias weights</td>
</tr>
<tr>
<td><a href="#metadata-json"><code>{s}.metadata.json</code></a></td>
<td>Meta</td>
<td>Global</td>
<td>Run parameters + QC</td>
</tr>
<tr>
<td><a href="#features-json"><code>{s}.features.json</code></a></td>
<td>All</td>
<td>All</td>
<td>Unified ML feature export</td>
</tr>
</tbody>
</table>
<blockquote>
<p><code>{s}</code> = sample name.</p>
</blockquote>
<hr>
<h2 id="on-target-vs-off-target-files">On-Target vs Off-Target Files</h2>
<p>Most fragmentomics features generate <strong>two parallel outputs</strong> in panel mode: a standard file (off-target reads) and an <code>.ontarget.tsv</code> variant (on-target reads). Understanding the difference is critical for choosing the right input to any ML model.</p>
<h3 id="what-on-target-and-off-target-mean">What "On-Target" and "Off-Target" Mean</h3>
<p>In panel sequencing (e.g. MSK-ACCESS), reads fall into two categories:</p>
<table>
<thead>
<tr>
<th>Category</th>
<th>Reads</th>
<th>Depth</th>
<th><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Off-target</strong></td>
<td>Do NOT overlap capture bait regions</td>
<td>Low (~15) but genome-wide</td>
<td>Unbiased  not affected by probe <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr></td>
</tr>
<tr>
<td><strong>On-target</strong></td>
<td>Overlap the capture bait regions</td>
<td>High (~5002000) but only at panel genes</td>
<td>Biased  capture efficiency varies by probe <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> content</td>
</tr>
</tbody>
</table>
<h3 id="why-the-split-matters">Why the Split Matters</h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>          All cfDNA fragments in BAM
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>                    
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>         
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>    Off-target             On-target
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>  (genome-wide)         (panel genes only)
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>                             
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>   FSC.tsv, FSR.tsv      FSC.ontarget.tsv
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>   MDS.tsv, OCF.tsv      MDS.ontarget.tsv
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>   FSD.tsv, ...          FSD.ontarget.tsv
</code></pre></div>
<p><strong>The off-target pool</strong> is sampled uniformly across the genome, unaffected by probe <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>, making it the <strong>gold standard for fragmentation features</strong> (<abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr>, <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>, <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr>). The <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction model is also trained exclusively on off-target fragments.</p>
<p><strong>The on-target pool</strong> has high depth at panel loci but is confounded by capture efficiency  probes with higher <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> content capture more efficiently, making fragment size distributions at those loci appear artificially different. On-target reads are primarily useful for gene-level copy number and region-specific analysis, not genome-wide fragmentation.</p>
<h3 id="which-to-use">Which to Use</h3>
<table>
<thead>
<tr>
<th>Task</th>
<th>Use</th>
<th>Reason</th>
</tr>
</thead>
<tbody>
<tr>
<td>Pan-cancer ML features (<abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr>, <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>, <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr>)</td>
<td><strong>Off-target</strong> (<code>.tsv</code>)</td>
<td>Unbiased, genome-wide, <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected with unbiased model</td>
</tr>
<tr>
<td>Gene-level copy number (<abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.gene.tsv)</td>
<td><strong>On-target</strong> internally</td>
<td>Gene <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> already uses on-target correction factors</td>
</tr>
<tr>
<td>Gene fragmentation composition (<abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.regions, E1)</td>
<td>On-target reads, but captured in gene/region TSVs</td>
<td>Not the raw <code>.ontarget.tsv</code>  use <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.gene / <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.regions</td>
</tr>
<tr>
<td>Motif features for tissue-of-origin (<abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>, <abbr title="End Dinucleotide Matrix - 16x16 matrix of terminal dinucleotide frequencies">EDM</abbr>, <abbr title="Breakpoint Motif - 4-mer motif frequencies at fragment breakpoints">BPM</abbr>)</td>
<td><strong>Off-target</strong> (<code>.tsv</code>)</td>
<td>On-target motifs are biased by probe sequence</td>
</tr>
<tr>
<td><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> on-target (when off-target too sparse)</td>
<td><code>.ontarget.tsv</code></td>
<td>Lower depth but gene-anchored signal</td>
</tr>
<tr>
<td><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> tissue-of-origin</td>
<td><strong>Off-target</strong> (<code>.tsv</code>)</td>
<td>On-target <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> biased by capture regions' tissue specificity</td>
</tr>
<tr>
<td>Building a <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></td>
<td><strong>Off-target</strong> only</td>
<td>Must match what the sample uses</td>
</tr>
</tbody>
</table>
<div class="admonition warning">
<p class="admonition-title">Do not mix off-target and on-target in the same model</p>
<p>Features from <code>FSR.tsv</code> (off-target) and <code>FSR.ontarget.tsv</code> (on-target) are not on the same scale. The on-target pool has different <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias, different fragment size distributions, and different effective depth. Always use the same variant consistently across all samples in a cohort.</p>
</div>
<h3 id="complete-on-target-off-target-file-inventory">Complete On-Target / Off-Target File Inventory</h3>
<table>
<thead>
<tr>
<th>Base file</th>
<th>On-target variant</th>
<th>Off-target variant</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>{s}.FSD.tsv</code></td>
<td><code>{s}.FSD.ontarget.tsv</code></td>
<td> (base is off-target)</td>
</tr>
<tr>
<td><code>{s}.FSR.tsv</code></td>
<td><code>{s}.FSR.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.FSC.tsv</code></td>
<td><code>{s}.FSC.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.EndMotif.tsv</code></td>
<td><code>{s}.EndMotif.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.BreakPointMotif.tsv</code></td>
<td><code>{s}.BreakPointMotif.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.MDS.tsv</code></td>
<td><code>{s}.MDS.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.OCF.tsv</code></td>
<td><code>{s}.OCF.ontarget.tsv</code></td>
<td><code>{s}.OCF.offtarget.tsv</code> </td>
</tr>
<tr>
<td><code>{s}.OCF.sync.tsv</code></td>
<td><code>{s}.OCF.ontarget.sync.tsv</code></td>
<td><code>{s}.OCF.offtarget.sync.tsv</code></td>
</tr>
<tr>
<td><code>{s}.TFBS.tsv</code></td>
<td><code>{s}.TFBS.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.TFBS.sync.tsv</code></td>
<td><code>{s}.TFBS.ontarget.sync.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.ATAC.tsv</code></td>
<td><code>{s}.ATAC.ontarget.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.ATAC.sync.tsv</code></td>
<td><code>{s}.ATAC.ontarget.sync.tsv</code></td>
<td></td>
</tr>
<tr>
<td><code>{s}.correction_factors.tsv</code></td>
<td><code>{s}.correction_factors.ontarget.tsv</code></td>
<td></td>
</tr>
</tbody>
</table>
<blockquote>
<p> <strong><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> <code>offtarget</code></strong> is a special case: <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> computes 3 variants  all reads, on-target reads, and a panel-specific off-target score derived from regions near but not in the capture baits. The <code>OCF.offtarget.tsv</code> is the panel-off-target score, not the same concept as the base <code>OCF.tsv</code>.</p>
</blockquote>
<h3 id="gc-correction-and-on-target"><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction and On-Target</h3>
<p><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction is trained on <strong>off-target reads only</strong>, then applied to both pools:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>Off-target reads  GC model training  correction_factors.tsv
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>On-target reads   correction_factors.ontarget.tsv (separate model)
</code></pre></div>
<p>The on-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model accounts for probe-specific capture bias. It is used internally when generating <code>FSC.gene.tsv</code> and <code>FSC.regions.tsv</code>  you do not need to apply it manually.</p>
<hr>
<h2 id="core-fragmentomics">Core Fragmentomics</h2>
<h3 id="fsd-fragment-size-distribution"><abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr> (Fragment Size Distribution)</h3>
<p><strong>File:</strong> <code>{sample}.FSD.tsv</code> / <code>{sample}.FSD.ontarget.tsv</code></p>
<p><abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr> measures how many fragments of each size (65400 bp, 5 bp bins) come from each chromosomal arm. Each row is one arm.</p>
<h4 id="columns">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>region</code></td>
<td>str</td>
<td>Arm label, e.g. <code>chr1p</code>, <code>chr17q</code></td>
</tr>
<tr>
<td><code>65-69</code>, <code>70-74</code>,  <code>395-399</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected fragment count in that 5 bp size bin</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td>Total <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected fragment count for this arm</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>ARM-LEVEL fragmentation fingerprint</strong>: Each arm's histogram reflects the chromatin state of that chromosomal region</li>
<li><strong>Aneuploidy / CNV detection</strong>: Arms with deletions or amplifications show altered absolute counts in <code>total</code></li>
<li><strong>Tumor-specific size shift</strong>: Cancer plasma shows a systematic shift toward shorter fragments genome-wide</li>
</ul>
<h4 id="ml-use-case">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a><span class="kn">import</span><span class="w"> </span><span class="nn">numpy</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">np</span>
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.FSD.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="s2">"region"</span><span class="p">)</span>
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a><span class="n">size_cols</span> <span class="o">=</span> <span class="p">[</span><span class="n">c</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span> <span class="k">if</span> <span class="n">c</span> <span class="o">!=</span> <span class="s2">"total"</span><span class="p">]</span>
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>
<a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="c1"># Feature vector: proportion of each size class per arm</span>
<a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a><span class="n">props</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">size_cols</span><span class="p">]</span><span class="o">.</span><span class="n">div</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s2">"total"</span><span class="p">]</span> <span class="o">+</span> <span class="mf">1e-9</span><span class="p">,</span> <span class="n">axis</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>
<a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a><span class="c1"># Reduce to 3-class: short (65-149), mono (150-259), long (260-400)</span>
<a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a><span class="n">props</span><span class="p">[</span><span class="s2">"short_frac"</span><span class="p">]</span> <span class="o">=</span> <span class="n">props</span><span class="p">[[</span><span class="n">c</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">size_cols</span> <span class="k">if</span> <span class="nb">int</span><span class="p">(</span><span class="n">c</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">"-"</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span> <span class="o">&lt;</span> <span class="mi">150</span><span class="p">]]</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a><span class="n">props</span><span class="p">[</span><span class="s2">"mono_frac"</span><span class="p">]</span>  <span class="o">=</span> <span class="n">props</span><span class="p">[[</span><span class="n">c</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">size_cols</span> <span class="k">if</span> <span class="mi">150</span> <span class="o">&lt;=</span> <span class="nb">int</span><span class="p">(</span><span class="n">c</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">"-"</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span> <span class="o">&lt;</span> <span class="mi">260</span><span class="p">]]</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
<a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a><span class="n">props</span><span class="p">[</span><span class="s2">"long_frac"</span><span class="p">]</span>  <span class="o">=</span> <span class="n">props</span><span class="p">[[</span><span class="n">c</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">size_cols</span> <span class="k">if</span> <span class="nb">int</span><span class="p">(</span><span class="n">c</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">"-"</span><span class="p">)[</span><span class="mi">0</span><span class="p">])</span> <span class="o">&gt;=</span> <span class="mi">260</span><span class="p">]]</span><span class="o">.</span><span class="n">sum</span><span class="p">(</span><span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>
</code></pre></div>
<p><strong>Best for</strong>: Arm-level <code>short_frac</code> as 46-feature input (2 arms  23 chromosomes) per sample for pan-cancer models.</p>
<div class="admonition note">
<p class="admonition-title">On-target variant</p>
<p><code>FSD.ontarget.tsv</code> uses only reads overlapping target capture regions. Useful for panel-specific copy number analysis, but capture-biased  prefer off-target for fragmentation ML features.</p>
</div>
<hr>
<h3 id="fsr-fragment-size-ratio"><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> (Fragment Size Ratio)</h3>
<p><strong>File:</strong> <code>{sample}.FSR.tsv</code> / <code>{sample}.FSR.ontarget.tsv</code></p>
<p><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> computes the <strong><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized short-to-long ratio</strong> per 5 Mb window. This is the primary genome-wide tumor fraction biomarker.</p>
<h4 id="columns_1">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>region</code></td>
<td>str</td>
<td>5 Mb window, e.g. <code>chr1:0-5000000</code></td>
</tr>
<tr>
<td><code>short_count</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count of short frags (65149 bp)</td>
</tr>
<tr>
<td><code>long_count</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count of long frags (261400 bp)</td>
</tr>
<tr>
<td><code>total_count</code></td>
<td>float</td>
<td>Total <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count</td>
</tr>
<tr>
<td><code>short_norm</code></td>
<td>float</td>
<td><code>short_count / PON_short_mean</code>  <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized short</td>
</tr>
<tr>
<td><code>long_norm</code></td>
<td>float</td>
<td><code>long_count / PON_long_mean</code>  <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized long</td>
</tr>
<tr>
<td><code>short_long_ratio</code></td>
<td>float</td>
<td><code>short_norm / long_norm</code>  primary biomarker</td>
</tr>
<tr>
<td><code>short_long_log2</code></td>
<td>float</td>
<td><code>log2(short_long_ratio)</code>  ML-ready signed metric</td>
</tr>
<tr>
<td><code>short_frac</code></td>
<td>float</td>
<td><code>short_count / total_count</code>  raw proportion</td>
</tr>
<tr>
<td><code>long_frac</code></td>
<td>float</td>
<td><code>long_count / total_count</code>  raw proportion</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_1">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Tumor fraction estimation</strong>: <code>short_long_ratio</code> increases proportionally with <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> fraction</li>
<li><strong>Genome-wide cancer screen</strong>: Profile of 500+ windows per sample captures focal and arm-level alterations</li>
<li><strong><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> comparison</strong>: <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> normalization (<code>short_norm</code>, <code>long_norm</code>) removes batch effects before ratio  critical for cross-batch comparison</li>
</ul>
<div class="admonition important">
<p class="admonition-title">Why not just use <code>short_frac</code> from <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>?</p>
<p><code>short_frac</code> is a raw proportion  it conflates true biology with library prep and <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias. <code>short_long_ratio</code> divides <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized values, canceling technical noise. Use <abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> for any cross-sample comparison.</p>
</div>
<h4 id="ml-use-case_1">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.FSR.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="c1"># Primary feature vector: ~500 windows  1 scalar</span>
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="n">feature_vec</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"short_long_log2"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># signed, mean ~0 in healthy</span>
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="c1"># Genome-wide statistics as compact features</span>
<a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="n">features</span> <span class="o">=</span> <span class="p">{</span>
<a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a>    <span class="s2">"fsr_mean"</span><span class="p">:</span>   <span class="n">df</span><span class="p">[</span><span class="s2">"short_long_log2"</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>
<a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a>    <span class="s2">"fsr_std"</span><span class="p">:</span>    <span class="n">df</span><span class="p">[</span><span class="s2">"short_long_log2"</span><span class="p">]</span><span class="o">.</span><span class="n">std</span><span class="p">(),</span>
<a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a>    <span class="s2">"fsr_q90"</span><span class="p">:</span>    <span class="n">df</span><span class="p">[</span><span class="s2">"short_long_log2"</span><span class="p">]</span><span class="o">.</span><span class="n">quantile</span><span class="p">(</span><span class="mf">0.9</span><span class="p">),</span>
<a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a>    <span class="s2">"fsr_n_elevated"</span><span class="p">:</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s2">"short_long_log2"</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mf">0.3</span><span class="p">)</span><span class="o">.</span><span class="n">sum</span><span class="p">(),</span>
<a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a><span class="p">}</span>
</code></pre></div>
<p><strong>Typical range</strong>: healthy ~0.0  0.15; high <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> &gt; +0.4 (more short frags than <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>)</p>
<hr>
<h3 id="fsc-fragment-size-coverage"><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> (Fragment Size Coverage)</h3>
<p><strong>File:</strong> <code>{sample}.FSC.tsv</code> / <code>{sample}.FSC.ontarget.tsv</code></p>
<p><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> counts fragments in 6 non-overlapping size channels across 5 Mb windows  the foundational multi-channel coverage feature.</p>
<h4 id="columns_2">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>chrom</code></td>
<td>str</td>
<td>Chromosome</td>
</tr>
<tr>
<td><code>start</code></td>
<td>int</td>
<td>Window start (0-based)</td>
</tr>
<tr>
<td><code>end</code></td>
<td>int</td>
<td>Window end</td>
</tr>
<tr>
<td><code>ultra_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count, 65100 bp</td>
</tr>
<tr>
<td><code>core_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count, 101149 bp</td>
</tr>
<tr>
<td><code>mono_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count, 150220 bp</td>
</tr>
<tr>
<td><code>di_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count, 221260 bp</td>
</tr>
<tr>
<td><code>long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count, 261400 bp</td>
</tr>
<tr>
<td><code>ultra_long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count, 4011000 bp</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected total, 651000 bp</td>
</tr>
<tr>
<td><code>mean_gc</code></td>
<td>float</td>
<td>Mean <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> fraction of fragments in window</td>
</tr>
<tr>
<td><code>*_log2</code></td>
<td>float</td>
<td>log(channel / PON_mean), with <code>--pon-model</code></td>
</tr>
<tr>
<td><code>*_reliability</code></td>
<td>float</td>
<td>1/(PON_variance + k)  weight for <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> columns</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_2">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Multi-channel fragmentation profile</strong>: Each channel represents fragments sharing a biological origin (nucleosomal, sub-nucleosomal, apoptotic)</li>
<li><strong>CNV proxy</strong>: <code>total</code> counts per arm reflect coverage depth  useful for detecting large-scale copy number events</li>
<li><strong><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> log2 ratios</strong>: <code>core_short_log2</code>, <code>mono_nucl_log2</code> etc. are analogous to CNV log-ratio tracks</li>
</ul>
<h4 id="ml-use-case_2">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.FSC.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a><span class="c1"># 6-channel feature matrix: windows  channels</span>
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="n">channels</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"ultra_short"</span><span class="p">,</span> <span class="s2">"core_short"</span><span class="p">,</span> <span class="s2">"mono_nucl"</span><span class="p">,</span> <span class="s2">"di_nucl"</span><span class="p">,</span> <span class="s2">"long"</span><span class="p">,</span> <span class="s2">"ultra_long"</span><span class="p">]</span>
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">channels</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># shape: (n_windows, 6)</span>
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>
<a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a><span class="c1"># Normalize to proportions (remove depth variation)</span>
<a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a><span class="n">X_prop</span> <span class="o">=</span> <span class="n">X</span> <span class="o">/</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s2">"total"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span><span class="p">[:,</span> <span class="kc">None</span><span class="p">]</span> <span class="o">+</span> <span class="mf">1e-9</span><span class="p">)</span>
<a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>
<a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a><span class="c1"># With PON: use log2 ratios directly (already depth-normalized)</span>
<a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a><span class="n">log2_cols</span> <span class="o">=</span> <span class="p">[</span><span class="n">c</span> <span class="o">+</span> <span class="s2">"_log2"</span> <span class="k">for</span> <span class="n">c</span> <span class="ow">in</span> <span class="n">channels</span> <span class="k">if</span> <span class="n">c</span> <span class="o">+</span> <span class="s2">"_log2"</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span><span class="p">]</span>
<a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a><span class="n">X_pon</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">log2_cols</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># shape: (n_windows, 6)  centered near 0 in healthy</span>
</code></pre></div>
<p><strong>Best for</strong>: Input to CNV callers, genome-wide fragmentation classifiers, and tumor fraction regression.</p>
<hr>
<h3 id="fsc-gene-level"><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> Gene-Level</h3>
<p><strong>File:</strong> <code>{sample}.FSC.gene.tsv</code><br>
<strong>Requires:</strong> <code>--assay xs2</code> (or other assay code) / <code>run-all</code></p>
<p>Aggregates <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> across all exons for each panel gene. Rows = genes.</p>
<h4 id="columns_3">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>gene</code></td>
<td>str</td>
<td>HGNC symbol (e.g. <code>ATM</code>, <code>TP53</code>)</td>
</tr>
<tr>
<td><code>n_regions</code></td>
<td>int</td>
<td>Number of exons/targets captured</td>
</tr>
<tr>
<td><code>total_bp</code></td>
<td>int</td>
<td>Total base pairs covered</td>
</tr>
<tr>
<td><code>ultra_short</code>  <code>long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count per size class</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td>Total <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected count</td>
</tr>
<tr>
<td><code>ultra_short_ratio</code>  <code>long_ratio</code></td>
<td>float</td>
<td><code>channel / total</code>  size composition</td>
</tr>
<tr>
<td><code>normalized_depth</code></td>
<td>float</td>
<td>RPKM-like: <code>(total  10) / (total_bp  total_frags)</code></td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_3">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Gene-level copy number</strong>: <code>normalized_depth</code> enables comparing coverage across genes in the same sample</li>
<li><strong>Gene fragmentation composition</strong>: <code>*_ratio</code> columns show whether a gene's reads are enriched for short (tumor) or long (normal) fragments</li>
<li><strong>Panel-level feature matrix</strong>: 146 genes  6 channels = 876 features per sample</li>
</ul>
<h4 id="ml-use-case_3">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.FSC.gene.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"gene"</span><span class="p">)</span>
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="c1"># Gene-level short enrichment</span>
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="n">df</span><span class="p">[</span><span class="s2">"tumor_signal"</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"ultra_short_ratio"</span><span class="p">]</span> <span class="o">+</span> <span class="n">df</span><span class="p">[</span><span class="s2">"core_short_ratio"</span><span class="p">]</span>
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a>
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a><span class="c1"># Full channel composition feature matrix: 146 genes  5 ratios</span>
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a><span class="n">ratio_cols</span> <span class="o">=</span> <span class="p">[</span><span class="s2">"ultra_short_ratio"</span><span class="p">,</span> <span class="s2">"core_short_ratio"</span><span class="p">,</span> <span class="s2">"mono_nucl_ratio"</span><span class="p">,</span> <span class="s2">"di_nucl_ratio"</span><span class="p">,</span> <span class="s2">"long_ratio"</span><span class="p">]</span>
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">ratio_cols</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># shape: (146, 5) per sample</span>
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a>
<a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a><span class="c1"># Normalized depth for CNV</span>
<a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a><span class="n">cnv_proxy</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"normalized_depth"</span><span class="p">]</span>  <span class="c1"># pivot across samples  CNV log-ratio</span>
</code></pre></div>
<p><strong>Best for</strong>: Gene-specific models, tissue-of-origin (which genes show altered fragmentation), copy number inference.</p>
<hr>
<h3 id="fsc-region-level"><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> Region-Level</h3>
<p><strong>File:</strong> <code>{sample}.FSC.regions.tsv</code><br>
<strong>Requires:</strong> <code>--assay</code> or <code>run-all</code></p>
<p>Per-exon/target fragment size coverage. Most granular <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> output.</p>
<h4 id="columns_4">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>chrom</code></td>
<td>str</td>
<td>Chromosome</td>
</tr>
<tr>
<td><code>start</code> / <code>end</code></td>
<td>int</td>
<td>Exon/target coordinates</td>
</tr>
<tr>
<td><code>gene</code></td>
<td>str</td>
<td>Gene symbol</td>
</tr>
<tr>
<td><code>region_name</code></td>
<td>str</td>
<td>Unique exon/target identifier</td>
</tr>
<tr>
<td><code>region_bp</code></td>
<td>int</td>
<td>Region size in bp</td>
</tr>
<tr>
<td><code>ultra_short</code>  <code>long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected counts</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td>Total count</td>
</tr>
<tr>
<td><code>ultra_short_ratio</code>  <code>long_ratio</code></td>
<td>float</td>
<td><code>channel / total</code></td>
</tr>
<tr>
<td><code>normalized_depth</code></td>
<td>float</td>
<td>RPKM-like depth</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_4">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Exon-level resolution</strong>: Useful when only specific exons (e.g., hotspot exons in <code>TP53</code>) show altered fragmentation</li>
<li><strong>Fine-grained CNV</strong>: Detect focal amplifications or deletions at single-exon scale</li>
<li><strong>Input for <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> building</strong>: Use to compute exon-level expected depth profiles</li>
</ul>
<h4 id="ml-use-case_4">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.FSC.regions.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span>
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a><span class="c1"># Pivot: regions  channels, one row per region</span>
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a><span class="n">pivot</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">pivot_table</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="s2">"region_name"</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="p">[</span><span class="s2">"core_short_ratio"</span><span class="p">,</span> <span class="s2">"mono_nucl_ratio"</span><span class="p">])</span>
<a id="__codelineno-6-5" name="__codelineno-6-5" href="#__codelineno-6-5"></a>
<a id="__codelineno-6-6" name="__codelineno-6-6" href="#__codelineno-6-6"></a><span class="c1"># Filter to well-covered regions</span>
<a id="__codelineno-6-7" name="__codelineno-6-7" href="#__codelineno-6-7"></a><span class="n">well_covered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s2">"total"</span><span class="p">]</span> <span class="o">&gt;</span> <span class="mi">50</span><span class="p">][</span><span class="s2">"region_name"</span><span class="p">]</span>
<a id="__codelineno-6-8" name="__codelineno-6-8" href="#__codelineno-6-8"></a><span class="n">df_filtered</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s2">"region_name"</span><span class="p">]</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">well_covered</span><span class="p">)]</span>
</code></pre></div>
<hr>
<h3 id="fsc-e1-only"><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> E1-Only</h3>
<p><strong>File:</strong> <code>{sample}.FSC.regions.e1only.tsv</code><br>
<strong>Requires:</strong> <code>--assay</code> or <code>run-all</code> (disable with <code>--disable-e1-aggregation</code>)</p>
<p>First exon (E1) per gene only. Same columns as <code>FSC.regions.tsv</code>.</p>
<h4 id="purpose-use-cases_5">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Promoter-proximal fragmentation</strong>: E1 = first exon = nucleosome-depleted region (NDR) near <abbr title="Transcription Start Site - Gene transcription initiation position">TSS</abbr>. NDRs have the most cancer-specific fragmentation patterns (Helzer et al. 2025)</li>
<li><strong>Highest cancer signal</strong>: E1 consistently outperforms whole-gene <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> in early cancer detection tasks</li>
<li><strong>Compact feature set</strong>: 146 genes  1 exon = compact, interpretable vector</li>
</ul>
<h4 id="ml-use-case_5">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.FSC.regions.e1only.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"gene"</span><span class="p">)</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a><span class="c1"># Primary cancer signal: promoter short enrichment</span>
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="n">df</span><span class="p">[</span><span class="s2">"promoter_short"</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"ultra_short_ratio"</span><span class="p">]</span> <span class="o">+</span> <span class="n">df</span><span class="p">[</span><span class="s2">"core_short_ratio"</span><span class="p">]</span>
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>
<a id="__codelineno-7-6" name="__codelineno-7-6" href="#__codelineno-7-6"></a><span class="c1"># 146-gene feature vector  best single FSC feature for early detection</span>
<a id="__codelineno-7-7" name="__codelineno-7-7" href="#__codelineno-7-7"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"promoter_short"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
</code></pre></div>
<div class="admonition tip">
<p class="admonition-title">Use E1 over gene-level for early detection models</p>
<p><code>e1only</code> routinely achieves lower AUC for early-stage cancer vs <code>FSC.gene.tsv</code> because E1 captures NDR-specific fragmentation that is washed out by whole-gene averaging.</p>
</div>
<hr>
<h3 id="fsc-counts-pre-correction"><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> Counts (Pre-Correction)</h3>
<p><strong>File:</strong> <code>{sample}.fsc_counts.tsv</code> / <code>{sample}.correction_factors.ontarget.tsv</code></p>
<p>Raw bin-level fragment counts before <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction  used internally for <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model training.</p>
<h4 id="purpose-use-cases_6">Purpose &amp; Use Cases</h4>
<ul>
<li><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias diagnostics</strong>: Compare observed vs expected counts per (length, <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>) bin</li>
<li><strong>Batch QC</strong>: Libraries with systematic <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias show large correction factors in specific bins</li>
<li><strong>Not for ML features</strong>: <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected values in <code>FSC.tsv</code> are the right input; <code>fsc_counts</code> is pre-correction</li>
</ul>
<hr>
<h2 id="nucleosome-positioning">Nucleosome Positioning</h2>
<h3 id="wps-windowed-protection-score"><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> (Windowed Protection Score)</h3>
<p><strong>File:</strong> <code>{sample}.WPS.parquet</code></p>
<p>Per-anchor nucleosome protection profiles. Each row is one genomic anchor (gene <abbr title="Transcription Start Site - Gene transcription initiation position">TSS</abbr> or <abbr title="CCCTC-binding factor - Chromatin architecture protein">CTCF</abbr> site). <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> captures how protected (nucleosome-covered) a region is to fragments of two sizes.</p>
<h4 id="columns-parquet">Columns (Parquet)</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>region_id</code></td>
<td>str</td>
<td>Anchor identifier (e.g. <code>ENSG00000142611_TSS</code>)</td>
</tr>
<tr>
<td><code>chrom</code></td>
<td>str</td>
<td>Chromosome</td>
</tr>
<tr>
<td><code>center</code></td>
<td>int</td>
<td>Anchor midpoint</td>
</tr>
<tr>
<td><code>strand</code></td>
<td>str</td>
<td><code>+</code> / <code>-</code></td>
</tr>
<tr>
<td><code>region_type</code></td>
<td>str</td>
<td><code>TSS</code>, <code>CTCF</code>, etc.</td>
</tr>
<tr>
<td><code>wps_nuc</code></td>
<td>float</td>
<td>Raw nucleosomal <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> (120180 bp fragments)</td>
</tr>
<tr>
<td><code>wps_tf</code></td>
<td>float</td>
<td>Raw TF footprint <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> (3580 bp fragments)</td>
</tr>
<tr>
<td><code>wps_nuc_smooth</code></td>
<td>float</td>
<td>Savitzky-Golay smoothed nucleosomal <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr></td>
</tr>
<tr>
<td><code>wps_tf_smooth</code></td>
<td>float</td>
<td>Savitzky-Golay smoothed TF <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr></td>
</tr>
<tr>
<td><code>wps_nuc_mean</code></td>
<td>float</td>
<td>Mean <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> across anchor window</td>
</tr>
<tr>
<td><code>wps_tf_mean</code></td>
<td>float</td>
<td>Mean TF <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> across anchor window</td>
</tr>
<tr>
<td><code>prot_frac_nuc</code></td>
<td>float</td>
<td>Fraction of window with <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> &gt; 0 (nucleosome-covered)</td>
</tr>
<tr>
<td><code>prot_frac_tf</code></td>
<td>float</td>
<td>Fraction of window with TF <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> &gt; 0</td>
</tr>
<tr>
<td><code>wps_nuc_z</code></td>
<td>float</td>
<td>Z-score vs <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> baseline (with <code>--pon-model</code>)</td>
</tr>
<tr>
<td><code>wps_tf_z</code></td>
<td>float</td>
<td>TF <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> Z-score vs <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_7">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Nucleosome positioning</strong>: <code>wps_nuc_smooth</code> detects nucleosome phasing at <abbr title="Transcription Start Site - Gene transcription initiation position">TSS</abbr>/<abbr title="CCCTC-binding factor - Chromatin architecture protein">CTCF</abbr> anchors</li>
<li><strong>TF accessibility</strong>: <code>wps_tf</code> / <code>prot_frac_tf</code> reflects accessible chromatin at TF binding sites</li>
<li><strong>Cancer signal</strong>: Tumor DNA shows flattened/disrupted <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> profiles at <abbr title="Transcription Start Site - Gene transcription initiation position">TSS</abbr> of active genes</li>
<li><strong>NRL (Nucleosome Repeat Length)</strong>: Computed in <code>WPS_background.parquet</code> from Alu stacking</li>
</ul>
<h4 id="ml-use-case_6">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">pandas</span><span class="w"> </span><span class="k">as</span><span class="w"> </span><span class="nn">pd</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_parquet</span><span class="p">(</span><span class="s2">"sample.WPS.parquet"</span><span class="p">)</span>
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a><span class="c1"># Feature vector per sample: mean WPS across all anchors</span>
<a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="n">features</span> <span class="o">=</span> <span class="p">{</span>
<a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a>    <span class="s2">"wps_nuc_global_mean"</span><span class="p">:</span> <span class="n">df</span><span class="p">[</span><span class="s2">"wps_nuc_mean"</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>
<a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a>    <span class="s2">"wps_nuc_global_std"</span><span class="p">:</span>  <span class="n">df</span><span class="p">[</span><span class="s2">"wps_nuc_mean"</span><span class="p">]</span><span class="o">.</span><span class="n">std</span><span class="p">(),</span>
<a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>    <span class="s2">"prot_frac_nuc_mean"</span><span class="p">:</span>  <span class="n">df</span><span class="p">[</span><span class="s2">"prot_frac_nuc"</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>
<a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a>    <span class="s2">"prot_frac_tf_mean"</span><span class="p">:</span>   <span class="n">df</span><span class="p">[</span><span class="s2">"prot_frac_tf"</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>
<a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a><span class="p">}</span>
<a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a>
<a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a><span class="c1"># Per-anchor feature matrix for gene-level models</span>
<a id="__codelineno-8-14" name="__codelineno-8-14" href="#__codelineno-8-14"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s2">"wps_nuc_mean"</span><span class="p">,</span> <span class="s2">"wps_tf_mean"</span><span class="p">,</span> <span class="s2">"prot_frac_nuc"</span><span class="p">,</span> <span class="s2">"prot_frac_tf"</span><span class="p">]]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-8-15" name="__codelineno-8-15" href="#__codelineno-8-15"></a><span class="c1"># shape: (~15,000 anchors, 4)  one row per TSS/CTCF</span>
</code></pre></div>
<p><strong>Best for</strong>: Deep learning input (<abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> profiles as 1D signals), nucleosome periodicity score, <abbr title="Transcription Start Site - Gene transcription initiation position">TSS</abbr> accessibility classifier.</p>
<hr>
<h3 id="wps-panel"><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> Panel</h3>
<p><strong>File:</strong> <code>{sample}.WPS.panel.parquet</code><br>
<strong>Requires:</strong> <code>--assay</code></p>
<p>Same columns as <code>WPS.parquet</code>, filtered to panel gene anchors (~1,820 for xs2). Rows are panel-gene <abbr title="Transcription Start Site - Gene transcription initiation position">TSS</abbr> and <abbr title="CCCTC-binding factor - Chromatin architecture protein">CTCF</abbr> sites.</p>
<h4 id="ml-use-case_7">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_parquet</span><span class="p">(</span><span class="s2">"sample.WPS.panel.parquet"</span><span class="p">)</span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="c1"># Compact panel feature: 1820 anchors  4 = 7280 features</span>
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s2">"wps_nuc_mean"</span><span class="p">,</span> <span class="s2">"wps_tf_mean"</span><span class="p">,</span> <span class="s2">"prot_frac_nuc"</span><span class="p">,</span> <span class="s2">"prot_frac_tf"</span><span class="p">]]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a><span class="c1"># Gene-indexed lookup</span>
<a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a><span class="n">df_gene</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"region_id"</span><span class="p">)</span>
<a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a><span class="n">tp53_wps</span> <span class="o">=</span> <span class="n">df_gene</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="s2">"TP53_TSS"</span><span class="p">,</span> <span class="s2">"wps_nuc_mean"</span><span class="p">]</span>
</code></pre></div>
<hr>
<h3 id="wps-background"><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> Background</h3>
<p><strong>File:</strong> <code>{sample}.WPS_background.parquet</code></p>
<p>Hierarchical Alu element stacking analysis capturing <strong>global chromatin state and nucleosome repeat length (NRL)</strong>.</p>
<h4 id="columns_5">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>group_id</code></td>
<td>str</td>
<td><code>Global_All</code>, <code>Family_AluY/S/J/Other</code>, <code>Chr{N}_All</code></td>
</tr>
<tr>
<td><code>stacked_wps_nuc</code></td>
<td>float[]</td>
<td>30-position binned stacked <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> (nucleosomal)</td>
</tr>
<tr>
<td><code>stacked_wps_tf</code></td>
<td>float[]</td>
<td>30-position binned stacked <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> (TF)</td>
</tr>
<tr>
<td><code>alu_count</code></td>
<td>int</td>
<td>Number of Alu elements in this group</td>
</tr>
<tr>
<td><code>mean_wps_nuc</code></td>
<td>float</td>
<td>Mean <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> amplitude</td>
</tr>
<tr>
<td><code>nrl_bp</code></td>
<td>float</td>
<td>Estimated Nucleosome Repeat Length in bp (~190 in healthy)</td>
</tr>
<tr>
<td><code>nrl_deviation_bp</code></td>
<td>float</td>
<td>Deviation from expected 190 bp NRL</td>
</tr>
<tr>
<td><code>periodicity_score</code></td>
<td>float</td>
<td>Signal-to-noise ratio of periodicity (01)</td>
</tr>
<tr>
<td><code>adjusted_score</code></td>
<td>float</td>
<td>Periodicity score penalized by NRL deviation</td>
</tr>
<tr>
<td><code>fragment_ratio</code></td>
<td>float</td>
<td>Ratio of short/long fragments at Alu sites</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_8">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Global chromatin compaction</strong>: <code>nrl_bp</code> shortens in cancer (chromatin opens globally)</li>
<li><strong>NRL as tumor biomarker</strong>: Healthy plasma NRL ~190 bp; cancer &lt; 185 bp</li>
<li><strong>Background correction</strong>: Used to compute "global_pon" baseline for <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> z-scores</li>
</ul>
<h4 id="ml-use-case_8">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_parquet</span><span class="p">(</span><span class="s2">"sample.WPS_background.parquet"</span><span class="p">)</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a><span class="n">global_row</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s2">"group_id"</span><span class="p">]</span> <span class="o">==</span> <span class="s2">"Global_All"</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a>
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="n">features</span> <span class="o">=</span> <span class="p">{</span>
<a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a>    <span class="s2">"nrl_bp"</span><span class="p">:</span> <span class="n">global_row</span><span class="p">[</span><span class="s2">"nrl_bp"</span><span class="p">],</span>
<a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a>    <span class="s2">"periodicity_score"</span><span class="p">:</span> <span class="n">global_row</span><span class="p">[</span><span class="s2">"periodicity_score"</span><span class="p">],</span>
<a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a>    <span class="s2">"adjusted_score"</span><span class="p">:</span> <span class="n">global_row</span><span class="p">[</span><span class="s2">"adjusted_score"</span><span class="p">],</span>
<a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a>    <span class="s2">"fragment_ratio_bg"</span><span class="p">:</span> <span class="n">global_row</span><span class="p">[</span><span class="s2">"fragment_ratio"</span><span class="p">],</span>
<a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a><span class="p">}</span>
</code></pre></div>
<p><strong>Best for</strong>: Global chromatin state features, cancer screening, NRL as continuous tumor fraction predictor.</p>
<hr>
<h2 id="motif-tissue-of-origin">Motif &amp; Tissue-of-Origin</h2>
<h3 id="endmotif">EndMotif</h3>
<p><strong>File:</strong> <code>{sample}.EndMotif.tsv</code> / <code>{sample}.EndMotif.ontarget.tsv</code></p>
<p>4-mer frequencies at fragment 5 ends. One row per sample, 256 columns (one per AAAATTTT 4-mer).</p>
<h4 id="columns_6">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>AAAA</code>  <code>TTTT</code></td>
<td>Frequency of that 4-mer at fragment ends (sums to 1.0)</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_9">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Tissue-of-origin</strong>: Different tissues have distinct end-motif preferences based on DNASE1L3 activity</li>
<li><strong>Cancer detection</strong>: DNASE1L3 is suppressed in cancer, producing a flattened, less-specific motif profile</li>
<li><strong><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> input</strong>: Raw material for Motif Diversity Score calculation</li>
</ul>
<h4 id="ml-use-case_9">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.EndMotif.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span>
<a id="__codelineno-11-2" name="__codelineno-11-2" href="#__codelineno-11-2"></a><span class="c1"># Single row: 256 4-mer frequencies</span>
<a id="__codelineno-11-3" name="__codelineno-11-3" href="#__codelineno-11-3"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># 256-dimensional feature vector</span>
<a id="__codelineno-11-4" name="__codelineno-11-4" href="#__codelineno-11-4"></a>
<a id="__codelineno-11-5" name="__codelineno-11-5" href="#__codelineno-11-5"></a><span class="c1"># Reduce by GC content group (64  5 groups)</span>
<a id="__codelineno-11-6" name="__codelineno-11-6" href="#__codelineno-11-6"></a><span class="n">gc_groups</span> <span class="o">=</span> <span class="p">{</span><span class="s2">"AT-rich"</span><span class="p">:</span> <span class="p">[</span><span class="n">k</span> <span class="k">for</span> <span class="n">k</span> <span class="ow">in</span> <span class="n">df</span><span class="o">.</span><span class="n">columns</span> <span class="k">if</span> <span class="n">k</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">"A"</span><span class="p">)</span> <span class="o">+</span> <span class="n">k</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">"T"</span><span class="p">)</span> <span class="o">&gt;=</span> <span class="mi">3</span><span class="p">],</span> <span class="o">...</span><span class="p">}</span>
</code></pre></div>
<hr>
<h3 id="endmotif-1-mer">EndMotif 1-mer</h3>
<p><strong>File:</strong> <code>{sample}.EndMotif1mer.tsv</code></p>
<p>Single-base (A/C/G/T) composition at fragment ends.</p>
<h4 id="columns-base-fraction">Columns: <code>base</code>, <code>fraction</code></h4>
<h4 id="purpose-use-cases_10">Purpose &amp; Use Cases</h4>
<ul>
<li><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias QC</strong>: Should be roughly balanced; extreme <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> skew indicates library quality issues</li>
<li><strong>DNASE1L3 proxy</strong>: Healthy <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> has distinct strand-asymmetric base preferences</li>
</ul>
<hr>
<h3 id="breakpointmotif">BreakPointMotif</h3>
<p><strong>File:</strong> <code>{sample}.BreakPointMotif.tsv</code> / <code>{sample}.BreakPointMotif.ontarget.tsv</code></p>
<p>4-mer frequencies at internal fragment <strong>breakpoints</strong> (rather than ends). Same 256-column format as EndMotif.</p>
<h4 id="purpose-vs-endmotif">Purpose vs EndMotif</h4>
<table>
<thead>
<tr>
<th></th>
<th>EndMotif</th>
<th>BreakPointMotif</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Measures</strong></td>
<td>DNASE1L3 cutting preference</td>
<td>Mechanical fragmentation patterns</td>
</tr>
<tr>
<td><strong>Cancer signal</strong></td>
<td>DNASE1L3 suppression</td>
<td>Chromatin compaction / MNase-like cleavage</td>
</tr>
<tr>
<td><strong>Correlation</strong></td>
<td>r ~ 0.6 with <abbr title="Breakpoint Motif - 4-mer motif frequencies at fragment breakpoints">BPM</abbr></td>
<td>Complementary signal</td>
</tr>
</tbody>
</table>
<p><strong>Best for</strong>: Combining both in multimodal ML models  they capture distinct biological processes.</p>
<hr>
<h3 id="mds-motif-diversity-score"><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> (Motif Diversity Score)</h3>
<p><strong>File:</strong> <code>{sample}.MDS.tsv</code> / <code>{sample}.MDS.ontarget.tsv</code></p>
<p>Single-number summary of end-motif randomness (Shannon entropy of 256 4-mers).</p>
<h4 id="columns-mds-float">Columns: <code>MDS</code> (float)</h4>
<p><strong>Range</strong>: Healthy plasma ~0.800.85; cancer (DNASE1L3 suppressed) &lt; 0.75</p>
<h4 id="ml-use-case_10">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a><span class="n">mds</span> <span class="o">=</span> <span class="nb">float</span><span class="p">(</span><span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.MDS.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)[</span><span class="s2">"MDS"</span><span class="p">]</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">0</span><span class="p">])</span>
<a id="__codelineno-12-2" name="__codelineno-12-2" href="#__codelineno-12-2"></a><span class="c1"># Single scalar  highly interpretable cancer feature</span>
</code></pre></div>
<hr>
<h3 id="mds-exon-level"><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> Exon-Level</h3>
<p><strong>File:</strong> <code>{sample}.MDS.exon.tsv</code><br>
<strong>Requires:</strong> <code>region-mds</code> command or <code>run-all</code></p>
<p><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> calculated per exon/target from <abbr title="Binary Alignment Map - Compressed sequence alignment file format">BAM</abbr> reads overlapping that region.</p>
<h4 id="columns_7">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>gene</code></td>
<td>Gene symbol</td>
</tr>
<tr>
<td><code>name</code></td>
<td>Exon identifier (<code>gene:exonN</code> for WGS; target name for panel)</td>
</tr>
<tr>
<td><code>chrom</code></td>
<td>Chromosome</td>
</tr>
<tr>
<td><code>start</code> / <code>end</code></td>
<td>Exon coordinates</td>
</tr>
<tr>
<td><code>strand</code></td>
<td>Strand</td>
</tr>
<tr>
<td><code>n_fragments</code></td>
<td>Fragments overlapping this exon</td>
</tr>
<tr>
<td><code>mds</code></td>
<td>Motif Diversity Score for this exon</td>
</tr>
</tbody>
</table>
<h4 id="ml-use-case_11">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.MDS.exon.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span>
<a id="__codelineno-13-2" name="__codelineno-13-2" href="#__codelineno-13-2"></a>
<a id="__codelineno-13-3" name="__codelineno-13-3" href="#__codelineno-13-3"></a><span class="c1"># Filter low-coverage exons</span>
<a id="__codelineno-13-4" name="__codelineno-13-4" href="#__codelineno-13-4"></a><span class="n">df_high</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s2">"n_fragments"</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="mi">20</span><span class="p">]</span>
<a id="__codelineno-13-5" name="__codelineno-13-5" href="#__codelineno-13-5"></a>
<a id="__codelineno-13-6" name="__codelineno-13-6" href="#__codelineno-13-6"></a><span class="c1"># Per-exon MDS as feature matrix (rows = exons)</span>
<a id="__codelineno-13-7" name="__codelineno-13-7" href="#__codelineno-13-7"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df_high</span><span class="p">[[</span><span class="s2">"mds"</span><span class="p">]]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># or pivot into gene  exon matrix</span>
</code></pre></div>
<hr>
<h3 id="mds-gene-level"><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> Gene-Level</h3>
<p><strong>File:</strong> <code>{sample}.MDS.gene.tsv</code><br>
<strong>Requires:</strong> <code>region-mds</code> command or <code>run-all</code></p>
<p>Gene-level aggregation of per-exon <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>, plus E1 (first exon) <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> as the promoter-proximal signal.</p>
<h4 id="columns_8">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>gene</code></td>
<td>Gene symbol</td>
</tr>
<tr>
<td><code>n_exons</code></td>
<td>Number of exons with data</td>
</tr>
<tr>
<td><code>n_fragments</code></td>
<td>Total fragments across all exons</td>
</tr>
<tr>
<td><code>mds_mean</code></td>
<td>Mean <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> across all exons</td>
</tr>
<tr>
<td><code>mds_e1</code></td>
<td><abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> of E1 (first exon) only</td>
</tr>
<tr>
<td><code>mds_std</code></td>
<td>Standard deviation of per-exon <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr></td>
</tr>
<tr>
<td><code>mds_z</code></td>
<td>Z-score vs <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> (with <code>--pon-model</code>)</td>
</tr>
<tr>
<td><code>mds_e1_z</code></td>
<td>E1 <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> z-score vs <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></td>
</tr>
</tbody>
</table>
<h4 id="ml-use-case_12">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.MDS.gene.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"gene"</span><span class="p">)</span>
<a id="__codelineno-14-2" name="__codelineno-14-2" href="#__codelineno-14-2"></a>
<a id="__codelineno-14-3" name="__codelineno-14-3" href="#__codelineno-14-3"></a><span class="c1"># E1 MDS is highest-signal feature (promoter-proximal NDR)</span>
<a id="__codelineno-14-4" name="__codelineno-14-4" href="#__codelineno-14-4"></a><span class="n">X_e1</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"mds_e1"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>           <span class="c1"># 146 features for panel</span>
<a id="__codelineno-14-5" name="__codelineno-14-5" href="#__codelineno-14-5"></a><span class="n">X_z</span>  <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"mds_e1_z"</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># PON-normalized  zero-centered in healthy</span>
</code></pre></div>
<p><strong>Best for</strong>: Gene-level cancer classifiers, promoter aberration detection, combining with <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>-E1.</p>
<hr>
<h3 id="ocf-orientation-aware-cfdna-fragmentation"><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> (Orientation-aware <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> Fragmentation)</h3>
<p><strong>File:</strong> <code>{sample}.OCF.tsv</code> / <code>{sample}.OCF.ontarget.tsv</code> / <code>{sample}.OCF.offtarget.tsv</code></p>
<p>Tissue-of-origin scores based on strand asymmetry of fragment ends at tissue-specific open chromatin regions.</p>
<h4 id="columns_9">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>tissue</code></td>
<td>Tissue type (Liver, Lung, Colon, Placenta, etc.)</td>
</tr>
<tr>
<td><code>OCF</code></td>
<td>Raw score = <code>(U - D) / (U + D)</code>  upstream/downstream asymmetry</td>
</tr>
<tr>
<td><code>ocf_z</code></td>
<td><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> z-score vs <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> (with <code>--pon-model</code>)</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_11">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Tissue-of-origin</strong>: Which tissue contributed most <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr>  useful for cancer site-of-origin</li>
<li><strong>Multi-tissue mixture deconvolution</strong>: Multiple elevated <code>ocf_z</code> values suggest multi-tissue contribution</li>
<li><strong><abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> fraction proxy</strong>: Overall <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> magnitude correlates with <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> purity</li>
</ul>
<h4 id="ml-use-case_13">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.OCF.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"tissue"</span><span class="p">)</span>
<a id="__codelineno-15-2" name="__codelineno-15-2" href="#__codelineno-15-2"></a>
<a id="__codelineno-15-3" name="__codelineno-15-3" href="#__codelineno-15-3"></a><span class="c1"># OCF score vector across tissues (e.g. 10 tissues = 10 features)</span>
<a id="__codelineno-15-4" name="__codelineno-15-4" href="#__codelineno-15-4"></a><span class="n">X_ocf</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"OCF"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-15-5" name="__codelineno-15-5" href="#__codelineno-15-5"></a><span class="n">X_z</span>   <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"ocf_z"</span><span class="p">]</span><span class="o">.</span><span class="n">fillna</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># zero-centered in healthy</span>
<a id="__codelineno-15-6" name="__codelineno-15-6" href="#__codelineno-15-6"></a>
<a id="__codelineno-15-7" name="__codelineno-15-7" href="#__codelineno-15-7"></a><span class="c1"># Tissue with max signal</span>
<a id="__codelineno-15-8" name="__codelineno-15-8" href="#__codelineno-15-8"></a><span class="n">top_tissue</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"ocf_z"</span><span class="p">]</span><span class="o">.</span><span class="n">idxmax</span><span class="p">()</span>
</code></pre></div>
<p><strong>Best for</strong>: Primary tumor site-of-origin classification, multi-class tissue deconvolution.</p>
<hr>
<h3 id="ocf-sync"><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> Sync</h3>
<p><strong>File:</strong> <code>{sample}.OCF.sync.tsv</code></p>
<p>Positional strand-specific protection profiles  detailed positional data underlying the <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> summary score.</p>
<h4 id="columns-label-count-mean_size-entropy">Columns: <code>label</code>, <code>count</code>, <code>mean_size</code>, <code>entropy</code></h4>
<p><strong>Use</strong>: Raw data for visualizing strand phasing; input to advanced nucleosome positioning models.</p>
<hr>
<h3 id="tfbs-transcription-factor-binding-site-entropy"><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> (Transcription Factor Binding Site Entropy)</h3>
<p><strong>File:</strong> <code>{sample}.TFBS.tsv</code> / <code>{sample}.TFBS.ontarget.tsv</code></p>
<p>Fragment size entropy at <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> regions for 808 transcription factors. Reflects chromatin accessibility and TF binding.</p>
<h4 id="columns_10">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>label</code></td>
<td>TF name (e.g. <code>CTCF</code>, <code>SP1</code>, <code>E2F1</code>)</td>
</tr>
<tr>
<td><code>count</code></td>
<td>Fragment count</td>
</tr>
<tr>
<td><code>mean_size</code></td>
<td>Mean fragment size at this TF's sites</td>
</tr>
<tr>
<td><code>entropy</code></td>
<td>Shannon entropy of fragment size distribution</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_12">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>TF accessibility</strong>: Low entropy = dominated by one size class (nucleosomal); high entropy = mixed, accessible</li>
<li><strong>TF-specific cancer signal</strong>: Some TFs (E2F family, SP1) show altered fragmentation in cancer</li>
<li><strong>808-feature vector</strong>: One entropy value per TF = large, rich feature set</li>
</ul>
<h4 id="ml-use-case_14">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.TFBS.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"label"</span><span class="p">)</span>
<a id="__codelineno-16-2" name="__codelineno-16-2" href="#__codelineno-16-2"></a>
<a id="__codelineno-16-3" name="__codelineno-16-3" href="#__codelineno-16-3"></a><span class="c1"># 808 TF entropy values  high-dimensional feature vector</span>
<a id="__codelineno-16-4" name="__codelineno-16-4" href="#__codelineno-16-4"></a><span class="n">X_tfbs</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"entropy"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-16-5" name="__codelineno-16-5" href="#__codelineno-16-5"></a>
<a id="__codelineno-16-6" name="__codelineno-16-6" href="#__codelineno-16-6"></a><span class="c1"># Mean size as complementary feature</span>
<a id="__codelineno-16-7" name="__codelineno-16-7" href="#__codelineno-16-7"></a><span class="n">X_size</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"mean_size"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-16-8" name="__codelineno-16-8" href="#__codelineno-16-8"></a>
<a id="__codelineno-16-9" name="__codelineno-16-9" href="#__codelineno-16-9"></a><span class="c1"># Combine</span>
<a id="__codelineno-16-10" name="__codelineno-16-10" href="#__codelineno-16-10"></a><span class="n">X</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">stack</span><span class="p">([</span><span class="n">X_tfbs</span><span class="p">,</span> <span class="n">X_size</span><span class="p">],</span> <span class="n">axis</span><span class="o">=</span><span class="mi">1</span><span class="p">)</span>  <span class="c1"># shape: (808, 2)</span>
</code></pre></div>
<hr>
<h3 id="tfbs-sync"><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> Sync</h3>
<p><strong>File:</strong> <code>{sample}.TFBS.sync.tsv</code></p>
<p>Per-TF  per-size distribution  the raw size histogram for each TF.</p>
<h4 id="columns-label-size-count-proportion">Columns: <code>label</code>, <code>size</code>, <code>count</code>, <code>proportion</code></h4>
<p><strong>Use</strong>: Size-resolved TF footprinting; input for detecting nucleosome-footprint transitions.</p>
<hr>
<h3 id="atac-chromatin-accessibility-entropy"><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> (Chromatin Accessibility Entropy)</h3>
<p><strong>File:</strong> <code>{sample}.ATAC.tsv</code> / <code>{sample}.ATAC.ontarget.tsv</code></p>
<p>Fragment size entropy at <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr>-seq accessible regions for 23 cancer-relevant tissue types.</p>
<h4 id="columns_11">Columns</h4>
<p>Same as <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>: <code>label</code> (tissue type), <code>count</code>, <code>mean_size</code>, <code>entropy</code></p>
<h4 id="purpose-use-cases_13">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Tissue-specific accessibility</strong>: Which tissue's <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> peaks show altered fragmentation</li>
<li><strong>Cancer type inference</strong>: Different cancer types show tissue-specific <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> entropy patterns</li>
<li><strong>Complementary to <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr></strong>: <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> uses strand asymmetry; <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> uses size entropy  different signal axis</li>
</ul>
<h4 id="ml-use-case_15">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-17-1" name="__codelineno-17-1" href="#__codelineno-17-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.ATAC.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"label"</span><span class="p">)</span>
<a id="__codelineno-17-2" name="__codelineno-17-2" href="#__codelineno-17-2"></a>
<a id="__codelineno-17-3" name="__codelineno-17-3" href="#__codelineno-17-3"></a><span class="c1"># 23 tissue entropy values  compact tissue-of-origin feature</span>
<a id="__codelineno-17-4" name="__codelineno-17-4" href="#__codelineno-17-4"></a><span class="n">X_atac</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s2">"entropy"</span><span class="p">]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-17-5" name="__codelineno-17-5" href="#__codelineno-17-5"></a>
<a id="__codelineno-17-6" name="__codelineno-17-6" href="#__codelineno-17-6"></a><span class="c1"># Combine OCF + ATAC tissue vectors for multimodal tissue classifier</span>
<a id="__codelineno-17-7" name="__codelineno-17-7" href="#__codelineno-17-7"></a><span class="n">X_combined</span> <span class="o">=</span> <span class="n">np</span><span class="o">.</span><span class="n">concatenate</span><span class="p">([</span><span class="n">ocf_scores</span><span class="p">,</span> <span class="n">atac_entropy</span><span class="p">])</span>
</code></pre></div>
<hr>
<h3 id="atac-sync"><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> Sync</h3>
<p><strong>File:</strong> <code>{sample}.ATAC.sync.tsv</code></p>
<p>Per-tissue  per-size fragment distributions.</p>
<h4 id="columns-label-size-count-proportion_1">Columns: <code>label</code>, <code>size</code>, <code>count</code>, <code>proportion</code></h4>
<hr>
<h2 id="variant-level">Variant-Level</h2>
<h3 id="mfsd-mutant-fragment-size-distribution"><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> (Mutant Fragment Size Distribution)</h3>
<p><strong>File:</strong> <code>{sample}.mFSD.tsv</code><br>
<strong>Requires:</strong> <code>--maf</code> / <code>run-all</code> with MAF input</p>
<p>Per-variant fragment size analysis. Compares ALT-bearing fragments vs REF, NonREF, and N (uncertain) allele classes.</p>
<h4 id="column-groups-46-total">Column Groups (46 total)</h4>
<table>
<thead>
<tr>
<th>Group</th>
<th>Columns</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Variant</strong> (5)</td>
<td><code>Chrom</code>, <code>Pos</code>, <code>Ref</code>, <code>Alt</code>, <code>VarType</code></td>
<td>Variant coordinates and type</td>
</tr>
<tr>
<td><strong>Raw Counts</strong> (5)</td>
<td><code>REF_Count</code>, <code>ALT_Count</code>, <code>NonREF_Count</code>, <code>N_Count</code>, <code>Total_Count</code></td>
<td>Fragments per allele class</td>
</tr>
<tr>
<td><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-Weighted</strong> (5)</td>
<td><code>REF_Weighted</code>, <code>ALT_Weighted</code>, <code>NonREF_Weighted</code>, <code>N_Weighted</code>, <code>VAF_GC_Corrected</code></td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected counts and VAF</td>
</tr>
<tr>
<td><strong>Log-Likelihood</strong> (2)</td>
<td><code>ALT_LLR</code>, <code>REF_LLR</code></td>
<td>For low-N variants (ALT_Count &lt; 5)</td>
</tr>
<tr>
<td><strong>Mean Sizes</strong> (4)</td>
<td><code>REF_MeanSize</code>, <code>ALT_MeanSize</code>, <code>NonREF_MeanSize</code>, <code>N_MeanSize</code></td>
<td>Mean fragment size per class</td>
</tr>
<tr>
<td><strong>KS Tests</strong> (18)</td>
<td><code>Delta_*/KS_*/KS_Pval_*</code>  6 pairings</td>
<td>KS distance + p-value for each allele pair</td>
</tr>
<tr>
<td><strong>Derived</strong> (5)</td>
<td><code>VAF_Proxy</code>, <code>Error_Rate</code>, <code>N_Rate</code>, <code>Size_Ratio</code>, <code>Quality_Score</code></td>
<td>Summary biomarkers</td>
</tr>
<tr>
<td><strong>Flags</strong> (2)</td>
<td><code>ALT_Confidence</code>, <code>KS_Valid</code></td>
<td>Quality indicators</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_14">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>MRD detection</strong>: <code>VAF_GC_Corrected</code> + <code>KS_Pval_ALT_REF</code> identify true somatic mutations vs noise</li>
<li><strong>Fragment size as orthogonal evidence</strong>: <code>Delta_ALT_REF</code> negative = ALT fragments shorter than REF = tumor-derived <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr></li>
<li><strong>Duplex support</strong>: <code>ALT_LLR</code> provides statistically valid evidence even at 12 fragment counts</li>
<li><strong>Multi-variant aggregation</strong>: Combine evidence across many variants to estimate <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> fraction</li>
</ul>
<h4 id="ml-use-case_16">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-18-1" name="__codelineno-18-1" href="#__codelineno-18-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.mFSD.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span>
<a id="__codelineno-18-2" name="__codelineno-18-2" href="#__codelineno-18-2"></a>
<a id="__codelineno-18-3" name="__codelineno-18-3" href="#__codelineno-18-3"></a><span class="c1"># Filter to high-quality variants</span>
<a id="__codelineno-18-4" name="__codelineno-18-4" href="#__codelineno-18-4"></a><span class="n">df_hq</span> <span class="o">=</span> <span class="n">df</span><span class="p">[(</span><span class="n">df</span><span class="p">[</span><span class="s2">"ALT_Confidence"</span><span class="p">]</span> <span class="o">==</span> <span class="s2">"HIGH"</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s2">"KS_Valid"</span><span class="p">]</span> <span class="o">==</span> <span class="kc">True</span><span class="p">)]</span>
<a id="__codelineno-18-5" name="__codelineno-18-5" href="#__codelineno-18-5"></a>
<a id="__codelineno-18-6" name="__codelineno-18-6" href="#__codelineno-18-6"></a><span class="c1"># Per-variant feature vector</span>
<a id="__codelineno-18-7" name="__codelineno-18-7" href="#__codelineno-18-7"></a><span class="n">features</span> <span class="o">=</span> <span class="n">df_hq</span><span class="p">[[</span><span class="s2">"VAF_GC_Corrected"</span><span class="p">,</span> <span class="s2">"Delta_ALT_REF"</span><span class="p">,</span> <span class="s2">"KS_ALT_REF"</span><span class="p">,</span>
<a id="__codelineno-18-8" name="__codelineno-18-8" href="#__codelineno-18-8"></a>                   <span class="s2">"Size_Ratio"</span><span class="p">,</span> <span class="s2">"Quality_Score"</span><span class="p">,</span> <span class="s2">"ALT_LLR"</span><span class="p">]]</span><span class="o">.</span><span class="n">values</span>
<a id="__codelineno-18-9" name="__codelineno-18-9" href="#__codelineno-18-9"></a>
<a id="__codelineno-18-10" name="__codelineno-18-10" href="#__codelineno-18-10"></a><span class="c1"># Sample-level aggregation: weighted average across variants</span>
<a id="__codelineno-18-11" name="__codelineno-18-11" href="#__codelineno-18-11"></a><span class="n">sample_vaf</span> <span class="o">=</span> <span class="p">(</span><span class="n">df_hq</span><span class="p">[</span><span class="s2">"VAF_GC_Corrected"</span><span class="p">]</span> <span class="o">*</span> <span class="n">df_hq</span><span class="p">[</span><span class="s2">"Quality_Score"</span><span class="p">])</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span> <span class="o">/</span> <span class="n">df_hq</span><span class="p">[</span><span class="s2">"Quality_Score"</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</code></pre></div>
<p><strong>Best for</strong>: MRD detection models, <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> fraction estimation from targeted panels, variant-level cancer classifiers.</p>
<hr>
<h3 id="mfsd-distributions"><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> Distributions</h3>
<p><strong>File:</strong> <code>{sample}.mFSD.distributions.tsv</code><br>
<strong>Requires:</strong> <code>--output-distributions</code> flag</p>
<p>Per-variant raw size histograms for manual inspection.</p>
<h4 id="columns-chrom-pos-ref-alt-category-size-count">Columns: <code>Chrom</code>, <code>Pos</code>, <code>Ref</code>, <code>Alt</code>, <code>Category</code>, <code>Size</code>, <code>Count</code></h4>
<p><strong>Use</strong>: Visualizing the ALT vs REF size shift for individual variants; QC and debugging.</p>
<hr>
<h2 id="methylation">Methylation</h2>
<h3 id="uxm-methylation"><abbr title="Fragment-level Methylation - Methylation state of individual fragments">UXM</abbr> (Methylation)</h3>
<p><strong>File:</strong> <code>{sample}.UXM.tsv</code><br>
<strong>Requires:</strong> Methylation-enabled <abbr title="Binary Alignment Map - Compressed sequence alignment file format">BAM</abbr> input</p>
<p>CpG methylation classification per genomic region. Each fragment is classified as Unmethylated (U), Partially-methylated (X), or fully Methylated (M).</p>
<h4 id="columns_12">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>region</code></td>
<td>Genomic region identifier</td>
</tr>
<tr>
<td><code>U</code></td>
<td>Fraction of unmethylated fragments</td>
</tr>
<tr>
<td><code>X</code></td>
<td>Fraction of partially methylated fragments</td>
</tr>
<tr>
<td><code>M</code></td>
<td>Fraction of fully methylated fragments</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_15">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Tumor DNA methylation</strong>: Cancer shows global hypomethylation (U) and focal hypermethylation (M at TSGs)</li>
<li><strong>Tissue-of-origin</strong>: Tissue-specific methylation patterns enable <abbr title="Circulating tumor DNA - Tumor-derived cfDNA fragments">ctDNA</abbr> source attribution</li>
<li><strong>Multi-modal fusion</strong>: Combine with <abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> and <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> for maximum sensitivity in early detection</li>
</ul>
<h4 id="ml-use-case_17">ML Use Case</h4>
<div class="highlight"><pre><span></span><code><a id="__codelineno-19-1" name="__codelineno-19-1" href="#__codelineno-19-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="s2">"sample.UXM.tsv"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"</span><span class="se">\t</span><span class="s2">"</span><span class="p">)</span><span class="o">.</span><span class="n">set_index</span><span class="p">(</span><span class="s2">"region"</span><span class="p">)</span>
<a id="__codelineno-19-2" name="__codelineno-19-2" href="#__codelineno-19-2"></a>
<a id="__codelineno-19-3" name="__codelineno-19-3" href="#__codelineno-19-3"></a><span class="c1"># 3-class methylation state per region</span>
<a id="__codelineno-19-4" name="__codelineno-19-4" href="#__codelineno-19-4"></a><span class="n">X</span> <span class="o">=</span> <span class="n">df</span><span class="p">[[</span><span class="s2">"U"</span><span class="p">,</span> <span class="s2">"X"</span><span class="p">,</span> <span class="s2">"M"</span><span class="p">]]</span><span class="o">.</span><span class="n">values</span>  <span class="c1"># shape: (n_regions, 3); rows sum to 1.0</span>
<a id="__codelineno-19-5" name="__codelineno-19-5" href="#__codelineno-19-5"></a>
<a id="__codelineno-19-6" name="__codelineno-19-6" href="#__codelineno-19-6"></a><span class="c1"># Sample-level statistics</span>
<a id="__codelineno-19-7" name="__codelineno-19-7" href="#__codelineno-19-7"></a><span class="n">features</span> <span class="o">=</span> <span class="p">{</span>
<a id="__codelineno-19-8" name="__codelineno-19-8" href="#__codelineno-19-8"></a>    <span class="s2">"global_U"</span><span class="p">:</span> <span class="n">df</span><span class="p">[</span><span class="s2">"U"</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>   <span class="c1"># high = hypomethylation (cancer signal)</span>
<a id="__codelineno-19-9" name="__codelineno-19-9" href="#__codelineno-19-9"></a>    <span class="s2">"global_M"</span><span class="p">:</span> <span class="n">df</span><span class="p">[</span><span class="s2">"M"</span><span class="p">]</span><span class="o">.</span><span class="n">mean</span><span class="p">(),</span>   <span class="c1"># varies by tissue</span>
<a id="__codelineno-19-10" name="__codelineno-19-10" href="#__codelineno-19-10"></a>    <span class="s2">"U_std"</span><span class="p">:</span> <span class="n">df</span><span class="p">[</span><span class="s2">"U"</span><span class="p">]</span><span class="o">.</span><span class="n">std</span><span class="p">(),</span>       <span class="c1"># heterogeneity across regions</span>
<a id="__codelineno-19-11" name="__codelineno-19-11" href="#__codelineno-19-11"></a><span class="p">}</span>
</code></pre></div>
<hr>
<h2 id="diagnostic-qc-outputs">Diagnostic / QC Outputs</h2>
<h3 id="gc-correction-factors"><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction Factors</h3>
<p><strong>File:</strong> <code>{sample}.correction_factors.tsv</code> / <code>{sample}.correction_factors.ontarget.tsv</code></p>
<p><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-bias correction weights per (fragment length bin, <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> content) pair.</p>
<h4 id="columns_13">Columns</h4>
<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>length_bin_min</code></td>
<td>Fragment length bin lower bound (bp)</td>
</tr>
<tr>
<td><code>length_bin_max</code></td>
<td>Fragment length bin upper bound (bp)</td>
</tr>
<tr>
<td><code>gc_percent</code></td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> content percentage (0100)</td>
</tr>
<tr>
<td><code>observed</code></td>
<td>Observed fragment count</td>
</tr>
<tr>
<td><code>expected</code></td>
<td>Expected count from <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model</td>
</tr>
<tr>
<td><code>correction_factor</code></td>
<td><code>expected / observed</code>  multiply raw counts by this</td>
</tr>
</tbody>
</table>
<h4 id="purpose-use-cases_16">Purpose &amp; Use Cases</h4>
<ul>
<li><strong>Library QC</strong>: Values far from 1.0 indicate <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias in sequencing/capture</li>
<li><strong>Batch effect debugging</strong>: Compare correction factors across runs to detect systematic issues</li>
<li><strong>Input to <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></strong>: Used to build <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> correction factor baselines</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Not for ML features</p>
<p><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction is already applied to <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>, <abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr>, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr>, and <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> outputs. Use those corrected values. <code>correction_factors.tsv</code> is diagnostic data.</p>
</div>
<hr>
<h3 id="metadata-json">Metadata JSON</h3>
<p><strong>File:</strong> <code>{sample}.metadata.json</code></p>
<p>Run parameters, QC metrics, and processing provenance.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-20-1" name="__codelineno-20-1" href="#__codelineno-20-1"></a><span class="p">{</span>
<a id="__codelineno-20-2" name="__codelineno-20-2" href="#__codelineno-20-2"></a><span class="w">  </span><span class="nt">"sample_id"</span><span class="p">:</span><span class="w"> </span><span class="s2">"sample_001"</span><span class="p">,</span>
<a id="__codelineno-20-3" name="__codelineno-20-3" href="#__codelineno-20-3"></a><span class="w">  </span><span class="nt">"krewlyzer_version"</span><span class="p">:</span><span class="w"> </span><span class="s2">"0.6.0"</span><span class="p">,</span>
<a id="__codelineno-20-4" name="__codelineno-20-4" href="#__codelineno-20-4"></a><span class="w">  </span><span class="nt">"genome"</span><span class="p">:</span><span class="w"> </span><span class="s2">"hg19"</span><span class="p">,</span>
<a id="__codelineno-20-5" name="__codelineno-20-5" href="#__codelineno-20-5"></a><span class="w">  </span><span class="nt">"assay"</span><span class="p">:</span><span class="w"> </span><span class="s2">"xs2"</span><span class="p">,</span>
<a id="__codelineno-20-6" name="__codelineno-20-6" href="#__codelineno-20-6"></a><span class="w">  </span><span class="nt">"total_fragments"</span><span class="p">:</span><span class="w"> </span><span class="mi">8234567</span><span class="p">,</span>
<a id="__codelineno-20-7" name="__codelineno-20-7" href="#__codelineno-20-7"></a><span class="w">  </span><span class="nt">"on_target_rate"</span><span class="p">:</span><span class="w"> </span><span class="mf">0.44</span><span class="p">,</span>
<a id="__codelineno-20-8" name="__codelineno-20-8" href="#__codelineno-20-8"></a><span class="w">  </span><span class="nt">"mean_fragment_size"</span><span class="p">:</span><span class="w"> </span><span class="mf">168.3</span><span class="p">,</span>
<a id="__codelineno-20-9" name="__codelineno-20-9" href="#__codelineno-20-9"></a><span class="w">  </span><span class="nt">"duplication_rate"</span><span class="p">:</span><span class="w"> </span><span class="mf">0.12</span><span class="p">,</span>
<a id="__codelineno-20-10" name="__codelineno-20-10" href="#__codelineno-20-10"></a><span class="w">  </span><span class="nt">"processing_time_s"</span><span class="p">:</span><span class="w"> </span><span class="mf">342.1</span>
<a id="__codelineno-20-11" name="__codelineno-20-11" href="#__codelineno-20-11"></a><span class="p">}</span>
</code></pre></div>
<p><strong>Use</strong>: Filter samples by QC thresholds before ML training (e.g. <code>total_fragments &gt; 5M</code>, <code>on_target_rate &gt; 0.3</code>).</p>
<hr>
<h3 id="features-json">Features JSON</h3>
<p><strong>File:</strong> <code>{sample}.features.json</code><br>
<strong>Requires:</strong> <code>--generate-json</code></p>
<p>Unified export of all features above in a single JSON file for ML pipelines. See <a href="../../features/output/json-output/">JSON Output Reference</a> for the complete schema.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-21-1" name="__codelineno-21-1" href="#__codelineno-21-1"></a><span class="kn">import</span><span class="w"> </span><span class="nn">json</span>
<a id="__codelineno-21-2" name="__codelineno-21-2" href="#__codelineno-21-2"></a><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">"sample.features.json"</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
<a id="__codelineno-21-3" name="__codelineno-21-3" href="#__codelineno-21-3"></a>    <span class="n">features</span> <span class="o">=</span> <span class="n">json</span><span class="o">.</span><span class="n">load</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
<a id="__codelineno-21-4" name="__codelineno-21-4" href="#__codelineno-21-4"></a>
<a id="__codelineno-21-5" name="__codelineno-21-5" href="#__codelineno-21-5"></a><span class="c1"># Access any output without reading individual TSVs</span>
<a id="__codelineno-21-6" name="__codelineno-21-6" href="#__codelineno-21-6"></a><span class="n">fsr_windows</span> <span class="o">=</span> <span class="n">features</span><span class="p">[</span><span class="s2">"fsr"</span><span class="p">][</span><span class="s2">"off_target"</span><span class="p">]</span>
<a id="__codelineno-21-7" name="__codelineno-21-7" href="#__codelineno-21-7"></a><span class="n">mds</span> <span class="o">=</span> <span class="n">features</span><span class="p">[</span><span class="s2">"motif"</span><span class="p">][</span><span class="s2">"mds"</span><span class="p">]</span>
<a id="__codelineno-21-8" name="__codelineno-21-8" href="#__codelineno-21-8"></a><span class="n">region_mds</span> <span class="o">=</span> <span class="n">features</span><span class="p">[</span><span class="s2">"region_mds"</span><span class="p">][</span><span class="s2">"gene"</span><span class="p">]</span>
</code></pre></div>
<hr>
<h2 id="feature-selection-guide-for-ml-models">Feature Selection Guide for ML Models</h2>
<table>
<thead>
<tr>
<th>Model type</th>
<th>Recommended features</th>
<th>File(s)</th>
</tr>
</thead>
<tbody>
<tr>
<td>Pan-cancer screen (WGS)</td>
<td><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> <code>short_long_log2</code>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>, <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> <code>prot_frac_nuc_mean</code>, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr> <code>short_frac</code></td>
<td><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>, WPS_background, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr></td>
</tr>
<tr>
<td>Pan-cancer screen (panel)</td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>-E1 <code>promoter_short</code>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>-E1, <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> <code>ocf_z</code>, <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> panel</td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.e1only, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>.gene, <abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr>, <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr>.panel</td>
</tr>
<tr>
<td>Tumor fraction regression</td>
<td><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> <code>short_long_log2</code> genome-wide, <abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> <code>nrl_bp</code></td>
<td><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr>, WPS_background</td>
</tr>
<tr>
<td>Cancer type / site-of-origin</td>
<td><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr> tissue vector, <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> tissue vector, <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> vector</td>
<td><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr>, <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr>, <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr></td>
</tr>
<tr>
<td>Gene-level amplification</td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> gene <code>normalized_depth</code>, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr> arm <code>total</code></td>
<td><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.gene, <abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr></td>
</tr>
<tr>
<td>MRD (residual disease)</td>
<td><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr> <code>VAF_GC_Corrected</code>, <code>Quality_Score</code>, <code>Delta_ALT_REF</code></td>
<td><abbr title="Mutant Fragment Size Distribution - Size distribution around somatic mutations">mFSD</abbr></td>
</tr>
<tr>
<td>Promoter accessibility</td>
<td><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr> E1 <code>wps_nuc_mean</code>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr> E1, <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>-E1 ratios</td>
<td><abbr title="Windowed Protection Score - Nucleosome positioning signal around regulatory sites">WPS</abbr>.panel, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>.gene, <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>.e1only</td>
</tr>
<tr>
<td>Methylation-augmented</td>
<td><abbr title="Fragment-level Methylation - Methylation state of individual fragments">UXM</abbr> <code>U</code>/<code>M</code> + <abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> + <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr></td>
<td><abbr title="Fragment-level Methylation - Methylation state of individual fragments">UXM</abbr>, <abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr>, <abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr></td>
</tr>
</tbody>
</table>
<hr>
<h2 id="see-also">See Also</h2>
<ul>
<li><a href="../../features/output/json-output/">JSON Output Reference</a>  unified JSON schema for all features</li>
<li><a href="../../features/core/fsc/#fsc-ratios-vs-fsr--not-redundant"><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr> vs <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> ratios</a>  why they differ</li>
<li><a href="../pon-models/"><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> Models</a>  normalization baselines</li>
<li><a href="../../guides/gc-correction/"><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction</a>  correction methodology</li>
</ul>
<!-- Abbreviation definitions for Krewlyzer documentation -->
<!-- These provide hover tooltips for technical acronyms -->







  
    
  
  
    
  


  <aside class="md-source-file">
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Last update">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="March 1, 2026 03:31:25 UTC"><span class="timeago" datetime="2026-03-01T03:31:25+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="March 1, 2026 03:31:25 UTC">2026-03-01</span>
  </span>

    
    
      
  <span class="md-source-file__fact">
    <span class="md-icon" title="Created">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
    </span>
    <span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="March 1, 2026 03:28:02 UTC"><span class="timeago" datetime="2026-03-01T03:28:02+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="March 1, 2026 03:28:02 UTC">2026-03-01</span>
  </span>

    
    
    
  </aside>





                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Developed by <a href="https://github.com/rhshah">Ronak H Shah</a> @ 
<a href="https://www.mskcc.org">MSKCC</a> | 
Built with <a href="https://antigravity.google/">Antigravity</a> | 
<a href="https://github.com/msk-access/krewlyzer/blob/main/LICENSE">AGPL-3.0</a>

    </div>
  
  
</div>
      
        
<div class="md-social">
  
    
    
    
    
      
      
    
    <a href="https://github.com/msk-access/krewlyzer" target="_blank" rel="noopener" title="github.com" class="md-social__link">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "../..", "features": ["navigation.instant", "navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.indexes", "navigation.path", "toc.follow", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"default": "stable", "provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.79ae519e.min.js"></script>
      
        <script src="../../js/timeago.min.js"></script>
      
        <script src="../../js/timeago_mkdocs_material.js"></script>
      
        <script src="../../javascripts/mathjax.js"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
        <script src="../../assets/javascripts/panzoom.min.js"></script>
      
        <script src="../../assets/javascripts/zoompan.js"></script>
      
    
  
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>