<!DOCTYPE html><html class="no-js" lang="en"><head>
<meta charset="utf-8">
<meta content="width=device-width,initial-scale=1" name="viewport">
<link href="https://msk-access.github.io/krewlyzer/dev/features/core/fsc/" rel="canonical">
<link href="../extract/" rel="prev">
<link href="../fsd/" rel="next">
<link href="../../../assets/images/favicon.png" rel="icon">
<meta content="mkdocs-1.6.1, mkdocs-material-9.7.3" name="generator">
<title>FSC - Krewlyzer</title>
<link href="../../../assets/stylesheets/main.484c7ddc.min.css" rel="stylesheet">
<link href="../../../assets/stylesheets/palette.ab4e12ef.min.css" rel="stylesheet">
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect">
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet">
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<link href="../../../css/timeago.css" rel="stylesheet">
<link href="../../../stylesheets/extra.css" rel="stylesheet">
<link href="../../../assets/stylesheets/panzoom.css" rel="stylesheet">
<script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<meta content="website" property="og:type">
<meta content="FSC - Krewlyzer" property="og:title">
<meta content="https://msk-access.github.io/krewlyzer/dev/assets/images/social/features/core/fsc.png" property="og:image">
<meta content="image/png" property="og:image:type">
<meta content="1200" property="og:image:width">
<meta content="630" property="og:image:height">
<meta content="https://msk-access.github.io/krewlyzer/dev/features/core/fsc/" property="og:url">
<meta content="summary_large_image" property="twitter:card">
<meta content="FSC - Krewlyzer" property="twitter:title">
<meta content="https://msk-access.github.io/krewlyzer/dev/assets/images/social/features/core/fsc.png" property="twitter:image">
<link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style><meta name="panzoom-theme" content="material">
<meta name="panzoom-data" content='{"selectors": [".mermaid", ".d2"], "initial_zoom_level": 1.0}'> </head>
<body data-md-color-accent="custom" data-md-color-primary="custom" data-md-color-scheme="default" dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox">
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox">
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#fragment-size-coverage-fsc">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<div data-md-color-scheme="default" data-md-component="outdated" hidden="">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Krewlyzer" class="md-header__button md-logo" data-md-component="logo" href="../../.." title="Krewlyzer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Krewlyzer
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              FSC
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="custom" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="custom" data-md-color-scheme="default" id="__palette_0" name="__palette" type="radio">
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
</label>
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="custom" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="custom" data-md-color-scheme="slate" id="__palette_1" name="__palette" type="radio">
<label class="md-header__button md-icon" for="__palette_0" hidden="" title="Switch to light mode">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
</label>
</form>
<script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
<label class="md-header__button md-icon" for="__search">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text">
<label class="md-search__icon md-icon" for="__search">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/msk-access/krewlyzer" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Krewlyzer" class="md-nav__button md-logo" data-md-component="logo" href="../../.." title="Krewlyzer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
    Krewlyzer
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/msk-access/krewlyzer" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../..">
<span class="md-ellipsis">
    
  
    Home
  

    
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../getting-started/">
<span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/installation/">
<span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/quickstart/">
<span class="md-ellipsis">
    
  
    Quickstart
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/concepts/">
<span class="md-ellipsis">
    
  
    Concepts
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../cli/">
<span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
            
  
    CLI Reference
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../cli/run-all/">
<span class="md-ellipsis">
    
  
    run-all
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../nextflow/">
<span class="md-ellipsis">
    
  
    Nextflow Pipeline
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
            
  
    Nextflow Pipeline
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/samplesheet/">
<span class="md-ellipsis">
    
  
    Samplesheet
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/parameters/">
<span class="md-ellipsis">
    
  
    Parameters
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/outputs/">
<span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/examples/">
<span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../">
<span class="md-ellipsis">
    
  
    Features
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="true" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
            
  
    Features
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5_2" type="checkbox">
<label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Core
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_5_2_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_2">
<span class="md-nav__icon md-icon"></span>
            
  
    Core
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../extract/">
<span class="md-ellipsis">
    
  
    Fragment Extraction
  

    
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox">
<label class="md-nav__link md-nav__link--active" for="__toc">
<span class="md-ellipsis">
    
  
    FSC
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
<span class="md-ellipsis">
    
  
    FSC
  

    
  </span>
</a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#purpose">
<span class="md-ellipsis">
      
        Purpose
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-context">
<span class="md-ellipsis">
      
        Biological Context
      
    </span>
</a>
<nav aria-label="Biological Context" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#why-fragment-sizes-matter">
<span class="md-ellipsis">
      
        Why Fragment Sizes Matter
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#6-channel-ml-features">
<span class="md-ellipsis">
      
        6-Channel ML Features
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#implementation-details">
<span class="md-ellipsis">
      
        Implementation Details
      
    </span>
</a>
<nav aria-label="Implementation Details" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#counting-pipeline">
<span class="md-ellipsis">
      
        Counting Pipeline
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#pythonrust-architecture">
<span class="md-ellipsis">
      
        Python/Rust Architecture
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#aggregation-strategy">
<span class="md-ellipsis">
      
        Aggregation Strategy
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gc-correction-details">
<span class="md-ellipsis">
      
        GC Correction Details
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#bin-assignment">
<span class="md-ellipsis">
      
        Bin Assignment
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usage">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cli-options">
<span class="md-ellipsis">
      
        CLI Options
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-format">
<span class="md-ellipsis">
      
        Output Format
      
    </span>
</a>
<nav aria-label="Output Format" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#base-columns-always-present">
<span class="md-ellipsis">
      
        Base Columns (always present)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#pon-columns-when-pon-model-provided">
<span class="md-ellipsis">
      
        PoN Columns (when --pon-model provided)
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#wgs-vs-panel-data-msk-access">
<span class="md-ellipsis">
      
        WGS vs Panel Data (MSK-ACCESS)
      
    </span>
</a>
<nav aria-label="WGS vs Panel Data (MSK-ACCESS)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-mode-details">
<span class="md-ellipsis">
      
        Panel Mode Details
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-recommendations">
<span class="md-ellipsis">
      
        Panel Recommendations
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ml-feature-engineering">
<span class="md-ellipsis">
      
        ML Feature Engineering
      
    </span>
</a>
<nav aria-label="ML Feature Engineering" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#raw-features-per-5mb-window">
<span class="md-ellipsis">
      
        Raw Features (per 5Mb window)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#derived-features-recommended">
<span class="md-ellipsis">
      
        Derived Features (recommended)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#example-short-to-long-ratio">
<span class="md-ellipsis">
      
        Example: Short-to-Long Ratio
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#normalization-order">
<span class="md-ellipsis">
      
        Normalization Order
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-mode-target-regions">
<span class="md-ellipsis">
      
        Panel Mode (--target-regions)
      
    </span>
</a>
<nav aria-label="Panel Mode (--target-regions)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#processing-with-target-regions">
<span class="md-ellipsis">
      
        Processing with Target Regions
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-files">
<span class="md-ellipsis">
      
        Output Files
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#when-to-use-on-target-fsc">
<span class="md-ellipsis">
      
        When to Use On-Target FSC
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gene-centric-fsc-msk-access">
<span class="md-ellipsis">
      
        Gene-Centric FSC (MSK-ACCESS)
      
    </span>
</a>
<nav aria-label="Gene-Centric FSC (MSK-ACCESS)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#output-files_1">
<span class="md-ellipsis">
      
        Output Files
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gene-fsc-output-format-samplefscgenetsv">
<span class="md-ellipsis">
      
        Gene FSC Output Format ({sample}.FSC.gene.tsv)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#region-fsc-output-format-samplefscregionstsv">
<span class="md-ellipsis">
      
        Region FSC Output Format ({sample}.FSC.regions.tsv)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#e1-only-fsc-output">
<span class="md-ellipsis">
      
        E1-Only FSC Output
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#normalized-depth-rpkm-like">
<span class="md-ellipsis">
      
        Normalized Depth (RPKM-like)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#supported-assays">
<span class="md-ellipsis">
      
        Supported Assays
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gc-correction-for-gene-fsc">
<span class="md-ellipsis">
      
        GC Correction for Gene FSC
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#see-also">
<span class="md-ellipsis">
      
        See Also
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#references">
<span class="md-ellipsis">
      
        References
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../fsd/">
<span class="md-ellipsis">
    
  
    FSD
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../fsr/">
<span class="md-ellipsis">
    
  
    FSR
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../wps/">
<span class="md-ellipsis">
    
  
    WPS
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_3" type="checkbox">
<label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Regulatory
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_3_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_3">
<span class="md-nav__icon md-icon"></span>
            
  
    Regulatory
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/motif/">
<span class="md-ellipsis">
    
  
    End Motif
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/ocf/">
<span class="md-ellipsis">
    
  
    OCF
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/region-entropy/">
<span class="md-ellipsis">
    
  
    Region Entropy
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../regulatory/region-mds/">
<span class="md-ellipsis">
    
  
    Region MDS
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_4" type="checkbox">
<label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Variant
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_4_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_4">
<span class="md-nav__icon md-icon"></span>
            
  
    Variant
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../variant/mfsd/">
<span class="md-ellipsis">
    
  
    mFSD
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_5" type="checkbox">
<label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Methylation
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_5_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_5">
<span class="md-nav__icon md-icon"></span>
            
  
    Methylation
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../methylation/uxm/">
<span class="md-ellipsis">
    
  
    UXM
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_6" type="checkbox">
<label class="md-nav__link" for="__nav_5_6" id="__nav_5_6_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Output
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_6_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_6">
<span class="md-nav__icon md-icon"></span>
            
  
    Output
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../output/json-output/">
<span class="md-ellipsis">
    
  
    JSON Export
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox">
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="">
<span class="md-ellipsis">
    
  
    Guides
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
            
  
    Guides
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/panel-mode/">
<span class="md-ellipsis">
    
  
    Panel Mode
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/msk-access-quickstart/">
<span class="md-ellipsis">
    
  
    MSK-ACCESS Quickstart
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/building-pon/">
<span class="md-ellipsis">
    
  
    Building PON
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/gc-correction/">
<span class="md-ellipsis">
    
  
    GC Correction
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox">
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="">
<span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/pon-models/">
<span class="md-ellipsis">
    
  
    PON Models
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/input-formats/">
<span class="md-ellipsis">
    
  
    Input Formats
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/architecture/">
<span class="md-ellipsis">
    
  
    Architecture
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/glossary/">
<span class="md-ellipsis">
    
  
    Glossary
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_8" type="checkbox">
<label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="">
<span class="md-ellipsis">
    
  
    Development
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_8_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_8">
<span class="md-nav__icon md-icon"></span>
            
  
    Development
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/developer-guide/">
<span class="md-ellipsis">
    
  
    Developer Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/testing-guide/">
<span class="md-ellipsis">
    
  
    Testing Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/release-guide/">
<span class="md-ellipsis">
    
  
    Release Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/changelog/">
<span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/contributing/">
<span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/contributing-data/">
<span class="md-ellipsis">
    
  
    Contributing Data
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_9" type="checkbox">
<label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="">
<span class="md-ellipsis">
    
  
    Resources
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_9_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_9">
<span class="md-nav__icon md-icon"></span>
            
  
    Resources
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/overview/">
<span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/citation/">
<span class="md-ellipsis">
    
  
    Citation
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/troubleshooting/">
<span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#purpose">
<span class="md-ellipsis">
      
        Purpose
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-context">
<span class="md-ellipsis">
      
        Biological Context
      
    </span>
</a>
<nav aria-label="Biological Context" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#why-fragment-sizes-matter">
<span class="md-ellipsis">
      
        Why Fragment Sizes Matter
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#6-channel-ml-features">
<span class="md-ellipsis">
      
        6-Channel ML Features
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#implementation-details">
<span class="md-ellipsis">
      
        Implementation Details
      
    </span>
</a>
<nav aria-label="Implementation Details" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#counting-pipeline">
<span class="md-ellipsis">
      
        Counting Pipeline
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#pythonrust-architecture">
<span class="md-ellipsis">
      
        Python/Rust Architecture
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#aggregation-strategy">
<span class="md-ellipsis">
      
        Aggregation Strategy
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gc-correction-details">
<span class="md-ellipsis">
      
        GC Correction Details
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#bin-assignment">
<span class="md-ellipsis">
      
        Bin Assignment
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usage">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cli-options">
<span class="md-ellipsis">
      
        CLI Options
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-format">
<span class="md-ellipsis">
      
        Output Format
      
    </span>
</a>
<nav aria-label="Output Format" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#base-columns-always-present">
<span class="md-ellipsis">
      
        Base Columns (always present)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#pon-columns-when-pon-model-provided">
<span class="md-ellipsis">
      
        PoN Columns (when --pon-model provided)
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#wgs-vs-panel-data-msk-access">
<span class="md-ellipsis">
      
        WGS vs Panel Data (MSK-ACCESS)
      
    </span>
</a>
<nav aria-label="WGS vs Panel Data (MSK-ACCESS)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-mode-details">
<span class="md-ellipsis">
      
        Panel Mode Details
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-recommendations">
<span class="md-ellipsis">
      
        Panel Recommendations
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ml-feature-engineering">
<span class="md-ellipsis">
      
        ML Feature Engineering
      
    </span>
</a>
<nav aria-label="ML Feature Engineering" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#raw-features-per-5mb-window">
<span class="md-ellipsis">
      
        Raw Features (per 5Mb window)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#derived-features-recommended">
<span class="md-ellipsis">
      
        Derived Features (recommended)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#example-short-to-long-ratio">
<span class="md-ellipsis">
      
        Example: Short-to-Long Ratio
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#normalization-order">
<span class="md-ellipsis">
      
        Normalization Order
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-mode-target-regions">
<span class="md-ellipsis">
      
        Panel Mode (--target-regions)
      
    </span>
</a>
<nav aria-label="Panel Mode (--target-regions)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#processing-with-target-regions">
<span class="md-ellipsis">
      
        Processing with Target Regions
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-files">
<span class="md-ellipsis">
      
        Output Files
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#when-to-use-on-target-fsc">
<span class="md-ellipsis">
      
        When to Use On-Target FSC
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gene-centric-fsc-msk-access">
<span class="md-ellipsis">
      
        Gene-Centric FSC (MSK-ACCESS)
      
    </span>
</a>
<nav aria-label="Gene-Centric FSC (MSK-ACCESS)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#output-files_1">
<span class="md-ellipsis">
      
        Output Files
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gene-fsc-output-format-samplefscgenetsv">
<span class="md-ellipsis">
      
        Gene FSC Output Format ({sample}.FSC.gene.tsv)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#region-fsc-output-format-samplefscregionstsv">
<span class="md-ellipsis">
      
        Region FSC Output Format ({sample}.FSC.regions.tsv)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#e1-only-fsc-output">
<span class="md-ellipsis">
      
        E1-Only FSC Output
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#normalized-depth-rpkm-like">
<span class="md-ellipsis">
      
        Normalized Depth (RPKM-like)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#supported-assays">
<span class="md-ellipsis">
      
        Supported Assays
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#gc-correction-for-gene-fsc">
<span class="md-ellipsis">
      
        GC Correction for Gene FSC
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#see-also">
<span class="md-ellipsis">
      
        See Also
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#references">
<span class="md-ellipsis">
      
        References
      
    </span>
</a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<nav aria-label="Navigation" class="md-path">
<ol class="md-path__list">
<li class="md-path__item">
<a class="md-path__link" href="../../..">
<span class="md-ellipsis">
    Home
  </span>
</a>
</li>
<li class="md-path__item">
<a class="md-path__link" href="../../">
<span class="md-ellipsis">
    Features
  </span>
</a>
</li>
<li class="md-path__item">
<a class="md-path__link" href="../extract/">
<span class="md-ellipsis">
    Core
  </span>
</a>
</li>
</ol>
</nav>
<article class="md-content__inner md-typeset">
<h1 id="fragment-size-coverage-fsc">Fragment Size Coverage (<abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr>)</h1>
<p><strong>Command</strong>: <code>krewlyzer fsc</code></p>
<div class="admonition info">
<p class="admonition-title">Plain English</p>
<p><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> counts how many DNA fragments of each size fall into each genomic region.
Think of it as a "heatmap" of fragment sizes across the genome.</p>
<p><strong>Use case</strong>: Copy number detection - regions with more short fragments suggest tumor amplification.</p>
</div>
<hr>
<h2 id="purpose">Purpose</h2>
<p>Computes <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-corrected coverage of <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> fragments in 6 biologically-meaningful size channels per genomic bin. Designed for ML feature extraction in cancer detection.</p>
<hr>
<h2 id="biological-context">Biological Context</h2>
<h3 id="why-fragment-sizes-matter">Why Fragment Sizes Matter</h3>
<p><abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> fragments are not randomâ€”their sizes reflect the chromatin state of their source cells:</p>
<table>
<thead>
<tr>
<th>Fragment Size</th>
<th>Source</th>
<th>Biological Meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Short (&lt;100bp)</strong></td>
<td>Open chromatin, active transcription</td>
<td>TF footprints, regulatory elements</td>
</tr>
<tr>
<td><strong>~167bp</strong></td>
<td>Mono-nucleosome</td>
<td>"Classic" <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> peak</td>
</tr>
<tr>
<td><strong>~334bp</strong></td>
<td>Di-nucleosome</td>
<td>Linked nucleosomes</td>
</tr>
<tr>
<td><strong>Long (&gt;260bp)</strong></td>
<td>Multi-nucleosome</td>
<td>Necrosis, incomplete digestion</td>
</tr>
</tbody>
</table>
<p><strong>Cancer signature</strong>: Tumors release shorter <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> fragments than healthy cells due to:
- Altered nucleosome positioning
- Different apoptotic pathways
- Chromatin accessibility changes</p>
<h3 id="6-channel-ml-features">6-Channel ML Features</h3>
<p><abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> partitions fragments into <strong>non-overlapping</strong> channels optimized for ML:</p>
<table>
<thead>
<tr>
<th>Channel</th>
<th>Size Range</th>
<th>Biological Meaning</th>
<th>Cancer Relevance</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>ultra_short</strong></td>
<td>65-100bp</td>
<td>Di-nucleosomal debris</td>
<td>Early apoptosis markers</td>
</tr>
<tr>
<td><strong>core_short</strong></td>
<td>101-149bp</td>
<td>Sub-nucleosomal</td>
<td>Specific chromatin states</td>
</tr>
<tr>
<td><strong>mono_nucl</strong></td>
<td>150-220bp</td>
<td>Mono-nucleosomal</td>
<td>Classic <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> (reference)</td>
</tr>
<tr>
<td><strong>di_nucl</strong></td>
<td>221-260bp</td>
<td>Di-nucleosomal</td>
<td>Transitional</td>
</tr>
<tr>
<td><strong>long</strong></td>
<td>261-400bp</td>
<td>Multi-nucleosomal</td>
<td>Necrosis-associated</td>
</tr>
<tr>
<td><strong>ultra_long</strong></td>
<td>401-1000bp</td>
<td>Extended fragments</td>
<td>Necrosis, fetal <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr>, late apoptosis</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Non-overlapping</p>
<p>Each fragment is counted in exactly one channel. This prevents multicollinearity in ML models.</p>
</div>
<hr>
<h2 id="implementation-details">Implementation Details</h2>
<h3 id="counting-pipeline">Counting Pipeline</h3>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart LR
    BED["sample.bed.gz"] --&gt; RUST["Rust Backend"]
    BINS["100kb Bins"] --&gt; RUST
    GC["GC Correction"] --&gt; RUST

    RUST --&gt; FSC["FSC.tsv"]

    subgraph "6 Channels"
        FSC --&gt; US["ultra_short (65-100bp)"]
        FSC --&gt; CS["core_short (101-149bp)"]
        FSC --&gt; MN["mono_nucl (150-220bp)"]
        FSC --&gt; DN["di_nucl (221-260bp)"]
        FSC --&gt; LG["long (261-400bp)"]
        FSC --&gt; UL["ultra_long (401-1000bp)"]
    end

    subgraph "With --pon-model"
        FSC --&gt; PON["PON logâ‚‚ ratios"]
    end

    subgraph "With --target-regions"
        RUST --&gt; FSC_ON["FSC.ontarget.tsv"]
    end
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<h3 id="pythonrust-architecture">Python/Rust Architecture</h3>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart TB
    subgraph "Python (CLI)"
        CLI["fsc.py"] --&gt; UP["unified_processor.py"]
        UP --&gt; ASSETS["AssetManager"]
    end

    subgraph "Rust Backend"
        UP --&gt; RUST["_core.run_unified_pipeline()"]
        RUST --&gt; GC["GC correction"]
        GC --&gt; COUNT["6-channel counting (65-1000bp)"]
    end

    subgraph "Python (Post-processing)"
        COUNT --&gt; PROC["fsc_processor.py"]
        PROC --&gt; AGG["Window aggregation"]
        AGG --&gt; PON["PON log2 ratios"]
        PON --&gt; OUT["FSC.tsv"]
        PON --&gt; GENE["FSC.gene.tsv"]
        GENE --&gt; REG["FSC.regions.tsv"]
        REG --&gt; E1["filter_fsc_to_e1()"]
        E1 --&gt; E1OUT["FSC.regions.e1only.tsv"]
    end
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<h3 id="aggregation-strategy">Aggregation Strategy</h3>
<div class="admonition warning">
<p class="admonition-title">Critical</p>
<p>Aggregation should match your analysis goal.</p>
</div>
<table>
<thead>
<tr>
<th>Data Type</th>
<th>Bin Input</th>
<th>Aggregation</th>
<th>Use Case</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>WGS</strong></td>
<td>100kb genome tiles</td>
<td>50 bins â†’ <strong>5Mb</strong></td>
<td>Arm-level CNV</td>
</tr>
<tr>
<td><strong>WGS focal</strong></td>
<td>100kb genome tiles</td>
<td><strong>No aggregation</strong></td>
<td>Focal amps (EGFR, MYC)</td>
</tr>
<tr>
<td><strong>Panel</strong></td>
<td>Exon/Gene targets</td>
<td><strong>No aggregation</strong></td>
<td>Gene-level resolution</td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">Auto-detection</p>
<p>When <code>--target-regions</code> is provided in <code>run-all</code>, aggregation is <strong>automatically disabled</strong> to preserve gene-level resolution for panel data.</p>
</div>
<p><strong>Why this matters:</strong>
- 5Mb aggregation is great for detecting <strong>arm-level</strong> events (e.g., 1p/19q co-deletion)
- Focal amplifications (e.g., EGFR amp &lt;1Mb) are <strong>washed out</strong> by 5Mb aggregation
- Panel targets are already gene-resolutionâ€”aggregation destroys their value</p>
<p><strong>Recommendation:</strong>
- For <strong>broad CNV detection</strong> (tumor fraction, aneuploidy): Use aggregated 5Mb windows
- For <strong>focal analysis</strong> (driver genes, amplicons): Preserve raw bin resolution</p>
<h3 id="gc-correction-details"><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction Details</h3>
<p>Each fragment receives a <strong>weight</strong> based on its (length, <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>%) bin:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>weight = correction_factors[(length, gc_pct)]
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>channel_count += weight  # Instead of += 1
</code></pre></div>
<p>This removes <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias <strong>before</strong> ML, not after. The correction factors come from:
- <strong>WGS</strong>: Computed from all genome-wide fragments
- <strong>Panel</strong>: Computed from <strong>off-target reads only</strong> (when <code>--target-regions</code> provided)</p>
<h3 id="bin-assignment">Bin Assignment</h3>
<p>Fragments are assigned to bins based on <strong>overlap</strong> (not midpoint):</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="c1">// If fragment overlaps bin, count it</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="n">tree</span><span class="p">.</span><span class="n">query</span><span class="p">(</span><span class="n">start</span><span class="p">,</span><span class="w"> </span><span class="n">end</span><span class="p">,</span><span class="w"> </span><span class="o">|</span><span class="n">bin</span><span class="o">|</span><span class="w"> </span><span class="p">{</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="w">    </span><span class="n">bin</span><span class="p">.</span><span class="n">channel_count</span><span class="w"> </span><span class="o">+=</span><span class="w"> </span><span class="n">weight</span><span class="p">;</span>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="p">});</span>
</code></pre></div>
<p>A fragment spanning two bins counts in <strong>both</strong>. This is intentional for coverage metrics.</p>
<hr>
<h2 id="usage">Usage</h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a><span class="c1"># Basic (auto-loads bundled 100kb bins)</span>
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>krewlyzer<span class="w"> </span>fsc<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span>--genome<span class="w"> </span>hg38
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a>
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a><span class="c1"># With PoN for log2 ratios</span>
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>krewlyzer<span class="w"> </span>fsc<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span>--pon-model<span class="w"> </span>cohort.pon
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a>
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a><span class="c1"># Custom bin size</span>
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a>krewlyzer<span class="w"> </span>fsc<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span>--bin-input<span class="w"> </span>custom_bins.bed
</code></pre></div>
<h2 id="cli-options">CLI Options</h2>
<table>
<thead>
<tr>
<th>Option</th>
<th>Short</th>
<th>Type</th>
<th>Default</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--input</code></td>
<td><code>-i</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>Input .bed.gz file (output from extract)</td>
</tr>
<tr>
<td><code>--output</code></td>
<td><code>-o</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>Output directory</td>
</tr>
<tr>
<td><code>--sample-name</code></td>
<td><code>-s</code></td>
<td>TEXT</td>
<td></td>
<td>Override sample name</td>
</tr>
<tr>
<td><code>--bin-input</code></td>
<td><code>-b</code></td>
<td>PATH</td>
<td></td>
<td>Custom bin file (default: 100kb genome-wide)</td>
</tr>
<tr>
<td><code>--pon-model</code></td>
<td><code>-P</code></td>
<td>PATH</td>
<td></td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> model for hybrid <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction</td>
</tr>
<tr>
<td><code>--pon-variant</code></td>
<td></td>
<td>TEXT</td>
<td>all_unique</td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> variant: <code>all_unique</code> or <code>duplex</code></td>
</tr>
<tr>
<td><code>--skip-pon</code></td>
<td></td>
<td>FLAG</td>
<td></td>
<td>Skip <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> z-score normalization</td>
</tr>
<tr>
<td><code>--target-regions</code></td>
<td><code>-T</code></td>
<td>PATH</td>
<td></td>
<td>Target <abbr title="Browser Extensible Data - Genomic interval file format">BED</abbr> (for on/off-target <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> split)</td>
</tr>
<tr>
<td><code>--skip-target-regions</code></td>
<td></td>
<td>FLAG</td>
<td></td>
<td>Force WGS mode (ignore bundled targets)</td>
</tr>
<tr>
<td><code>--assay</code></td>
<td><code>-A</code></td>
<td>TEXT</td>
<td></td>
<td>Assay type (xs1/xs2) for gene-centric <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
</tr>
<tr>
<td><code>--windows</code></td>
<td><code>-w</code></td>
<td>INT</td>
<td>100000</td>
<td>Window size for aggregation</td>
</tr>
<tr>
<td><code>--continue-n</code></td>
<td><code>-c</code></td>
<td>INT</td>
<td>50</td>
<td>Consecutive window number</td>
</tr>
<tr>
<td><code>--genome</code></td>
<td><code>-G</code></td>
<td>TEXT</td>
<td>hg19</td>
<td>Genome build (hg19/hg38)</td>
</tr>
<tr>
<td><code>--gc-correct</code></td>
<td></td>
<td>FLAG</td>
<td>True</td>
<td>Apply <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> bias correction</td>
</tr>
<tr>
<td><code>--threads</code></td>
<td><code>-t</code></td>
<td>INT</td>
<td>0</td>
<td>Number of threads (0=all cores)</td>
</tr>
<tr>
<td><code>--verbose</code></td>
<td><code>-v</code></td>
<td>FLAG</td>
<td></td>
<td>Enable verbose logging</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="output-format">Output Format</h2>
<p>Output: <code>{sample}.FSC.tsv</code></p>
<h3 id="base-columns-always-present">Base Columns (always present)</h3>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>chrom</code></td>
<td>str</td>
<td>Chromosome</td>
</tr>
<tr>
<td><code>start</code></td>
<td>int</td>
<td>Window start (0-based)</td>
</tr>
<tr>
<td><code>end</code></td>
<td>int</td>
<td>Window end</td>
</tr>
<tr>
<td><code>ultra_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (65â€“100 bp)</td>
</tr>
<tr>
<td><code>core_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (101â€“149 bp)</td>
</tr>
<tr>
<td><code>mono_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (150â€“220 bp)</td>
</tr>
<tr>
<td><code>di_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (221â€“260 bp)</td>
</tr>
<tr>
<td><code>long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (261â€“400 bp)</td>
</tr>
<tr>
<td><code>ultra_long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (401â€“1000 bp)</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted total (65â€“1000 bp)</td>
</tr>
<tr>
<td><code>mean_gc</code></td>
<td>float</td>
<td>Mean <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> fraction of fragments in this window</td>
</tr>
</tbody>
</table>
<h3 id="pon-columns-when-pon-model-provided">PoN Columns (when <code>--pon-model</code> provided)</h3>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>*_log2</code></td>
<td>float</td>
<td>log2(channel / PoN_mean)</td>
</tr>
<tr>
<td><code>*_reliability</code></td>
<td>float</td>
<td>1 / (PoN_variance + k)</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Log2 ratios are signed: positive = above PoN mean, negative = below.</p>
</div>
<hr>
<h2 id="wgs-vs-panel-data-msk-access">WGS vs Panel Data (MSK-ACCESS)</h2>
<table>
<thead>
<tr>
<th>Aspect</th>
<th>WGS</th>
<th>Panel (MSK-ACCESS)</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Coverage</strong></td>
<td>Uniform genome-wide</td>
<td>High on-target, sparse off-target</td>
</tr>
<tr>
<td><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction source</strong></td>
<td>All fragments</td>
<td><strong>Off-target only</strong></td>
</tr>
<tr>
<td><strong>Typical depth</strong></td>
<td>~30x genome</td>
<td>~1000x on-target</td>
</tr>
<tr>
<td><strong>Best bins</strong></td>
<td>All bins reliable</td>
<td>On-target bins only</td>
</tr>
</tbody>
</table>
<h3 id="panel-mode-details">Panel Mode Details</h3>
<p>When you provide <code>--target-regions</code> to <code>run-all</code>:</p>
<ol>
<li><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model training</strong>: Uses only off-target reads (unbiased by capture)</li>
<li><strong>Counting</strong>: All reads are counted (on-target + off-target)</li>
<li><strong>Interpretation</strong>: On-target bins have high counts, off-target bins are sparse</li>
</ol>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>sample.bam<span class="w"> </span>-g<span class="w"> </span>ref.fa<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a><span class="w">    </span>--target-regions<span class="w"> </span>msk_access_baits.bed
</code></pre></div>
<h3 id="panel-recommendations">Panel Recommendations</h3>
<table>
<thead>
<tr>
<th>Feature</th>
<th>Recommendation</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>Which bins to use</strong></td>
<td>Filter to high-coverage bins (&gt;100 fragments)</td>
</tr>
<tr>
<td><strong>Channel ratios</strong></td>
<td>More robust than absolute counts</td>
</tr>
<tr>
<td><strong>PoN</strong></td>
<td>Build from same panel type only</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="ml-feature-engineering">ML Feature Engineering</h2>
<h3 id="raw-features-per-5mb-window">Raw Features (per 5Mb window)</h3>
<ul>
<li>5 channel counts (ultra_short, core_short, mono_nucl, di_nucl, long)</li>
<li>1 total count</li>
</ul>
<h3 id="derived-features-recommended">Derived Features (recommended)</h3>
<ul>
<li><strong>Channel ratios</strong>: <code>short / long</code>, <code>mono_nucl / total</code></li>
<li><strong>Log2 ratios vs PoN</strong>: Tumor-specific deviations</li>
<li><strong>Reliability-weighted</strong>: Use reliability scores in loss functions</li>
</ul>
<h3 id="example-short-to-long-ratio">Example: Short-to-Long Ratio</h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a><span class="n">df</span><span class="p">[</span><span class="s1">'short_long_ratio'</span><span class="p">]</span> <span class="o">=</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">'ultra_short'</span><span class="p">]</span> <span class="o">+</span> <span class="n">df</span><span class="p">[</span><span class="s1">'core_short'</span><span class="p">])</span> <span class="o">/</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">'long'</span><span class="p">]</span> <span class="o">+</span> <span class="mf">1e-9</span><span class="p">)</span>
</code></pre></div>
<p>Higher ratio = more short fragments = potential tumor signal</p>
<hr>
<h2 id="normalization-order">Normalization Order</h2>
<ol>
<li><strong><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighting</strong> (Rust): Raw counts Ã— correction factor per (length, <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>) bin</li>
<li><strong>Window aggregation</strong> (Python): 50 bins â†’ 5Mb windows</li>
<li><strong>PoN log-ratio</strong> (Python): log2(sample / PoN mean) when PoN model provided</li>
</ol>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction is applied <strong>first</strong> in Rust, not after. This ensures all downstream features are <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-unbiased.</p>
</div>
<hr>
<h2 id="panel-mode-target-regions">Panel Mode (--target-regions)</h2>
<p>For targeted sequencing panels (MSK-ACCESS), use <code>--target-regions</code> to generate <strong>separate on/off-target outputs</strong>:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>krewlyzer<span class="w"> </span>fsc<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a><span class="w">    </span>--target-regions<span class="w"> </span>MSK-ACCESS_targets.bed<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a><span class="w">    </span>--bin-input<span class="w"> </span>gene_level_bins.bed
</code></pre></div>
<h3 id="processing-with-target-regions">Processing with Target Regions</h3>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart TB
    BED["sample.bed.gz"] --&gt; SPLIT{"Fragment Location"}

    SPLIT --&gt;|"Overlaps target"| ON["On-Target"]
    SPLIT --&gt;|"Does not overlap"| OFF["Off-Target"]

    ON --&gt; FSC_ON["FSC.ontarget.tsv"]
    OFF --&gt; FSC_OFF["FSC.tsv"]
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<h3 id="output-files">Output Files</h3>
<table>
<thead>
<tr>
<th>File</th>
<th>Contents</th>
<th>Use Case</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>{sample}.FSC.tsv</code></td>
<td><strong>Off-target</strong> fragments</td>
<td>Unbiased global signal (primary)</td>
</tr>
<tr>
<td><code>{sample}.FSC.ontarget.tsv</code></td>
<td><strong>On-target</strong> fragments</td>
<td>Gene-level local signal</td>
</tr>
</tbody>
</table>
<div class="admonition important">
<p class="admonition-title">Important</p>
<p><strong>Off-target = unbiased</strong> â€“ preferred for fragmentomics biomarkers.<br>
<strong>On-target = capture-biased</strong> â€“ reflects library prep + target selection.</p>
</div>
<h3 id="when-to-use-on-target-fsc">When to Use On-Target <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></h3>
<table>
<thead>
<tr>
<th>Use Case</th>
<th>Recommended</th>
</tr>
</thead>
<tbody>
<tr>
<td>CNV detection</td>
<td>Off-target</td>
</tr>
<tr>
<td>Tumor fraction</td>
<td>Off-target</td>
</tr>
<tr>
<td>Gene-level amplification</td>
<td><strong>On-target</strong></td>
</tr>
<tr>
<td>Panel-specific features</td>
<td>Both</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="gene-centric-fsc-msk-access">Gene-Centric <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> (MSK-ACCESS)</h2>
<p>For MSK-ACCESS panels, use <code>--assay</code> to aggregate fragment counts by <strong>gene</strong> instead of genomic windows:</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>krewlyzer<span class="w"> </span>fsc<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span>--assay<span class="w"> </span>xs2
</code></pre></div>
<h3 id="output-files_1">Output Files</h3>
<table>
<thead>
<tr>
<th>File</th>
<th>Description</th>
<th>Rows</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>{sample}.FSC.tsv</code></td>
<td>Standard window-based <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
<td>~28,000</td>
</tr>
<tr>
<td><code>{sample}.FSC.gene.tsv</code></td>
<td>Gene-level <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
<td>146 (xs2)</td>
</tr>
<tr>
<td><code>{sample}.FSC.regions.tsv</code></td>
<td>Per-exon/target <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></td>
<td>~1,000</td>
</tr>
</tbody>
</table>
<h3 id="gene-fsc-output-format-samplefscgenetsv">Gene <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> Output Format (<code>{sample}.FSC.gene.tsv</code>)</h3>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>gene</code></td>
<td>str</td>
<td>HGNC gene symbol</td>
</tr>
<tr>
<td><code>n_regions</code></td>
<td>int</td>
<td>Number of exons/targets for this gene</td>
</tr>
<tr>
<td><code>total_bp</code></td>
<td>int</td>
<td>Total covered base pairs</td>
</tr>
<tr>
<td><code>ultra_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (65â€“100 bp)</td>
</tr>
<tr>
<td><code>core_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (101â€“149 bp)</td>
</tr>
<tr>
<td><code>mono_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (150â€“220 bp)</td>
</tr>
<tr>
<td><code>di_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (221â€“260 bp)</td>
</tr>
<tr>
<td><code>long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (261â€“400 bp)</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted total count</td>
</tr>
<tr>
<td><code>ultra_short_ratio</code></td>
<td>float</td>
<td><code>ultra_short / total</code></td>
</tr>
<tr>
<td><code>core_short_ratio</code></td>
<td>float</td>
<td><code>core_short / total</code></td>
</tr>
<tr>
<td><code>mono_nucl_ratio</code></td>
<td>float</td>
<td><code>mono_nucl / total</code></td>
</tr>
<tr>
<td><code>di_nucl_ratio</code></td>
<td>float</td>
<td><code>di_nucl / total</code></td>
</tr>
<tr>
<td><code>long_ratio</code></td>
<td>float</td>
<td><code>long / total</code></td>
</tr>
<tr>
<td><code>normalized_depth</code></td>
<td>float</td>
<td>RPKM-like: <code>(total Ã— 10â¹) / (total_bp Ã— total_frags)</code></td>
</tr>
</tbody>
</table>
<h3 id="region-fsc-output-format-samplefscregionstsv">Region <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> Output Format (<code>{sample}.FSC.regions.tsv</code>)</h3>
<p>Per-exon/target output for fine-grained copy number analysis:</p>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>chrom</code></td>
<td>str</td>
<td>Chromosome</td>
</tr>
<tr>
<td><code>start</code></td>
<td>int</td>
<td>Region start (0-based)</td>
</tr>
<tr>
<td><code>end</code></td>
<td>int</td>
<td>Region end</td>
</tr>
<tr>
<td><code>gene</code></td>
<td>str</td>
<td>Gene symbol</td>
</tr>
<tr>
<td><code>region_name</code></td>
<td>str</td>
<td>Exon/target identifier</td>
</tr>
<tr>
<td><code>region_bp</code></td>
<td>int</td>
<td>Region length in bp</td>
</tr>
<tr>
<td><code>ultra_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (65â€“100 bp)</td>
</tr>
<tr>
<td><code>core_short</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (101â€“149 bp)</td>
</tr>
<tr>
<td><code>mono_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (150â€“220 bp)</td>
</tr>
<tr>
<td><code>di_nucl</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (221â€“260 bp)</td>
</tr>
<tr>
<td><code>long</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted count (261â€“400 bp)</td>
</tr>
<tr>
<td><code>total</code></td>
<td>float</td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr>-weighted total count</td>
</tr>
<tr>
<td><code>ultra_short_ratio</code></td>
<td>float</td>
<td><code>ultra_short / total</code></td>
</tr>
<tr>
<td><code>core_short_ratio</code></td>
<td>float</td>
<td><code>core_short / total</code></td>
</tr>
<tr>
<td><code>mono_nucl_ratio</code></td>
<td>float</td>
<td><code>mono_nucl / total</code></td>
</tr>
<tr>
<td><code>di_nucl_ratio</code></td>
<td>float</td>
<td><code>di_nucl / total</code></td>
</tr>
<tr>
<td><code>long_ratio</code></td>
<td>float</td>
<td><code>long / total</code></td>
</tr>
<tr>
<td><code>normalized_depth</code></td>
<td>float</td>
<td>RPKM-like depth</td>
</tr>
</tbody>
</table>
<h3 id="e1-only-fsc-output">E1-Only <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> Output</h3>
<p><strong>File</strong>: <code>{sample}.FSC.regions.e1only.tsv</code></p>
<p>E1 (first exon) filtering extracts only the first exon per gene by genomic position. Per Helzer et al. (2025), promoter-proximal regions (E1) are Nucleosome Depleted Regions (NDRs) with distinct fragmentation patterns, often showing stronger cancer signal than whole-gene averages.</p>
<p><strong>Usage</strong>:
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="c1"># Default: E1-only file generated automatically with --assay</span>
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-r<span class="w"> </span>ref.fa<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span>-A<span class="w"> </span>xs2
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a>
<a href="#__codelineno-7-4" id="__codelineno-7-4" name="__codelineno-7-4"></a><span class="c1"># Disable E1-only generation</span>
<a href="#__codelineno-7-5" id="__codelineno-7-5" name="__codelineno-7-5"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-r<span class="w"> </span>ref.fa<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span>-A<span class="w"> </span>xs2<span class="w"> </span>--disable-e1-aggregation
</code></pre></div><p></p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>E1-only <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> is particularly useful for <strong>early cancer detection</strong> where promoter fragmentation changes are an early marker.</p>
</div>
<h3 id="normalized-depth-rpkm-like">Normalized Depth (RPKM-like)</h3>
<p>Both gene and region outputs include <code>normalized_depth</code>:</p>
<div class="arithmatex">\[
\text{normalized_depth} = \frac{\text{count} \times 10^9}{\text{region_bp} \times \text{total_fragments}}
\]</div>
<p>This enables cross-sample depth comparisons independent of library size and region size.</p>
<h3 id="supported-assays">Supported Assays</h3>
<table>
<thead>
<tr>
<th>Assay</th>
<th>Flag</th>
<th style="text-align: center;">Genes</th>
</tr>
</thead>
<tbody>
<tr>
<td>MSK-ACCESS v1</td>
<td><code>--assay xs1</code></td>
<td style="text-align: center;">128</td>
</tr>
<tr>
<td>MSK-ACCESS v2</td>
<td><code>--assay xs2</code></td>
<td style="text-align: center;">146</td>
</tr>
</tbody>
</table>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>Gene-level <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> is useful for <strong>gene-specific amplification</strong> detection and <strong>integration with variant calling</strong> pipelines.</p>
</div>
<h3 id="gc-correction-for-gene-fsc"><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction for Gene <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr></h3>
<p>In panel mode, gene-level <abbr title="Fragment Size Coverage - Coverage depth by fragment size across genomic bins">FSC</abbr> uses <strong>on-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction factors</strong> (<code>.correction_factors.ontarget.tsv</code>) for accurate copy number estimates.</p>
<p><strong>Why this matters:</strong>
- Different genes have different <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> content
- High-<abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> genes (e.g., <em>EGFR</em>) capture with different efficiency than low-<abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> genes
- Without correction, high-<abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> genes appear <strong>falsely deleted</strong>, low-<abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> genes appear <strong>amplified</strong></p>
<p><strong>How it works:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a><span class="c1"># Instead of raw counting (+= 1):</span>
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a><span class="n">weight</span> <span class="o">=</span> <span class="n">correction_factors</span><span class="p">[(</span><span class="n">len_bin</span><span class="p">,</span> <span class="n">gc_pct</span><span class="p">)]</span>
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a><span class="n">gene_count</span> <span class="o">+=</span> <span class="n">weight</span>  <span class="c1"># GC-corrected counting</span>
</code></pre></div><p></p>
<p><strong>Log output example:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>Aggregating FSC by gene: 146 genes, GC correction: ON (weighted counting)
<a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a>Processed 2.4M fragments, 2.4M assigned to genes
<a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a>  GC correction: avg_weight=1.898, missing_gc=0
</code></pre></div><p></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>On-target factors are automatically used when available. If not found, raw counting is used with a debug log message.</p>
</div>
<hr>
<h2 id="see-also">See Also</h2>
<ul>
<li><strong>Citation</strong>: <a href="../../../resources/citation/#fsc">Cristiano et al. (2019)</a> - DELFI fragmentomics paper</li>
<li><strong><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></strong>: <a href="../../../reference/pon-models/">Z-Score Normalization</a> - Log2 ratio computation</li>
<li><strong>Inputs</strong>: <a href="../../../reference/input-formats/#bins">File Formats</a> - Bin file format</li>
<li><strong>Related</strong>: <a href="../fsr/"><abbr title="Fragment Size Ratio - Ratio of short to long fragments">FSR</abbr></a> (ratios), <a href="../fsd/"><abbr title="Fragment Size Distribution - Distribution of fragment lengths across chromosome arms">FSD</abbr></a> (distribution), <a href="../extract/">Extract</a> (<abbr title="Browser Extensible Data - Genomic interval file format">BED</abbr>.gz source)</li>
<li><strong>Guides</strong>: <a href="../../../guides/panel-mode/">Panel Mode</a>, <a href="../../../guides/gc-correction/"><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction</a></li>
<li><strong>CLI</strong>: <a href="../../../cli/run-all/"><code>run-all</code></a> - Unified pipeline</li>
<li><strong>Nextflow</strong>: <a href="../../../nextflow/parameters/">Parameters</a> - Batch processing</li>
</ul>
<h2 id="references">References</h2>
<div class="admonition quote">
<p class="admonition-title">References</p>
<p>Snyder et al. (2016). Cell-free DNA comprises an in vivo nucleosome footprint that informs its tissues-of-origin. <em>Cell</em>, 164(1-2), 57-68.</p>
<p>Cristiano et al. (2019). Genome-wide cell-free DNA fragmentation in patients with cancer. <em>Nature</em>, 570(7761), 385-389.</p>
</div>
<!-- Abbreviation definitions for Krewlyzer documentation -->
<!-- These provide hover tooltips for technical acronyms -->
<aside class="md-source-file">
<span class="md-source-file__fact">
<span class="md-icon" title="Last update">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
</span>
<span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="March 1, 2026 03:15:26 UTC"><span class="timeago" datetime="2026-03-01T03:15:26+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="March 1, 2026 03:15:26 UTC">2026-03-01</span>
</span>
<span class="md-source-file__fact">
<span class="md-icon" title="Created">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
</span>
<span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="November 26, 2025 16:12:36 UTC"><span class="timeago" datetime="2025-11-26T16:12:36+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="November 26, 2025 16:12:36 UTC">2025-11-26</span>
</span>
</aside>
</article>
</div>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
<div class="md-copyright__highlight">
      Developed by <a href="https://github.com/rhshah">Ronak H Shah</a> @ 
<a href="https://www.mskcc.org">MSKCC</a> | 
Built with <a href="https://antigravity.google/">Antigravity</a> | 
<a href="https://github.com/msk-access/krewlyzer/blob/main/LICENSE">AGPL-3.0</a>
</div>
</div>
<div class="md-social">
<a class="md-social__link" href="https://github.com/msk-access/krewlyzer" rel="noopener" target="_blank" title="github.com">
<svg viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M173.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6m-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3m44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9M252.8 8C114.1 8 8 113.3 8 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C436.2 457.8 504 362.9 504 252 504 113.3 391.5 8 252.8 8M105.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1m-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7m32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1m-11.4-14.7c-1.6 1-1.6 3.6 0 5.9s4.3 3.3 5.6 2.3c1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2"></path></svg>
</a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"annotate": null, "base": "../../..", "features": ["navigation.instant", "navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.indexes", "navigation.path", "toc.follow", "content.code.copy", "content.code.annotate"], "search": "../../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": {"default": "stable", "provider": "mike"}}</script>
<script src="../../../assets/javascripts/bundle.79ae519e.min.js"></script>
<script src="../../../js/timeago.min.js"></script>
<script src="../../../js/timeago_mkdocs_material.js"></script>
<script src="../../../javascripts/mathjax.js"></script>
<script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
<script src="../../../assets/javascripts/panzoom.min.js"></script>
<script src="../../../assets/javascripts/zoompan.js"></script>
<script type="module">import mermaid from "https://unpkg.com/mermaid@11.4.0/dist/mermaid.esm.min.mjs";
window.mermaidConfig = {default: {
    startOnLoad: false
}};</script>
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>