<!DOCTYPE html><html class="no-js" lang="en"><head>
<meta charset="utf-8">
<meta content="width=device-width,initial-scale=1" name="viewport">
<link href="https://msk-access.github.io/krewlyzer/features/regulatory/region-entropy/" rel="canonical">
<link href="../ocf/" rel="prev">
<link href="../region-mds/" rel="next">
<link href="../../../assets/images/favicon.png" rel="icon">
<meta content="mkdocs-1.6.1, mkdocs-material-9.7.1" name="generator">
<title>Region Entropy - Krewlyzer</title>
<link href="../../../assets/stylesheets/main.484c7ddc.min.css" rel="stylesheet">
<link href="../../../assets/stylesheets/palette.ab4e12ef.min.css" rel="stylesheet">
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect">
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet">
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<link href="../../../css/timeago.css" rel="stylesheet">
<link href="../../../stylesheets/extra.css" rel="stylesheet">
<link href="../../../assets/stylesheets/panzoom.css" rel="stylesheet">
<script>__md_scope=new URL("../../..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<meta content="website" property="og:type">
<meta content="Region Entropy - Krewlyzer" property="og:title">
<meta content="https://msk-access.github.io/krewlyzer/assets/images/social/features/regulatory/region-entropy.png" property="og:image">
<meta content="image/png" property="og:image:type">
<meta content="1200" property="og:image:width">
<meta content="630" property="og:image:height">
<meta content="https://msk-access.github.io/krewlyzer/features/regulatory/region-entropy/" property="og:url">
<meta content="summary_large_image" property="twitter:card">
<meta content="Region Entropy - Krewlyzer" property="twitter:title">
<meta content="https://msk-access.github.io/krewlyzer/assets/images/social/features/regulatory/region-entropy.png" property="twitter:image">
<link href="../../../assets/stylesheets/glightbox.min.css" rel="stylesheet"><script src="../../../assets/javascripts/glightbox.min.js"></script><style id="glightbox-style">
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color); }
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color); }
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color); }
        </style><meta name="panzoom-theme" content="material">
<meta name="panzoom-data" content='{"selectors": [".mermaid", ".d2"], "initial_zoom_level": 1.0}'> </head>
<body data-md-color-accent="custom" data-md-color-primary="custom" data-md-color-scheme="default" dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox">
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox">
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#region-entropy-tfbsatac-size-entropy">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Krewlyzer" class="md-header__button md-logo" data-md-component="logo" href="../../.." title="Krewlyzer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Krewlyzer
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Region Entropy
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="custom" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="custom" data-md-color-scheme="default" id="__palette_0" name="__palette" type="radio">
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
</label>
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="custom" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="custom" data-md-color-scheme="slate" id="__palette_1" name="__palette" type="radio">
<label class="md-header__button md-icon" for="__palette_0" hidden="" title="Switch to light mode">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12z"></path></svg>
</label>
</form>
<script>var palette=__md_get("__palette");if(palette&&palette.color){if("(prefers-color-scheme)"===palette.color.media){var media=matchMedia("(prefers-color-scheme: light)"),input=document.querySelector(media.matches?"[data-md-color-media='(prefers-color-scheme: light)']":"[data-md-color-media='(prefers-color-scheme: dark)']");palette.color.media=input.getAttribute("data-md-color-media"),palette.color.scheme=input.getAttribute("data-md-color-scheme"),palette.color.primary=input.getAttribute("data-md-color-primary"),palette.color.accent=input.getAttribute("data-md-color-accent")}for(var[key,value]of Object.entries(palette.color))document.body.setAttribute("data-md-color-"+key,value)}</script>
<label class="md-header__button md-icon" for="__search">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text">
<label class="md-search__icon md-icon" for="__search">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/msk-access/krewlyzer" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Krewlyzer" class="md-nav__button md-logo" data-md-component="logo" href="../../.." title="Krewlyzer">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54"></path></svg>
</a>
    Krewlyzer
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/msk-access/krewlyzer" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    msk-access/krewlyzer
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../..">
<span class="md-ellipsis">
    
  
    Home
  

    
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_2" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../getting-started/">
<span class="md-ellipsis">
    
  
    Getting Started
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
            
  
    Getting Started
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/installation/">
<span class="md-ellipsis">
    
  
    Installation
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/quickstart/">
<span class="md-ellipsis">
    
  
    Quickstart
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../getting-started/concepts/">
<span class="md-ellipsis">
    
  
    Concepts
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_3" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../cli/">
<span class="md-ellipsis">
    
  
    CLI Reference
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
            
  
    CLI Reference
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../cli/run-all/">
<span class="md-ellipsis">
    
  
    run-all
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_4" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../../nextflow/">
<span class="md-ellipsis">
    
  
    Nextflow Pipeline
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_4" id="__nav_4_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="false" aria-labelledby="__nav_4_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_4">
<span class="md-nav__icon md-icon"></span>
            
  
    Nextflow Pipeline
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/samplesheet/">
<span class="md-ellipsis">
    
  
    Samplesheet
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/parameters/">
<span class="md-ellipsis">
    
  
    Parameters
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/outputs/">
<span class="md-ellipsis">
    
  
    Outputs
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../nextflow/examples/">
<span class="md-ellipsis">
    
  
    Examples
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5" type="checkbox">
<div class="md-nav__link md-nav__container">
<a class="md-nav__link" href="../../">
<span class="md-ellipsis">
    
  
    Features
  

    
  </span>
</a>
<label class="md-nav__link" for="__nav_5" id="__nav_5_label" tabindex="">
<span class="md-nav__icon md-icon"></span>
</label>
</div>
<nav aria-expanded="true" aria-labelledby="__nav_5_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_5">
<span class="md-nav__icon md-icon"></span>
            
  
    Features
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_2" type="checkbox">
<label class="md-nav__link" for="__nav_5_2" id="__nav_5_2_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Core
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_2_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_2">
<span class="md-nav__icon md-icon"></span>
            
  
    Core
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/extract/">
<span class="md-ellipsis">
    
  
    Fragment Extraction
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/fsc/">
<span class="md-ellipsis">
    
  
    FSC
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/fsd/">
<span class="md-ellipsis">
    
  
    FSD
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/fsr/">
<span class="md-ellipsis">
    
  
    FSR
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../core/wps/">
<span class="md-ellipsis">
    
  
    WPS
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--active md-nav__item--nested">
<input checked="" class="md-nav__toggle md-toggle" id="__nav_5_3" type="checkbox">
<label class="md-nav__link" for="__nav_5_3" id="__nav_5_3_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Regulatory
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="true" aria-labelledby="__nav_5_3_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_3">
<span class="md-nav__icon md-icon"></span>
            
  
    Regulatory
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../motif/">
<span class="md-ellipsis">
    
  
    End Motif
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../ocf/">
<span class="md-ellipsis">
    
  
    OCF
  

    
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox">
<label class="md-nav__link md-nav__link--active" for="__toc">
<span class="md-ellipsis">
    
  
    Region Entropy
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
<span class="md-ellipsis">
    
  
    Region Entropy
  

    
  </span>
</a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#purpose">
<span class="md-ellipsis">
      
        Purpose
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#scientific-background">
<span class="md-ellipsis">
      
        Scientific Background
      
    </span>
</a>
<nav aria-label="Scientific Background" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#from-helzer-et-al-2025">
<span class="md-ellipsis">
      
        From Helzer et al. 2025
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-mechanism">
<span class="md-ellipsis">
      
        Biological Mechanism
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#processing-flowchart">
<span class="md-ellipsis">
      
        Processing Flowchart
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#shannon-entropy-formula">
<span class="md-ellipsis">
      
        Shannon Entropy Formula
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usage">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cli-options">
<span class="md-ellipsis">
      
        CLI Options
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#assets">
<span class="md-ellipsis">
      
        Assets
      
    </span>
</a>
<nav aria-label="Assets" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#tfbs-regions">
<span class="md-ellipsis">
      
        TFBS Regions
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#atac-regions">
<span class="md-ellipsis">
      
        ATAC Regions
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-mode-dual-output">
<span class="md-ellipsis">
      
        Panel Mode (Dual Output)
      
    </span>
</a>
<nav aria-label="Panel Mode (Dual Output)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#usage_1">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-files-panel-mode">
<span class="md-ellipsis">
      
        Output Files (Panel Mode)
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#pon-normalization">
<span class="md-ellipsis">
      
        PON Normalization
      
    </span>
</a>
<nav aria-label="PON Normalization" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#building-pon-with-tfbsatac">
<span class="md-ellipsis">
      
        Building PON with TFBS/ATAC
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#applying-pon">
<span class="md-ellipsis">
      
        Applying PON
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-format">
<span class="md-ellipsis">
      
        Output Format
      
    </span>
</a>
<nav aria-label="Output Format" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#tfbs-output-sampletfbstsv">
<span class="md-ellipsis">
      
        TFBS Output: {sample}.TFBS.tsv
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#atac-output-sampleatactsv">
<span class="md-ellipsis">
      
        ATAC Output: {sample}.ATAC.tsv
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#interpretation">
<span class="md-ellipsis">
      
        Interpretation
      
    </span>
</a>
<nav aria-label="Interpretation" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#entropy-values">
<span class="md-ellipsis">
      
        Entropy Values
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#z-scores">
<span class="md-ellipsis">
      
        Z-Scores
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#algorithm-details">
<span class="md-ellipsis">
      
        Algorithm Details
      
    </span>
</a>
<nav aria-label="Algorithm Details" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#rust-backend-region_entropyrs">
<span class="md-ellipsis">
      
        Rust Backend (region_entropy.rs)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#python-processing-region_entropy_processorpy">
<span class="md-ellipsis">
      
        Python Processing (region_entropy_processor.py)
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#performance">
<span class="md-ellipsis">
      
        Performance
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#citation">
<span class="md-ellipsis">
      
        Citation
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#see-also">
<span class="md-ellipsis">
      
        See Also
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../region-mds/">
<span class="md-ellipsis">
    
  
    Region MDS
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_4" type="checkbox">
<label class="md-nav__link" for="__nav_5_4" id="__nav_5_4_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Variant
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_4_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_4">
<span class="md-nav__icon md-icon"></span>
            
  
    Variant
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../variant/mfsd/">
<span class="md-ellipsis">
    
  
    mFSD
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_5" type="checkbox">
<label class="md-nav__link" for="__nav_5_5" id="__nav_5_5_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Methylation
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_5_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_5">
<span class="md-nav__icon md-icon"></span>
            
  
    Methylation
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../methylation/uxm/">
<span class="md-ellipsis">
    
  
    UXM
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_5_6" type="checkbox">
<label class="md-nav__link" for="__nav_5_6" id="__nav_5_6_label" tabindex="0">
<span class="md-ellipsis">
    
  
    Output
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_5_6_label" class="md-nav" data-md-level="2">
<label class="md-nav__title" for="__nav_5_6">
<span class="md-nav__icon md-icon"></span>
            
  
    Output
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../output/json-output/">
<span class="md-ellipsis">
    
  
    JSON Export
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_6" type="checkbox">
<label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="">
<span class="md-ellipsis">
    
  
    Guides
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_6_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_6">
<span class="md-nav__icon md-icon"></span>
            
  
    Guides
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/panel-mode/">
<span class="md-ellipsis">
    
  
    Panel Mode
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/msk-access-quickstart/">
<span class="md-ellipsis">
    
  
    MSK-ACCESS Quickstart
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/building-pon/">
<span class="md-ellipsis">
    
  
    Building PON
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../guides/gc-correction/">
<span class="md-ellipsis">
    
  
    GC Correction
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_7" type="checkbox">
<label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="">
<span class="md-ellipsis">
    
  
    Reference
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_7_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_7">
<span class="md-nav__icon md-icon"></span>
            
  
    Reference
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/pon-models/">
<span class="md-ellipsis">
    
  
    PON Models
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/input-formats/">
<span class="md-ellipsis">
    
  
    Input Formats
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/architecture/">
<span class="md-ellipsis">
    
  
    Architecture
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../reference/glossary/">
<span class="md-ellipsis">
    
  
    Glossary
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_8" type="checkbox">
<label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="">
<span class="md-ellipsis">
    
  
    Development
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_8_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_8">
<span class="md-nav__icon md-icon"></span>
            
  
    Development
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/developer-guide/">
<span class="md-ellipsis">
    
  
    Developer Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/testing-guide/">
<span class="md-ellipsis">
    
  
    Testing Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/release-guide/">
<span class="md-ellipsis">
    
  
    Release Guide
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/changelog/">
<span class="md-ellipsis">
    
  
    Changelog
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/contributing/">
<span class="md-ellipsis">
    
  
    Contributing
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../development/contributing-data/">
<span class="md-ellipsis">
    
  
    Contributing Data
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--section md-nav__item--nested">
<input class="md-nav__toggle md-toggle md-toggle--indeterminate" id="__nav_9" type="checkbox">
<label class="md-nav__link" for="__nav_9" id="__nav_9_label" tabindex="">
<span class="md-ellipsis">
    
  
    Resources
  

    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_9_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_9">
<span class="md-nav__icon md-icon"></span>
            
  
    Resources
  

          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/overview/">
<span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/citation/">
<span class="md-ellipsis">
    
  
    Citation
  

    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../../../resources/troubleshooting/">
<span class="md-ellipsis">
    
  
    Troubleshooting
  

    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#purpose">
<span class="md-ellipsis">
      
        Purpose
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#scientific-background">
<span class="md-ellipsis">
      
        Scientific Background
      
    </span>
</a>
<nav aria-label="Scientific Background" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#from-helzer-et-al-2025">
<span class="md-ellipsis">
      
        From Helzer et al. 2025
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#biological-mechanism">
<span class="md-ellipsis">
      
        Biological Mechanism
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#processing-flowchart">
<span class="md-ellipsis">
      
        Processing Flowchart
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#shannon-entropy-formula">
<span class="md-ellipsis">
      
        Shannon Entropy Formula
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usage">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#cli-options">
<span class="md-ellipsis">
      
        CLI Options
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#assets">
<span class="md-ellipsis">
      
        Assets
      
    </span>
</a>
<nav aria-label="Assets" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#tfbs-regions">
<span class="md-ellipsis">
      
        TFBS Regions
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#atac-regions">
<span class="md-ellipsis">
      
        ATAC Regions
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#panel-mode-dual-output">
<span class="md-ellipsis">
      
        Panel Mode (Dual Output)
      
    </span>
</a>
<nav aria-label="Panel Mode (Dual Output)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#usage_1">
<span class="md-ellipsis">
      
        Usage
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-files-panel-mode">
<span class="md-ellipsis">
      
        Output Files (Panel Mode)
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#pon-normalization">
<span class="md-ellipsis">
      
        PON Normalization
      
    </span>
</a>
<nav aria-label="PON Normalization" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#building-pon-with-tfbsatac">
<span class="md-ellipsis">
      
        Building PON with TFBS/ATAC
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#applying-pon">
<span class="md-ellipsis">
      
        Applying PON
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#output-format">
<span class="md-ellipsis">
      
        Output Format
      
    </span>
</a>
<nav aria-label="Output Format" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#tfbs-output-sampletfbstsv">
<span class="md-ellipsis">
      
        TFBS Output: {sample}.TFBS.tsv
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#atac-output-sampleatactsv">
<span class="md-ellipsis">
      
        ATAC Output: {sample}.ATAC.tsv
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#interpretation">
<span class="md-ellipsis">
      
        Interpretation
      
    </span>
</a>
<nav aria-label="Interpretation" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#entropy-values">
<span class="md-ellipsis">
      
        Entropy Values
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#z-scores">
<span class="md-ellipsis">
      
        Z-Scores
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#algorithm-details">
<span class="md-ellipsis">
      
        Algorithm Details
      
    </span>
</a>
<nav aria-label="Algorithm Details" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#rust-backend-region_entropyrs">
<span class="md-ellipsis">
      
        Rust Backend (region_entropy.rs)
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#python-processing-region_entropy_processorpy">
<span class="md-ellipsis">
      
        Python Processing (region_entropy_processor.py)
      
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#performance">
<span class="md-ellipsis">
      
        Performance
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#citation">
<span class="md-ellipsis">
      
        Citation
      
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#see-also">
<span class="md-ellipsis">
      
        See Also
      
    </span>
</a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<nav aria-label="Navigation" class="md-path">
<ol class="md-path__list">
<li class="md-path__item">
<a class="md-path__link" href="../../..">
<span class="md-ellipsis">
    Home
  </span>
</a>
</li>
<li class="md-path__item">
<a class="md-path__link" href="../../">
<span class="md-ellipsis">
    Features
  </span>
</a>
</li>
<li class="md-path__item">
<a class="md-path__link" href="../motif/">
<span class="md-ellipsis">
    Regulatory
  </span>
</a>
</li>
</ol>
</nav>
<article class="md-content__inner md-typeset">
<h1 id="region-entropy-tfbsatac-size-entropy">Region Entropy (<abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>/<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> Size Entropy)</h1>
<p><strong>Command</strong>: <code>krewlyzer region-entropy</code></p>
<div class="admonition info">
<p class="admonition-title">Plain English</p>
<p>Region Entropy calculates the diversity of fragment sizes at regulatory regions.
A high entropy value indicates many different fragment sizes; low entropy indicates uniform sizes.</p>
<p><strong>Use case</strong>: Cancer detection and subtyping - tumor <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> shows altered nucleosome positioning at specific regulatory elements.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This feature is based on <strong>Helzer KT, et al. (2025)</strong> "Analysis of <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> fragmentomics metrics and commercial targeted sequencing panels" published in <em>Nature Communications</em>.</p>
</div>
<hr>
<h2 id="purpose">Purpose</h2>
<p>Calculates Shannon entropy of fragment size distributions at:
- <strong><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr></strong>: Transcription Factor Binding Sites (808 factors from GTRD)
- <strong><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr></strong>: Cancer-specific <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr>-seq peaks (23 cancer types from TCGA)</p>
<p>These metrics enable cancer phenotyping from targeted sequencing panels without requiring whole genome sequencing.</p>
<hr>
<h2 id="scientific-background">Scientific Background</h2>
<h3 id="from-helzer-et-al-2025">From Helzer et al. 2025</h3>
<p>Fragmentomics-based analysis of cell-free DNA (<abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr>) has emerged as a method to infer epigenetic and transcriptional data. While many reports analyze whole genome sequencing (WGS), <strong>targeted exon panels can be similarly employed for cancer phenotyping with minimal decrease in performance</strong> despite their smaller genomic coverage.</p>
<p>The study assessed 13 fragmentomics metrics including:
- Fragment length proportions (small fragments, Shannon entropy)
- Normalized fragment read depth
- End motif diversity score (<abbr title="Motif Diversity Score - Entropy/diversity of end motif patterns">MDS</abbr>)
- <strong><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> entropy</strong> - fragments overlapping transcription factor binding sites
- <strong><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> entropy</strong> - fragments overlapping cancer-specific open chromatin regions</p>
<p>Key findings relevant to <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>/<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> entropy:
- Diversity metrics like Shannon entropy measure the <strong>spread of fragment sizes</strong> in a region
- <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> and <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> entropy work well for <strong>cancer detection and subtyping</strong>
- These metrics can be applied to <strong>commercial targeted sequencing panels</strong></p>
<h3 id="biological-mechanism">Biological Mechanism</h3>
<p>Fragment size distributions at regulatory regions reflect <strong>nucleosome positioning</strong>:</p>
<ul>
<li><strong>Nucleosome-bound DNA</strong>: ~147bp core + ~20bp linker = ~167bp</li>
<li><strong>Open chromatin (active TF binding)</strong>: Variable sizes due to transcription factor binding</li>
<li><strong>Tumor alterations</strong>: Aberrant nucleosome positioning → altered size distributions</li>
</ul>
<p>Cancer cells exhibit:
- <strong>Epigenetic dysregulation</strong> → Changed <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> accessibility
- <strong>Altered enhancer usage</strong> → Different <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> peak patterns<br>
- <strong>Tissue-specific signatures</strong> → Cancer type identification</p>
<hr>
<h2 id="processing-flowchart">Processing Flowchart</h2>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart LR
    BED["sample.bed.gz"] --&gt; RUST["Rust Backend<br>(par_iter)"]
    TFBS["TFBS regions"] --&gt; RUST
    ATAC["ATAC regions"] --&gt; RUST

    RUST --&gt; ENTROPY["Entropy Calculation"]

    subgraph "Per Region Label (Parallel)"
        ENTROPY --&gt; COUNT["Fragment count"]
        ENTROPY --&gt; SIZES["Size distribution"]
        SIZES --&gt; SHANNON["Shannon Entropy"]
    end

    SHANNON --&gt; TSV["TFBS.tsv / ATAC.tsv"]
    TSV --&gt; PON["PON Z-score"]
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<div class="admonition tip">
<p class="admonition-title">Performance</p>
<p>Region-level parallelization via Rayon <code>par_iter()</code> enables efficient multi-core processing of <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>/<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> regions.</p>
</div>
<hr>
<h2 id="shannon-entropy-formula">Shannon Entropy Formula</h2>
<div class="arithmatex">\[
H = -\sum_{i} p_i \log_2(p_i)
\]</div>
<p>Where:
- <span class="arithmatex">\(p_i\)</span> = Proportion of fragments with size <span class="arithmatex">\(i\)</span>
- High entropy = Many equally represented sizes (diverse)
- Low entropy = One dominant size (uniform)</p>
<p>As described in Helzer et al.: "Shannon entropy was calculated on the frequency of the fragment lengths... This yielded a single entropy value for each [TF/cancer type] in each sample."</p>
<hr>
<h2 id="usage">Usage</h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1"># Basic usage (computes both TFBS and ATAC)</span>
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>krewlyzer<span class="w"> </span>region-entropy<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output_dir/
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a><span class="c1"># TFBS only</span>
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>krewlyzer<span class="w"> </span>region-entropy<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span>--no-atac
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a>
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a><span class="c1"># ATAC only with PON normalization</span>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a>krewlyzer<span class="w"> </span>region-entropy<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>output/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="w">    </span>--no-tfbs<span class="w"> </span>--pon-model<span class="w"> </span>healthy.pon.parquet
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a>
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a><span class="c1"># Via run-all (automatic when assets available)</span>
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-r<span class="w"> </span>hg19.fa<span class="w"> </span>-o<span class="w"> </span>output/
</code></pre></div>
<hr>
<h2 id="cli-options">CLI Options</h2>
<table>
<thead>
<tr>
<th>Option</th>
<th>Short</th>
<th>Type</th>
<th>Default</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>--input</code></td>
<td><code>-i</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>Input .bed.gz file (from extract)</td>
</tr>
<tr>
<td><code>--output</code></td>
<td><code>-o</code></td>
<td>PATH</td>
<td><em>required</em></td>
<td>Output directory</td>
</tr>
<tr>
<td><code>--sample-name</code></td>
<td><code>-s</code></td>
<td>TEXT</td>
<td></td>
<td>Override sample name</td>
</tr>
<tr>
<td><code>--tfbs/--no-tfbs</code></td>
<td></td>
<td>FLAG</td>
<td><code>--tfbs</code></td>
<td>Enable/disable <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> entropy</td>
</tr>
<tr>
<td><code>--atac/--no-atac</code></td>
<td></td>
<td>FLAG</td>
<td><code>--atac</code></td>
<td>Enable/disable <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> entropy</td>
</tr>
<tr>
<td><code>--tfbs-regions</code></td>
<td></td>
<td>PATH</td>
<td></td>
<td>Custom <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> regions <abbr title="Browser Extensible Data - Genomic interval file format">BED</abbr>.gz</td>
</tr>
<tr>
<td><code>--atac-regions</code></td>
<td></td>
<td>PATH</td>
<td></td>
<td>Custom <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> regions <abbr title="Browser Extensible Data - Genomic interval file format">BED</abbr>.gz</td>
</tr>
<tr>
<td><code>--genome</code></td>
<td><code>-G</code></td>
<td>TEXT</td>
<td>hg19</td>
<td>Genome build (hg19/GRCh37/hg38/GRCh38)</td>
</tr>
<tr>
<td><code>--gc-factors</code></td>
<td><code>-F</code></td>
<td>PATH</td>
<td></td>
<td><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction factors <abbr title="Tab-Separated Values - Plain text tabular data format">TSV</abbr></td>
</tr>
<tr>
<td><code>--pon-model</code></td>
<td><code>-P</code></td>
<td>PATH</td>
<td></td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> model for z-score normalization</td>
</tr>
<tr>
<td><code>--pon-variant</code></td>
<td></td>
<td>TEXT</td>
<td>all_unique</td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> variant: <code>all_unique</code> or <code>duplex</code></td>
</tr>
<tr>
<td><code>--skip-pon</code></td>
<td></td>
<td>FLAG</td>
<td></td>
<td>Skip <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> z-score normalization (for ML negatives)</td>
</tr>
<tr>
<td><code>--target-regions</code></td>
<td><code>-T</code></td>
<td>PATH</td>
<td></td>
<td>Target regions <abbr title="Browser Extensible Data - Genomic interval file format">BED</abbr> (panel mode: generates .ontarget.tsv)</td>
</tr>
<tr>
<td><code>--skip-target-regions</code></td>
<td></td>
<td>FLAG</td>
<td></td>
<td>Force WGS mode (ignore bundled targets from --assay)</td>
</tr>
<tr>
<td><code>--threads</code></td>
<td><code>-t</code></td>
<td>INT</td>
<td>0</td>
<td>Number of threads (0 = all cores)</td>
</tr>
<tr>
<td><code>--verbose</code></td>
<td><code>-v</code></td>
<td>FLAG</td>
<td></td>
<td>Enable verbose logging</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="assets">Assets</h2>
<h3 id="tfbs-regions"><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> Regions</h3>
<p><strong>Source</strong>: <a href="https://gtrd.biouml.org/">Gene Transcription Regulation Database (GTRD)</a> v19.10</p>
<p>As described in Helzer et al.: "A collection of consensus Homo sapiens TFBSs was downloaded from the Gene Transcription Regulation Database (GTRD, v19.10). For each TF, the <strong>top 5000 sites with the greatest amount of experimental support</strong> were used for analysis. TFs with fewer than 5000 sites were discarded, leaving a total of <strong>808 TFs</strong> used for the analysis."</p>
<table>
<thead>
<tr>
<th>Genome</th>
<th>File</th>
<th>TF Count</th>
<th>Sites per TF</th>
</tr>
</thead>
<tbody>
<tr>
<td>GRCh37</td>
<td><code>TFBS.GRCh37.bed.gz</code></td>
<td>808</td>
<td>5,000</td>
</tr>
<tr>
<td>GRCh38</td>
<td><code>TFBS.GRCh38.bed.gz</code></td>
<td>808</td>
<td>5,000</td>
</tr>
</tbody>
</table>
<p><strong>Format:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>chr1  10000  10500  CTCF
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>chr1  15000  15200  FOXA1
</code></pre></div><p></p>
<h3 id="atac-regions"><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> Regions</h3>
<p><strong>Source</strong>: <a href="https://gdc.cancer.gov/about-data/publications/ATACseq-AWG">TCGA <abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr>-seq Pan-Cancer Atlas</a></p>
<p>As described in Helzer et al.: "Consensus genomic regions from Assay for Transposase Accessible Chromatin with sequencing (<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr>-seq) data was downloaded from The Cancer Genome Atlas (TCGA) for <strong>23 different cancer types</strong>."</p>
<table>
<thead>
<tr>
<th>Genome</th>
<th>File</th>
<th>Cancer Types</th>
</tr>
</thead>
<tbody>
<tr>
<td>GRCh37</td>
<td><code>ATAC.GRCh37.bed.gz</code></td>
<td>23</td>
</tr>
<tr>
<td>GRCh38</td>
<td><code>ATAC.GRCh38.bed.gz</code></td>
<td>23</td>
</tr>
</tbody>
</table>
<p><strong>Cancer Types:</strong>
ACC, BLCA, BRCA, CESC, CHOL, COAD, ESCA, GBM, HNSC, KIRC, KIRP, LAML, LGG, LIHC, LUAD, LUSC, MESO, PCPG, PRAD, SKCM, STAD, TGCT, THCA, UCEC</p>
<p><strong>Format:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>chr1  10000  10500  BRCA
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>chr1  15000  15200  LUAD
</code></pre></div><p></p>
<p><strong>Data Source</strong>: Region files are from <a href="https://github.com/Zhao-Lab-UW-DHO/fragmentomics_metrics/">Zhao-Lab-UW-DHO/fragmentomics_metrics</a></p>
<hr>
<h2 id="panel-mode-dual-output">Panel Mode (Dual Output)</h2>
<p>For targeted sequencing panels (like MSK-ACCESS), krewlyzer generates <strong>dual output</strong>:</p>
<ol>
<li><strong>Genome-wide (<code>.tsv</code>)</strong>: All fragments across all <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>/<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> regions → WGS-comparable baseline</li>
<li><strong>Panel-specific (<code>.ontarget.tsv</code>)</strong>: Uses pre-intersected panel regions → panel-specific signal</li>
</ol>
<div class="panzoom-box" data-key="none" id="panzoom0" oncontextmenu="return false;">
<nav class="panzoom-nav">
<button class="panzoom-info panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M464 256A208 208 0 1 0 48 256a208 208 0 1 0 416 0zM0 256a256 256 0 1 1 512 0A256 256 0 1 1 0 256zm169.8-90.7c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"></path>
</svg>
</button><button class="panzoom-reset panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 512 512" xmlns="http://www.w3.org/2000/svg">
<path d="M125.7 160l50.3 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L48 224c-17.7 0-32-14.3-32-32L16 64c0-17.7 14.3-32 32-32s32 14.3 32 32l0 51.2L97.6 97.6c87.5-87.5 229.3-87.5 316.8 0s87.5 229.3 0 316.8s-229.3 87.5-316.8 0c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0c62.5 62.5 163.8 62.5 226.3 0s62.5-163.8 0-226.3s-163.8-62.5-226.3 0L125.7 160z"></path>
</svg>
</button><button class="panzoom-max panzoom-button">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M32 32C14.3 32 0 46.3 0 64l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L32 32zM64 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 32c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM448 352c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96z"></path>
</svg>
</button><button class="panzoom-min panzoom-button panzoom-hidden">
<svg class="panzoom-icon" viewBox="0 0 448 512" xmlns="http://www.w3.org/2000/svg">
<path d="M160 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 64-64 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l96 0c17.7 0 32-14.3 32-32l0-96zM32 320c-17.7 0-32 14.3-32 32s14.3 32 32 32l64 0 0 64c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96c0-17.7-14.3-32-32-32l-96 0zM352 64c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-64 0 0-64zM320 320c-17.7 0-32 14.3-32 32l0 96c0 17.7 14.3 32 32 32s32-14.3 32-32l0-64 64 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-96 0z"></path>
</svg>
</button>
</nav>
<pre class="mermaid"><code>flowchart LR
    subgraph "Genome-wide Output"
        GW_TFBS["All TFBS regions"]
        GW_ATAC["All ATAC regions"]
    end

    subgraph "Panel-specific Output"
        PS_TFBS["xs1/xs2 TFBS regions"]
        PS_ATAC["xs1/xs2 ATAC regions"]
    end

    GW_TFBS --&gt; TSV1["sample.TFBS.tsv"]
    GW_ATAC --&gt; TSV2["sample.ATAC.tsv"]
    PS_TFBS --&gt; TSV3["sample.TFBS.ontarget.tsv"]
    PS_ATAC --&gt; TSV4["sample.ATAC.ontarget.tsv"]
</code></pre>
<div class="panzoom-info-box panzoom-info-box-bottom panzoom-hidden">
Use mouse to pan and zoom
</div>
</div>
<h3 id="usage_1">Usage</h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="c1"># With --assay → auto-loads panel-specific region files</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-r<span class="w"> </span>hg19.fa<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span>--assay<span class="w"> </span>xs2
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="c1"># Or with target regions → enables panel mode</span>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>krewlyzer<span class="w"> </span>run-all<span class="w"> </span>-i<span class="w"> </span>sample.bam<span class="w"> </span>-r<span class="w"> </span>hg19.fa<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a><span class="w">    </span>-T<span class="w"> </span>msk-access-v2.targets.bed
</code></pre></div>
<h3 id="output-files-panel-mode">Output Files (Panel Mode)</h3>
<table>
<thead>
<tr>
<th>File</th>
<th>Description</th>
<th><abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> Correction</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>{sample}.TFBS.tsv</code></td>
<td>All 808 TFs (genome-wide)</td>
<td>Off-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model</td>
</tr>
<tr>
<td><code>{sample}.TFBS.ontarget.tsv</code></td>
<td>TFs overlapping panel</td>
<td><strong>On-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model</strong></td>
</tr>
<tr>
<td><code>{sample}.TFBS.sync.tsv</code></td>
<td>Detailed size distributions</td>
<td>-</td>
</tr>
<tr>
<td><code>{sample}.TFBS.ontarget.sync.tsv</code></td>
<td>Panel size distributions</td>
<td>-</td>
</tr>
<tr>
<td><code>{sample}.ATAC.tsv</code></td>
<td>All 23 cancer types</td>
<td>Off-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model</td>
</tr>
<tr>
<td><code>{sample}.ATAC.ontarget.tsv</code></td>
<td>Cancer types in panel</td>
<td><strong>On-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> model</strong></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>On-target outputs use on-target <abbr title="Guanine-Cytosine content - Percentage of G and C nucleotides">GC</abbr> correction factors when available,
providing better accuracy for capture-biased data.</p>
</div>
<hr>
<h2 id="pon-normalization"><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> Normalization</h2>
<p>With a <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> model, raw entropy is converted to Z-scores:</p>
<div class="arithmatex">\[
Z = \frac{\text{entropy} - \mu_{\text{PON}}}{\sigma_{\text{PON}}}
\]</div>
<h3 id="building-pon-with-tfbsatac">Building <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> with <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>/<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr></h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>krewlyzer<span class="w"> </span>build-pon<span class="w"> </span>-i<span class="w"> </span>samples.txt<span class="w"> </span>-r<span class="w"> </span>hg19.fa<span class="w"> </span>-o<span class="w"> </span>healthy.pon.parquet
</code></pre></div>
<p>The <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> model stores:
- <code>tfbs_baseline</code>: Per-TF mean/std entropy from healthy samples
- <code>atac_baseline</code>: Per-cancer-type mean/std entropy from healthy samples</p>
<h3 id="applying-pon">Applying <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>krewlyzer<span class="w"> </span>region-entropy<span class="w"> </span>-i<span class="w"> </span>sample.bed.gz<span class="w"> </span>-o<span class="w"> </span>out/<span class="w"> </span><span class="se">\</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a><span class="w">    </span>-P<span class="w"> </span>healthy.pon.parquet
</code></pre></div>
<p><strong>Output with <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>:</strong>
</p><div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>label   count   mean_size   entropy   z_score
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a>CTCF    1234    167.2       5.23      1.45
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a>FOXA1   892     165.8       4.98      -0.32
</code></pre></div><p></p>
<hr>
<h2 id="output-format">Output Format</h2>
<h3 id="tfbs-output-sampletfbstsv"><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> Output: <code>{sample}.TFBS.tsv</code></h3>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>label</code></td>
<td>TEXT</td>
<td>Transcription factor name (e.g., <abbr title="CCCTC-binding factor - Chromatin architecture protein">CTCF</abbr>, FOXA1)</td>
</tr>
<tr>
<td><code>count</code></td>
<td>INT</td>
<td>Number of fragments overlapping TF regions</td>
</tr>
<tr>
<td><code>mean_size</code></td>
<td>FLOAT</td>
<td>Mean fragment size at these regions</td>
</tr>
<tr>
<td><code>entropy</code></td>
<td>FLOAT</td>
<td>Shannon entropy of size distribution (bits)</td>
</tr>
<tr>
<td><code>z_score</code></td>
<td>FLOAT</td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized z-score (0 if no <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>)</td>
</tr>
</tbody>
</table>
<h3 id="atac-output-sampleatactsv"><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> Output: <code>{sample}.ATAC.tsv</code></h3>
<table>
<thead>
<tr>
<th>Column</th>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>label</code></td>
<td>TEXT</td>
<td>Cancer type (e.g., BRCA, LUAD, COAD)</td>
</tr>
<tr>
<td><code>count</code></td>
<td>INT</td>
<td>Number of fragments overlapping cancer peaks</td>
</tr>
<tr>
<td><code>mean_size</code></td>
<td>FLOAT</td>
<td>Mean fragment size at these regions</td>
</tr>
<tr>
<td><code>entropy</code></td>
<td>FLOAT</td>
<td>Shannon entropy of size distribution (bits)</td>
</tr>
<tr>
<td><code>z_score</code></td>
<td>FLOAT</td>
<td><abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>-normalized z-score (0 if no <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr>)</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="interpretation">Interpretation</h2>
<h3 id="entropy-values">Entropy Values</h3>
<table>
<thead>
<tr>
<th>Range</th>
<th>Interpretation</th>
<th>Clinical Significance</th>
</tr>
</thead>
<tbody>
<tr>
<td>0-2</td>
<td>Very low entropy</td>
<td>Single dominant fragment size</td>
</tr>
<tr>
<td>2-4</td>
<td>Low entropy</td>
<td>Few distinct sizes</td>
</tr>
<tr>
<td>4-6</td>
<td>Moderate entropy</td>
<td>Normal healthy range</td>
</tr>
<tr>
<td>6-8</td>
<td>High entropy</td>
<td>Many distinct sizes</td>
</tr>
<tr>
<td>&gt; 8</td>
<td>Very high entropy</td>
<td>Possible tumor signal</td>
</tr>
</tbody>
</table>
<h3 id="z-scores">Z-Scores</h3>
<table>
<thead>
<tr>
<th>Z-Score</th>
<th>Interpretation</th>
</tr>
</thead>
<tbody>
<tr>
<td>-2 to +2</td>
<td>Within normal range</td>
</tr>
<tr>
<td>+2 to +3</td>
<td>Elevated (possible cancer signal)</td>
</tr>
<tr>
<td>&gt; +3</td>
<td>Significantly elevated</td>
</tr>
<tr>
<td>&lt; -2</td>
<td>Significantly reduced</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="algorithm-details">Algorithm Details</h2>
<h3 id="rust-backend-region_entropyrs">Rust Backend (<code>region_entropy.rs</code>)</h3>
<p>Following the methodology from Helzer et al.:</p>
<ol>
<li><strong>Load regions</strong>: Read <abbr title="Browser Extensible Data - Genomic interval file format">BED</abbr>.gz with label in 4th column</li>
<li><strong>Intersect fragments</strong>: For each region, collect fragments with minimum 1bp overlap</li>
<li><strong>Build histograms</strong>: Count fragments per size (20-500bp) per label</li>
<li><strong>Compute entropy</strong>: Shannon entropy from normalized histogram</li>
<li><strong>Output</strong>: <abbr title="Tab-Separated Values - Plain text tabular data format">TSV</abbr> with label, count, mean_size, entropy</li>
</ol>
<h3 id="python-processing-region_entropy_processorpy">Python Processing (<code>region_entropy_processor.py</code>)</h3>
<ol>
<li><strong>Load raw output</strong>: Read Rust-generated <abbr title="Tab-Separated Values - Plain text tabular data format">TSV</abbr></li>
<li><strong>Apply <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> baseline</strong>: If model provided, compute Z-scores</li>
<li><strong>Write final output</strong>: <abbr title="Tab-Separated Values - Plain text tabular data format">TSV</abbr> with z_score column</li>
</ol>
<hr>
<h2 id="performance">Performance</h2>
<table>
<thead>
<tr>
<th>Dataset</th>
<th>Regions</th>
<th>Time</th>
<th>Memory</th>
</tr>
</thead>
<tbody>
<tr>
<td><abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr> (808 TFs)</td>
<td>~4M regions</td>
<td>~30s</td>
<td>~500MB</td>
</tr>
<tr>
<td><abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> (23 types)</td>
<td>~700K regions</td>
<td>~20s</td>
<td>~400MB</td>
</tr>
</tbody>
</table>
<hr>
<h2 id="citation">Citation</h2>
<p>If you use this feature, please cite:</p>
<div class="admonition quote">
<p class="admonition-title">Reference</p>
<p><strong>Helzer KT, Sharifi MN, Sperger JM, et al.</strong> Analysis of <abbr title="Cell-free DNA - Circulating DNA fragments in blood plasma">cfDNA</abbr> fragmentomics metrics and commercial targeted sequencing panels. <em>Nat Commun</em> <strong>16</strong>, 9122 (2025). https://doi.org/10.1038/s41467-025-64153-z</p>
</div>
<p><strong>Data source:</strong>
- GitHub: <a href="https://github.com/Zhao-Lab-UW-DHO/fragmentomics_metrics/">Zhao-Lab-UW-DHO/fragmentomics_metrics</a></p>
<hr>
<h2 id="see-also">See Also</h2>
<ul>
<li><a href="../../core/extract/">Extract</a> – Generate input .bed.gz files</li>
<li><a href="../../../guides/building-pon/">Build <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr></a> – Create <abbr title="Panel of Normals - Reference model from healthy plasma samples">PON</abbr> models with <abbr title="Transcription Factor Binding Sites - DNA sequences bound by TFs">TFBS</abbr>/<abbr title="Assay for Transposase-Accessible Chromatin - Open chromatin assay">ATAC</abbr> baselines</li>
<li><a href="../ocf/"><abbr title="Orientation-aware cfDNA Fragmentation - Fragment end patterns near open chromatin">OCF</abbr></a> – Related open chromatin feature</li>
<li><a href="../../../resources/citation/#region-entropy">Citation</a> – All scientific references</li>
</ul>
<!-- Abbreviation definitions for Krewlyzer documentation -->
<!-- These provide hover tooltips for technical acronyms -->
<aside class="md-source-file">
<span class="md-source-file__fact">
<span class="md-icon" title="Last update">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M21 13.1c-.1 0-.3.1-.4.2l-1 1 2.1 2.1 1-1c.2-.2.2-.6 0-.8l-1.3-1.3c-.1-.1-.2-.2-.4-.2m-1.9 1.8-6.1 6V23h2.1l6.1-6.1zM12.5 7v5.2l4 2.4-1 1L11 13V7zM11 21.9c-5.1-.5-9-4.8-9-9.9C2 6.5 6.5 2 12 2c5.3 0 9.6 4.1 10 9.3-.3-.1-.6-.2-1-.2s-.7.1-1 .2C19.6 7.2 16.2 4 12 4c-4.4 0-8 3.6-8 8 0 4.1 3.1 7.5 7.1 7.9l-.1.2z"></path></svg>
</span>
<span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="February 6, 2026 07:09:13 UTC"><span class="timeago" datetime="2026-02-06T07:09:13+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="February 6, 2026 07:09:13 UTC">2026-02-06</span>
</span>
<span class="md-source-file__fact">
<span class="md-icon" title="Created">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M14.47 15.08 11 13V7h1.5v5.25l3.08 1.83c-.41.28-.79.62-1.11 1m-1.39 4.84c-.36.05-.71.08-1.08.08-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8c0 .37-.03.72-.08 1.08.69.1 1.33.32 1.92.64.1-.56.16-1.13.16-1.72 0-5.5-4.5-10-10-10S2 6.5 2 12s4.47 10 10 10c.59 0 1.16-.06 1.72-.16-.32-.59-.54-1.23-.64-1.92M18 15v3h-3v2h3v3h2v-3h3v-2h-3v-3z"></path></svg>
</span>
<span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-timeago" title="February 6, 2026 07:09:13 UTC"><span class="timeago" datetime="2026-02-06T07:09:13+00:00" locale="en"></span></span><span class="git-revision-date-localized-plugin git-revision-date-localized-plugin-iso_date" title="February 6, 2026 07:09:13 UTC">2026-02-06</span>
</span>
</aside>
</article>
</div>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8z"></path></svg>
  Back to top
</button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
<div class="md-copyright__highlight">
      Developed by <a href="https://github.com/rhshah">Ronak H Shah</a> @  <a href="https://www.mskcc.org">MSKCC</a> |  Built with <a href="https://antigravity.google/">Antigravity</a> |  <a href="https://github.com/msk-access/krewlyzer/blob/main/LICENSE">AGPL-3.0</a>
</div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"annotate": null, "base": "../../..", "features": ["navigation.instant", "navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.indexes", "navigation.path", "toc.follow", "content.code.copy", "content.code.annotate"], "search": "../../../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
<script src="../../../assets/javascripts/bundle.79ae519e.min.js"></script>
<script src="../../../js/timeago.min.js"></script>
<script src="../../../js/timeago_mkdocs_material.js"></script>
<script src="../../../javascripts/mathjax.js"></script>
<script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
<script src="../../../assets/javascripts/panzoom.min.js"></script>
<script src="../../../assets/javascripts/zoompan.js"></script>
<script type="module">import mermaid from "https://unpkg.com/mermaid@11.4.0/dist/mermaid.esm.min.mjs";
window.mermaidConfig = {default: {
    startOnLoad: false
}};</script>
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
document$.subscribe(()=>{ lightbox.reload(); });
</script></body></html>